--- lastmp.orig	2007-03-29 19:29:16.000000000 +0300
+++ lastmp	2007-04-17 00:00:20.000000000 +0300
@@ -168,7 +168,7 @@
         try:
             sub = song.dict()
             self.log.info('Sending to daemon')
-            lastfm.submit([sub])
+            lastfm.submit([sub], conf.spool_path)
         except ValueError, e:
             self.log.error('Missing tags: %s' % e)
         except IOError:
--- lastfmsubmitd.orig	2006-09-15 03:25:04.000000000 +0300
+++ lastfmsubmitd	2007-04-17 00:00:35.000000000 +0300
@@ -111,7 +111,7 @@
         for f in self.files:
             os.unlink(f)
         if self.subs:
-            newfile = lastfm.submit(self.subs)
+            newfile = lastfm.submit(self.subs, conf.spool_path)
             self.files = [newfile]
         else:
             self.files = []
