--- ./src/core/core.pro.orig	2007-09-22 01:52:47.000000000 +0300
+++ ./src/core/core.pro	2007-09-22 12:49:34.000000000 +0300
@@ -3,7 +3,8 @@
 PRECOMPILED_HEADER = precompile.h 
 
 LIBS += -ltraversocommands \
-        -ltraversoaudiobackend
+        -ltraversoaudiobackend \
+        -ltraversoaudiofileio
 
 INCLUDEPATH += ../commands \
 	../engine \
--- ./src/engine/engine.pro.orig	2007-09-22 01:52:47.000000000 +0300
+++ ./src/engine/engine.pro	2007-09-22 12:42:36.000000000 +0300
@@ -47,6 +47,8 @@
 !contains(DEFINES, JACK_SUPPORT){
     SOURCES -= JackDriver.cpp
     HEADERS -= JackDriver.h
+} else{
+    LIBS += -ljack
 }
 
 !contains(DEFINES, ALSA_SUPPORT){
--- ./src/audiofileio/audiofileio.pro.orig	2007-09-22 01:52:47.000000000 +0300
+++ ./src/audiofileio/audiofileio.pro	2007-09-22 13:42:07.000000000 +0300
@@ -7,6 +7,13 @@
 QMAKE_LIBDIR = ../../lib 
 TARGET = traversoaudiofileio 
 DESTDIR = ../../lib 
+LIBS += -lsamplerate \
+    -lsndfile \
+    -lvorbisfile \
+    -lmad \
+    -lmp3lame \
+    -lFLAC \
+    -lwavpack
 
 TEMPLATE = lib 
 
--- ./src/plugins/plugins.pro.orig	2007-09-22 01:52:47.000000000 +0300
+++ ./src/plugins/plugins.pro	2007-09-22 12:46:31.000000000 +0300
@@ -13,6 +13,7 @@
 	native \
 	../3rdparty/slv2
 
+LIBS += -lfftw3
 SOURCES += PluginChain.cpp \
 Plugin.cpp \
 PluginManager.cpp \
