--- a/xsd/xsd/buildfile
+++ b/xsd/xsd/buildfile
@@ -7,14 +7,14 @@ import libs += libxerces-c%lib{xerces-c}
 
 options_topics = options cxx/options cxx/parser/options cxx/tree/options
 
-./: exe{xsd}: {hxx ixx txx cxx}{** -{$options_topics} -version} \
+./: exe{codesynthesis-xsd}: {hxx ixx txx cxx}{** -{$options_topics} -version} \
               {hxx ixx cxx}{$options_topics}                    \
               {hxx}{version}                                    \
               $libs
 
 # Target metadata, see also --build2-metadata in xsd.cxx.
 #
-exe{xsd}:
+exe{codesynthesis-xsd}:
 {
   export.metadata = 1 xsd
   xsd.name = [string] xsd
--- a/xsd/doc/buildfile
+++ b/xsd/doc/buildfile
@@ -45,7 +45,7 @@ import? ps2pdf14 = ps2pdf14%exe{ps2pdf14
 #
 if ($cli != [null])
 {
-  ./: {man1 xhtml}{xsd}
+  ./: {man1 xhtml}{codesynthesis-xsd}
 
   man_topics = $(src_root)/xsd/cxx/cli{options tree/options parser/options}
 
@@ -57,7 +57,7 @@ if ($cli != [null])
                 --suppress-undocumented          \
                 --exclude-base
 
-  man1{xsd}: $man_topics            \
+  man1{codesynthesis-xsd}: $man_topics            \
              man1{xsd-prologue      \
                   xsd-tree-header   \
                   xsd-parser-header \
@@ -91,7 +91,7 @@ if ($cli != [null])
          $path($<[2]) >>$path($>)
   }}
 
-  xhtml{xsd}: $man_topics             \
+  xhtml{codesynthesis-xsd}: $man_topics             \
               xhtml{xsd-prologue      \
                     xsd-tree-header   \
                     xsd-parser-header \
