--- a/src/babelfont/convertors/truetype.py
+++ b/src/babelfont/convertors/truetype.py
@@ -3,7 +3,7 @@ from babelfont import *
 from fontTools.fontBuilder import FontBuilder
 from fontTools.pens.ttGlyphPen import TTGlyphPen
 from fontTools.pens.recordingPen import RecordingPen
-from cu2qu.ufo import glyphs_to_quadratic
+from fontTools.cu2qu.ufo import glyphs_to_quadratic
 from fontTools.misc.timeTools import epoch_diff, timestampSinceEpoch
 from fontTools.ttLib.tables.TupleVariation import TupleVariation
 from babelfont.fontFilters.featureWriters import build_all_features
