--- a/pyproject.toml
+++ b/pyproject.toml
@@ -12,7 +12,6 @@ fonttools = ">=4.21.1"
 ufoLib2 = ">=0.11.1"
 openstep-plist = ">=0.2.2"
 glyphsLib = ">=5.3.2"
-cu2qu = "^1.6.7"
 fontfeatures = "^1.0.6"
 
 [tool.poetry.dev-dependencies]
--- a/src/babelfont/convertors/truetype.py
+++ b/src/babelfont/convertors/truetype.py
@@ -3,7 +3,7 @@ from babelfont import *
 from fontTools.fontBuilder import FontBuilder
 from fontTools.pens.ttGlyphPen import TTGlyphPen
 from fontTools.pens.recordingPen import RecordingPen
-from cu2qu.ufo import glyphs_to_quadratic
+from fontTools.cu2qu.ufo import glyphs_to_quadratic
 from fontTools.misc.timeTools import epoch_diff, timestampSinceEpoch
 from fontTools.ttLib.tables.TupleVariation import TupleVariation
 from babelfont.fontFilters.featureWriters import build_all_features
