Index: server/mp4live/Makefile.am
===================================================================
RCS file: /cvsroot/mpeg4ip/mpeg4ip/server/mp4live/Makefile.am,v
retrieving revision 1.61
diff -u -r1.61 Makefile.am
--- server/mp4live/Makefile.am	30 May 2006 19:48:12 -0000	1.61
+++ server/mp4live/Makefile.am	14 Jun 2006 21:22:18 -0000
@@ -170,6 +170,6 @@
 	@SRTPLIB@ \
 	-lpthread \
 	@SDL_LIBS@ $(GUILIBS) @FFMPEG_LIB@ @LIBVORBIS_LIB@ $(XVID_LIB) \
-	@X264_LIB@ 
+	@X264_LIB@ @ALSA_LIBS@
 
 EXTRA_DIST = $(ed2) video_v4l_source.cpp 
