Index: configure.in
===================================================================
RCS file: /cvsroot/mpeg4ip/mpeg4ip/configure.in,v
retrieving revision 1.154
diff -u -r1.154 configure.in
--- configure.in	17 Dec 2004 19:32:40 -0000	1.154
+++ configure.in	22 Dec 2004 08:43:18 -0000
@@ -367,8 +367,8 @@
 if test x$disable_ffmpeg = xfalse; then
   if test x$have_ffmpeg_arg = xtrue; then
      AC_MSG_NOTICE([have enable ffmpeg $enable_ffmpeg])
-     FFMPEG_INC=-I${enable_ffmpeg}/libavcodec
-     FFMPEG_LIB="${enable_ffmpeg}/libavcodec/libavcodec.a -lz"
+     FFMPEG_INC=-I/usr/include/ffmpeg
+     FFMPEG_LIB="-lavcodec -lz"
      saveCFLAGS="$CFLAGS"
      CFLAGS="$CFLAGS $FFMPEG_INC"
      AC_CHECK_TYPES([AVRational], , , [#include <avcodec.h>])
Index: util/testfaac.sh
===================================================================
RCS file: /cvsroot/mpeg4ip/mpeg4ip/util/testfaac.sh,v
retrieving revision 1.2
diff -u -r1.2 testfaac.sh
--- util/testfaac.sh	6 Oct 2003 23:49:35 -0000	1.2
+++ util/testfaac.sh	22 Dec 2004 08:55:10 -0000
@@ -9,11 +9,8 @@
 do
   word=`echo $i | tr '[a-z]' '[A-Z]'`
   if test $found_libfaac = 'no'; then
-     if test $word = "LIBFAAC"; then
+     if test $word = "FAAC"; then
         found_libfaac=yes
-     fi
-  else
-     if test $word = "VERSION"; then
         echo "yes"
         exit 0
      fi
Index: player/lib/audio/faad/Makefile.am
===================================================================
RCS file: /cvsroot/mpeg4ip/mpeg4ip/player/lib/audio/faad/Makefile.am,v
retrieving revision 1.11
diff -u -r1.11 Makefile.am
--- player/lib/audio/faad/Makefile.am	12 Feb 2004 18:13:38 -0000	1.11
+++ player/lib/audio/faad/Makefile.am	6 May 2005 22:33:35 -0000
@@ -33,6 +33,6 @@
 	transfo.h\
 	util.h
 INCLUDES = -I$(top_srcdir)/include
-AM_CFLAGS = -O2 -Werror -Wall -fexceptions -fno-strict-aliasing
+AM_CFLAGS = -O2 -Wall -fexceptions -fno-strict-aliasing
 
 EXTRA_DIST = COPYING libfaad60.dsp README
