--- configure.in.orig	2006-10-24 13:46:58.000000000 +0300
+++ configure.in	2006-10-24 13:46:46.000000000 +0300
@@ -33,9 +33,6 @@
    ;;
 esac
 
-if test $have_macosx = false; then
-   AC_CONFIG_SUBDIRS(lib/SDLAudio)
-fi
 AM_CONDITIONAL(HAVE_MAC_OSX, $have_macosx)
 AC_CONFIG_SUBDIRS(common/video/iso-mpeg4)
 
--- lib/Makefile.am.orig	2006-10-05 01:08:42.000000000 +0300
+++ lib/Makefile.am	2006-10-24 13:48:33.000000000 +0300
@@ -2,7 +2,7 @@
   extradir = 
 else
 if PLAYER
-  extradir = SDLAudio
+  extradir = 
 else
   extradir = 
 endif
--- player/src/audio_sdl.h.orig	2006-02-18 02:57:16.000000000 +0200
+++ player/src/audio_sdl.h	2006-10-24 15:15:00.000000000 +0300
@@ -27,7 +27,7 @@
 #define __AUDIO_SDL_H__ 1
 
 #include "audio_buffer.h"
-#if 1
+#if 0
 #include "Our_SDL_audio.h"
 #define SDL_AUDIODELAY()  Our_SDL_AudioDelay()
 #define SDL_AUDIOINIT   Our_SDL_AudioInit
