Index: configure.ac
===================================================================
--- configure.ac	(revision 16690)
+++ configure.ac	(working copy)
@@ -4234,13 +4234,15 @@
   [  --enable-portaudio      Portaudio library support (default disabled)],
   [if test "${enable_portaudio}" = "yes"
    then
+   PKG_CHECK_MODULES( PORTAUDIO, portaudio-2.0, [
      VLC_ADD_PLUGINS([portaudio])
-     VLC_ADD_CXXFLAGS([portaudio],[])
+     VLC_ADD_CFLAGS([portaudio],[$PORTAUDIO_CFLAGS])
      if test "${SYS}" = "mingw32"; then
         VLC_ADD_LDFLAGS([portaudio],[-lportaudio -lwinmm -lole32])
      else
-        VLC_ADD_LDFLAGS([portaudio],[-lportaudio])
+        VLC_ADD_LDFLAGS([portaudio],[$PORTAUDIO_LIBS])
      fi
+  ])
    fi])
 
 dnl
