Index: bootstrap
===================================================================
--- bootstrap	(revision 20823)
+++ bootstrap	(working copy)
@@ -22,8 +22,6 @@
 CYGWIN=binmode
 export CYGWIN
 
-set -e
-set -x
 
 ##
 ## Check for various tools
@@ -60,6 +58,10 @@
   exit 1
 fi
 
+if true; then
+GETTEXT=yes
+PKGCONFIG=yes
+else
 # Check for autoconf
 rm -f m4/autoconf260.m4
 case "$(autoconf --version|head -n 1)" in
@@ -136,6 +138,7 @@
 else
   PKGCONFIG=no
 fi
+fi
 
 aclocal=aclocal${amsuff}
 automake=automake${amsuff}
@@ -427,7 +430,6 @@
 ###
 ###  classic bootstrap stuff
 ###
-set -x
 
 # remove autotools cruft
 rm -f aclocal.m4 configure config.log config.h config.h.in
@@ -453,6 +455,9 @@
   printf "" > ABOUT-NLS
 fi
 
+if true; then
+autopoint --force > /dev/null 2>&1 || exit 1
+else
 # Libtoolize directory
 ${libtoolize} --copy --force
 if test -f "ltmain.sh"; then
@@ -483,6 +488,7 @@
 ${autoconf}
 ${autoheader}
 ${automake} --add-missing --copy -Wall
+fi
 
 ##
 ##  files which need to be regenerated
