Index: modules/codec/flac.c
===================================================================
--- modules/codec/flac.c	(revision 16087)
+++ modules/codec/flac.c	(working copy)
@@ -725,10 +725,12 @@
     case FLAC__STREAM_DECODER_ABORTED:
         msg_Warn( p_dec, "the decoder was aborted by the read callback." );
         break;
-    case FLAC__STREAM_DECODER_UNPARSEABLE_STREAM:
+    #if 0
+    case FLAC__STREAM_DECODER_ERROR_STATUS_UNPARSEABLE_STREAM:
         msg_Warn( p_dec, "the decoder encountered reserved fields in use "
                  "in the stream." );
         break;
+    #endif
     case FLAC__STREAM_DECODER_MEMORY_ALLOCATION_ERROR:
         msg_Err( p_dec, "error when allocating memory." );
         break;
