Index: configure.ac
===================================================================
--- configure.ac	(revision 15710)
+++ configure.ac	(working copy)
@@ -2980,7 +2980,12 @@
         VLC_ADD_LDFLAGS([a52tofloat32],[${LDFLAGS_test} -la52])
         VLC_ADD_CPPFLAGS([a52tofloat32],[${CPPFLAGS_test}])
       ],[
+      PKG_CHECK_MODULES( LIBA52, liba52, 
+       [VLC_ADD_PLUGINS([a52tofloat32])
+        VLC_ADD_LDFLAGS([a52tofloat32],[$LIBA52_LIBS])
+        VLC_ADD_CFLAGS([a52tofloat32],[$LIBA52_CFLAGS])],[
         AC_MSG_ERROR([Could not find liba52 on your system: you may get it from http://liba52.sf.net/. Alternatively you can use --disable-a52 to disable the a52 plugin.])
+        ])
         ])
     ],[
       AC_MSG_ERROR([Could not find liba52 on your system: you may get it from http://liba52.sf.net/. Alternatively you can use --disable-a52 to disable the a52 plugin.])
