Index: modules/misc/freetype.c
===================================================================
--- modules/misc/freetype.c	(revision 18869)
+++ modules/misc/freetype.c	(working copy)
@@ -55,8 +55,8 @@
 #elif defined( WIN32 )
 #define DEFAULT_FONT "" /* Default font found at run-time */
 #else
-#define DEFAULT_FONT "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
-#define FC_DEFAULT_FONT "Serif Bold"
+#define DEFAULT_FONT "/usr/share/fonts/ttf-bitstream-vera/VeraBd.ttf"
+#define FC_DEFAULT_FONT "Sans Bold"
 #endif
 
 #if defined(HAVE_FRIBIDI)
