diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.4.4.orig/poppler/CairoOutputDev.cc poppler-0.4.4/poppler/CairoOutputDev.cc
--- poppler-0.4.4.orig/poppler/CairoOutputDev.cc	2005-12-12 17:24:01.000000000 -0500
+++ poppler-0.4.4/poppler/CairoOutputDev.cc	2006-01-19 12:42:12.000000000 -0500
@@ -189,7 +189,11 @@
 
 void CairoOutputDev::updateLineWidth(GfxState *state) {
   LOG(printf ("line width: %f\n", state->getTransformedLineWidth()));
-  cairo_set_line_width (cairo, state->getTransformedLineWidth());
+  if (state->getTransformedLineWidth() == 0.0) {
+    cairo_set_line_width (cairo, 72.0/300.0);
+  } else {
+    cairo_set_line_width (cairo, state->getTransformedLineWidth());
+  }
 }
 
 void CairoOutputDev::updateFillColor(GfxState *state) {
