diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.4.3.orig/pdf2xml.dtd poppler-0.4.3/pdf2xml.dtd
--- pdf2xml.dtd	1969-12-31 19:00:00.000000000 -0500
+++ pdf2xml.dtd	2006-01-16 17:44:43.000000000 -0500
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<!ELEMENT pdf2xml (page+)>
+<!ELEMENT page (fontspec*, text*)>
+<!ATTLIST page
+	number CDATA #REQUIRED
+	position CDATA #REQUIRED
+	top CDATA #REQUIRED
+	left CDATA #REQUIRED
+	height CDATA #REQUIRED
+	width CDATA #REQUIRED
+>
+<!ELEMENT fontspec EMPTY>
+<!ATTLIST fontspec
+	id CDATA #REQUIRED
+	size CDATA #REQUIRED
+	family CDATA #REQUIRED
+	color CDATA #REQUIRED
+>
+<!ELEMENT text (#PCDATA | b | i)*>
+<!ATTLIST text
+	top CDATA #REQUIRED
+	left CDATA #REQUIRED
+	width CDATA #REQUIRED
+	height CDATA #REQUIRED
+	font CDATA #REQUIRED
+>
+<!ELEMENT b (#PCDATA)>
+<!ELEMENT i (#PCDATA)>
