diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.4.4.orig/poppler/JBIG2Stream.cc poppler-0.4.4/poppler/JBIG2Stream.cc
--- poppler-0.4.4.orig/poppler/JBIG2Stream.cc	2006-01-10 13:53:54.000000000 -0500
+++ poppler-0.4.4/poppler/JBIG2Stream.cc	2006-01-19 12:35:17.000000000 -0500
@@ -7,6 +7,7 @@
 //========================================================================
 
 #include <config.h>
+#include <limits.h>
 
 #ifdef USE_GCC_PRAGMAS
 #pragma implementation
@@ -2924,6 +2925,11 @@
     return NULL;
   }
 
+  if (w < 0 || h <= 0 || w >= INT_MAX / h) {
+    error(-1, "invalid width/height");
+    return NULL;
+  }
+
   bitmap = new JBIG2Bitmap(0, w, h);
   bitmap->clearToZero();
 
