--- a/experimental/ffmpeg/SkVideoDecoder.cpp
+++ b/experimental/ffmpeg/SkVideoDecoder.cpp
@@ -372,7 +372,7 @@ bool SkVideoDecoder::loadStream(std::uni
         return false;
     }
 
-    AVCodec* codec;
+    const AVCodec* codec;
     fStreamIndex = av_find_best_stream(fFormatCtx, AVMEDIA_TYPE_VIDEO, -1, -1, &codec, 0);
     if (fStreamIndex < 0) {
         SkDebugf("av_find_best_stream failed %d\n", fStreamIndex);
--- a/experimental/ffmpeg/SkVideoEncoder.cpp
+++ b/experimental/ffmpeg/SkVideoEncoder.cpp
@@ -163,12 +163,12 @@ bool SkVideoEncoder::init(int fps) {
     SkASSERT(fFormatCtx);
     fFormatCtx->pb = fStreamCtx;
 
-    AVOutputFormat *output_format = fFormatCtx->oformat;
+    const AVOutputFormat *output_format = fFormatCtx->oformat;
 
     if (output_format->video_codec == AV_CODEC_ID_NONE) {
         return false;
     }
-    AVCodec* codec = avcodec_find_encoder(output_format->video_codec);
+    const AVCodec* codec = avcodec_find_encoder(output_format->video_codec);
     SkASSERT(codec);
 
     fStream = avformat_new_stream(fFormatCtx, codec);
