Index: ccmain/tfacep.h
===================================================================
--- ccmain/tfacep.h	(revision 72)
+++ ccmain/tfacep.h	(working copy)
@@ -103,7 +103,6 @@
   //extern int                                    enable_assoc;
   //extern int                                    chop_enable;
   //extern int                                    permute_only_top;
-  extern int display_ratings;
 
 };
 
Index: ccmain/control.cpp
===================================================================
--- ccmain/control.cpp	(revision 72)
+++ ccmain/control.cpp	(working copy)
@@ -148,7 +148,7 @@
 EXTERN double_VAR (test_pt_y, 99999.99, "ycoord");
 
 extern int MatcherDebugLevel;
-extern "C" { extern int display_ratings; }
+extern int display_ratings;
 extern int number_debug;
 extern int adjust_debug;
 /*
Index: cutil/globals.h
===================================================================
--- cutil/globals.h	(revision 72)
+++ cutil/globals.h	(working copy)
@@ -45,8 +45,6 @@
 extern int debugs[MAXPROC];      /*debug flags */
 extern int plots[MAXPROC];       /*plot flags */
 extern int corners[4];           /*corners of scan window */
-extern int optind;               /*option index */
-extern char *optarg;             /*option argument */
                                  /*image file name */
 extern char imagefile[FILENAMESIZE];
                                  /* main directory */
Index: training/mfTraining.cpp
===================================================================
--- training/mfTraining.cpp	(revision 72)
+++ training/mfTraining.cpp	(working copy)
@@ -31,6 +31,7 @@
 #include "emalloc.h"
 #include "featdefs.h"
 #include "tessopt.h"
+#include "globals.h"
 #include "ocrfeatures.h"
 #include "general.h"
 #include "clusttool.h"
Index: training/cnTraining.cpp
===================================================================
--- training/cnTraining.cpp	(revision 72)
+++ training/cnTraining.cpp	(working copy)
@@ -29,6 +29,7 @@
 #include "emalloc.h"
 #include "featdefs.h"
 #include "tessopt.h"
+#include "globals.h"
 #include "ocrfeatures.h"
 #include "general.h"
 #include "clusttool.h"
