--- a/client/widgets/MainAction.cpp
+++ b/client/widgets/MainAction.cpp
@@ -24,6 +24,7 @@
 
 #include <QtCore/QSettings>
 #include <QtGui/QPainter>
+#include <QtGui/QPainterPath>
 #include <QtGui/QPaintEvent>
 
 using namespace ria::qdigidoc4;
--- a/client/widgets/CheckBox.cpp
+++ b/client/widgets/CheckBox.cpp
@@ -22,6 +22,7 @@
 #include <QBrush>
 #include <QPaintEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyleOptionButton>
 
 CheckBox::CheckBox(QWidget *parent)
