Index: Makefile
===================================================================
--- Makefile	(revision 27482)
+++ Makefile	(working copy)
@@ -703,14 +703,13 @@
 
 ALLHEADERS = $(foreach dir,$(DIRS),$(wildcard $(dir)/*.h))
 
-PARTS = libavcodec \
+PARTS = \
         libavformat \
         libavutil \
-        libpostproc \
         libswscale \
 
 FFMPEGLIBS  = $(foreach part, $(PARTS), $(part)/$(part).a)
-FFMPEGFILES = $(foreach part, $(PARTS), $(part)/*.[chS] libavcodec/*/*.[chS])
+FFMPEGFILES = $(foreach part, $(PARTS), $(part)/*.[chS])
 
 
 
