Index: version.sh
===================================================================
--- version.sh	(revision 22932)
+++ version.sh	(working copy)
@@ -2,7 +2,7 @@
 
 test "$1" && extra="-$1"
 
-svn_revision=`LC_ALL=C svn info 2> /dev/null | grep Revision | cut -d' ' -f2`
+svn_revision="${ESVN_WC_REVISION}"
 test $svn_revision || svn_revision=`grep revision .svn/entries 2>/dev/null | cut -d '"' -f2`
 test $svn_revision || svn_revision=`sed -n -e '/^dir$/{n;p;q;}' .svn/entries 2>/dev/null`
 test $svn_revision || svn_revision=UNKNOWN
