Index: configure
===================================================================
--- configure	(revision 21332)
+++ configure	(working copy)
@@ -59,9 +59,9 @@
   echo >> "$TMPLOG"
   cat "$1" >> "$TMPLOG"
   echo >> "$TMPLOG"
-  echo "$_cc $CFLAGS $_inc_extra $_ld_static $_ld_extra $_libs_mplayer $_libs_mencoder -o $TMPO $@" >> "$TMPLOG"
+  echo "$_cc $CFLAGS $_inc_extra -o $TMPO $@" $_ld_static $_ld_extra $_libs_mplayer $_libs_mencoder >> "$TMPLOG"
   rm -f "$TMPO"
-  $_cc $CFLAGS $_inc_extra $_ld_static $_ld_extra $_libs_mplayer $_libs_mencoder -o "$TMPO" "$@" >> "$TMPLOG" 2>&1
+  $_cc $CFLAGS $_inc_extra -o "$TMPO" "$@" $_ld_static $_ld_extra $_libs_mplayer $_libs_mencoder >> "$TMPLOG" 2>&1
   TMP="$?"
   echo >> "$TMPLOG"
   echo "ldd $TMPO" >> "$TMPLOG"
