Index: configure
===================================================================
--- configure	(revision 308)
+++ configure	(working copy)
@@ -177,7 +177,7 @@
             ;;
         --enable-mp4-output)
             CFLAGS="$CFLAGS -DMP4_OUTPUT"
-            LDFLAGS="$LDFLAGS -lgpac_static"
+            LDFLAGS="$LDFLAGS -lgpac"
             if [ $SYS = CYGWIN -o $SYS = MINGW ]; then
                 LDFLAGS="$LDFLAGS -lwinmm"
             fi
