Index: gtk/Makefile
===================================================================
--- gtk/Makefile	(revision 622)
+++ gtk/Makefile	(working copy)
@@ -49,7 +49,7 @@
 EXTERNAL_DEPS= ../muxers.o ../matroska.o ../libx264.a
 
 
-all: $(ENCODE_BIN) $(TEST_BIN) $(MO_FILES) $(X264GTK_PC)
+all: $(ENCODE_BIN) $(MO_FILES) $(X264GTK_PC)
 
 # Already provides iconv/intl
 CPPFLAGS = -g `pkg-config --cflags gtk+-2.0 gthread-2.0` -I.. -DX264_DATA_DIR=\"${datadir}\"
