diff --git a/version.sh b/version.sh
index 37b9908..4fd0ede 100755
--- a/version.sh
+++ b/version.sh
@@ -2,11 +2,9 @@
 git-rev-list HEAD | sort > config.git-hash
 LOCALVER=`wc -l config.git-hash | awk '{print $1}'`
 if [ $LOCALVER \> 1 ] ; then
-    VER=`git-rev-list origin/master | sort | join config.git-hash - | wc -l | awk '{print $1}'`
+    VER=`git-rev-list master | sort | join config.git-hash - | wc -l | awk '{print $1}'`
     if [ $VER != $LOCALVER ] ; then
         VER="$VER+$(($LOCALVER-$VER))"
-    elif git-status | grep -q "modified:" ; then
-        VER="${VER}M"
     fi
     VER="$VER $(git-rev-list HEAD -n 1 | head -c 7)"
     echo "#define X264_VERSION \" r$VER\"" >> config.h
