--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,5 +65,3 @@ include(desktopapp_configuration)
 # Load the MEGA targets
 add_subdirectory(src)
 
-include(get_clang_format)
-get_clang_format()
--- a/cmake/modules/desktopapp_configuration.cmake
+++ b/cmake/modules/desktopapp_configuration.cmake
@@ -29,8 +29,6 @@ if(UNIX AND NOT APPLE)
     # Note: using cmake --install --prefix /some/prefix will keep the RPATH as configured above.
     #       Used for building packages: in which install dir is a path construction folder that will not be there in packages
 
-    set(CMAKE_INSTALL_LIBDIR "opt/megasync/lib") # override default "lib" from GNUInstallDirs
-    set(CMAKE_INSTALL_BINDIR "usr/bin") # override default "bin" from GNUInstallDirs
 
     # Override CMAKE_INSTALL_PREFIX
     if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT) # In consecutive runs it will always be undef/false
@@ -38,7 +36,6 @@ if(UNIX AND NOT APPLE)
         set(CMAKE_INSTALL_PREFIX "" CACHE PATH "" FORCE) # Set value in cache for consecutive runs
     endif()
 
-    list(APPEND CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
 
     message(STATUS "Current installation path for ${PROJECT_NAME} project: \"${CMAKE_INSTALL_PREFIX}\"")
 
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,7 +3,6 @@
 # Load MEGA Desktop app project
 
 if(ENABLE_DESKTOP_APP OR ENABLE_DESKTOP_UPDATE_GEN)
-    add_subdirectory(MEGASync/mega)
 endif()
 
 if(ENABLE_DESKTOP_APP)
--- a/src/MEGAAutoTests/UnitTests/CMakeLists.txt
+++ b/src/MEGAAutoTests/UnitTests/CMakeLists.txt
@@ -2,6 +2,9 @@ cmake_minimum_required(VERSION 3.18)
 cmake_policy(SET CMP0091 NEW)
 
 find_package(Qt5 REQUIRED COMPONENTS Widgets Core Gui Network Qml Quick QuickWidgets)
+find_package(sdklib REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(breakpad REQUIRED IMPORTED_TARGET breakpad breakpad-client)
 
 find_package(Catch2 REQUIRED)
 find_package(TrompeLoeil REQUIRED)
@@ -20,6 +23,7 @@ set_target_properties(UnitTests
 
 target_compile_definitions(UnitTests
     PUBLIC
+    __STDC_FORMAT_MACROS
     $<$<BOOL:${WIN32}>:PUBLIC NOMINMAX WIN32_LEAN_AND_MEAN UNICODE>
     $<$<BOOL:${ENABLE_ISOLATED_GFX}>:ENABLE_SDK_ISOLATED_GFX>
     $<$<BOOL:${USE_BREAKPAD}>:USE_BREAKPAD>
@@ -40,7 +44,6 @@ set(UNIT_TEST_FILES
 )
 
 if(USE_BREAKPAD)
-    find_package(unofficial-breakpad CONFIG REQUIRED)
 
     set(CRASH_BACKEND_URL "$ENV{MEGA_CRASH_BACKEND_URL}" CACHE STRING "Crash backend URL")
     target_compile_definitions(UnitTests PRIVATE CRASH_BACKEND_URL="${CRASH_BACKEND_URL}")
@@ -83,12 +86,11 @@ target_link_libraries(UnitTests
     PRIVATE
     trompeloeil
     MEGA::SDKlib
-    MEGA::SDKQtBindings
+    SDKQtBindings
     $<$<BOOL:${WIN32}>:Qt5::WinExtras>
     $<$<BOOL:${APPLE}>:Qt5::MacExtras>
     $<$<BOOL:${UNIX}>:Qt5::Svg>
-    $<$<BOOL:${USE_BREAKPAD}>:unofficial::breakpad::libbreakpad>
-    $<$<BOOL:${USE_BREAKPAD}>:unofficial::breakpad::libbreakpad_client>
+    $<$<BOOL:${USE_BREAKPAD}>:PkgConfig::breakpad>
     Qt5::Widgets
     Qt5::Core
     Qt5::Gui
--- a/src/MEGASync/CMakeLists.txt
+++ b/src/MEGASync/CMakeLists.txt
@@ -82,7 +82,8 @@ set_target_properties(MEGAsync
 )
 
 
-find_package(Qt5 REQUIRED COMPONENTS Widgets Core Gui Network Qml Quick QuickWidgets LinguistTools)
+find_package(Qt5 REQUIRED COMPONENTS Widgets Core Gui Network Qml Quick QuickWidgets)
+find_package(Qt5LinguistTools REQUIRED)
 qt5_create_translation(QM_OUTPUT ${CMAKE_CURRENT_SOURCE_DIR} gui/translations/translation.source.ts
     OPTIONS -locations none -no-ui-lines -no-obsolete)
 
@@ -91,6 +92,7 @@ add_custom_target(generate_ts DEPENDS ${QM_OUTPUT})
 add_dependencies(MEGAsync generate_ts)
 
 target_include_directories(MEGAsync PRIVATE ${CMAKE_CURRENT_LIST_DIR})
+target_include_directories(MEGAsync PRIVATE ${CMAKE_CURRENT_LIST_DIR}/mega/bindings/qt)
 
 set(ExecutableTarget MEGAsync)
 
@@ -106,6 +108,7 @@ include(UserAttributesRequests/userattributesrequests.cmake)
 
 target_compile_definitions(MEGAsync
     PUBLIC
+    __STDC_FORMAT_MACROS
     QT_NO_CAST_FROM_ASCII QT_NO_CAST_TO_ASCII
     $<$<CONFIG:Debug>:LOG_TO_STDOUT LOG_TO_LOGGER CREATE_COMPATIBLE_MINIDUMPS>
     $<$<BOOL:${WIN32}>:PUBLIC NOMINMAX WIN32_LEAN_AND_MEAN UNICODE>
@@ -115,7 +118,6 @@ target_compile_definitions(MEGAsync
 
 # Load and link needed libraries for the Desktop App target
 if(USE_BREAKPAD)
-    find_package(unofficial-breakpad CONFIG REQUIRED)
 
     if(NOT DEFINED CACHE{CRASH_BACKEND_URL})
         set(CRASH_BACKEND_URL "" CACHE STRING "Crash backend URL")
@@ -139,17 +141,18 @@ else()
     find_package(Qt5 REQUIRED COMPONENTS Svg)
 endif()
 
+find_package(sdklib REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(breakpad IMPORTED_TARGET breakpad breakpad-client)
 message(STATUS "Building the MEGA Desktop App with Qt v${Qt5_VERSION} from ${Qt5_DIR}")
 
 target_link_libraries(MEGAsync
     PUBLIC
     MEGA::SDKlib
-    MEGA::SDKQtBindings
+    SDKQtBindings
     $<$<BOOL:${WIN32}>:Qt5::WinExtras>
     $<$<BOOL:${APPLE}>:Qt5::MacExtras>
     $<$<BOOL:${UNIX}>:Qt5::Svg>
-    $<$<BOOL:${USE_BREAKPAD}>:unofficial::breakpad::libbreakpad>
-    $<$<BOOL:${USE_BREAKPAD}>:unofficial::breakpad::libbreakpad_client>
     Qt5::Widgets
     Qt5::Core
     Qt5::Gui
@@ -157,6 +160,8 @@ target_link_libraries(MEGAsync
     Qt5::Qml
     Qt5::Quick
     Qt5::QuickWidgets
+    $<$<BOOL:${USE_BREAKPAD}>:PkgConfig::breakpad>
+    z
 )
 
 ## Adjust compilation flags for warnings and errors ##
@@ -192,13 +197,6 @@ if (UNIX AND NOT APPLE)
 
     set(vcpkg_lib_folder "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/$<$<CONFIG:Debug>:debug/>lib/")
 
-    install(DIRECTORY "${vcpkg_lib_folder}"
-        TYPE LIB # In CMAKE_INSTALL_LIBDIR directory
-        FILES_MATCHING
-        PATTERN "*.so*"
-        PATTERN "manual-link" EXCLUDE
-        PATTERN "pkgconfig" EXCLUDE
-    )
 
     if(DEPLOY_QT_LIBRARIES)
         include(desktopapp_deploy_qt)
