--- a/src/MEGAShellExtDolphin/megasync-plugin-overlay.cpp
+++ b/src/MEGAShellExtDolphin/megasync-plugin-overlay.cpp
@@ -161,8 +161,8 @@ public:
             if (!icon_path.isEmpty())
             {
                 // qDebug() << "MEGASYNCOVERLAYPLUGIN: icon_path =" << icon_path;
-                if (icon_path == "mega" ||
-                    (icon_path.contains("/usr/share/icons") && icon_path.contains("apps/mega.png")))
+                if (icon_path == QString::fromLatin1("mega") ||
+                    (icon_path.contains(QString::fromLatin1("/usr/share/icons")) && icon_path.contains(QString::fromLatin1("apps/mega.png"))))
                 {
                     // qDebug() << "MEGASYNCOVERLAYPLUGIN: Restore " << qStrURL << " to default icon";
                     // Restore to default icon
@@ -261,10 +261,10 @@ private:
         }
 
         QString iconPath;
-        KDesktopFile desktopFile(folderPath + "/.directory");
-        if (desktopFile.hasGroup("Desktop Entry"))
+        KDesktopFile desktopFile(folderPath + QString::fromLatin1("/.directory"));
+        if (desktopFile.hasGroup(QString::fromLatin1("Desktop Entry")))
         {
-            KConfigGroup group = desktopFile.group("Desktop Entry");
+            KConfigGroup group = desktopFile.group(QString::fromLatin1("Desktop Entry"));
             iconPath = group.readEntry("Icon", QString());
         }
 
@@ -280,12 +280,12 @@ private:
             return;
         }
 
-        QString desktopFilePath = folderPath + "/.directory";
+        QString desktopFilePath = folderPath + QString::fromLatin1("/.directory");
         KDesktopFile desktopFile(desktopFilePath);
 
-        if (desktopFile.hasGroup("Desktop Entry"))
+        if (desktopFile.hasGroup(QString::fromLatin1("Desktop Entry")))
         {
-            KConfigGroup group = desktopFile.group("Desktop Entry");
+            KConfigGroup group = desktopFile.group(QString::fromLatin1("Desktop Entry"));
 
             if (iconPath.isEmpty())
             {
