--- a/src/MEGA.pro
+++ b/src/MEGA.pro
@@ -18,9 +18,9 @@ win32 {
 unix:!macx {
     CONFIG(with_ext) {
         SUBDIRS += MEGAShellExtNautilus
+    }
+    CONFIG(with_thu) {
         SUBDIRS += MEGAShellExtThunar
-        SUBDIRS += MEGAShellExtDolphin
-        SUBDIRS += MEGAShellExtNemo
     }
 }
 
--- a/src/MEGAShellExtNautilus/MEGAShellExtNautilus.pro
+++ b/src/MEGAShellExtNautilus/MEGAShellExtNautilus.pro
@@ -1,6 +1,7 @@
 QT       -= core gui
 
-TARGET = MEGAShellExtNautilus
+TARGET = libMEGAShellExtNautilus
+CONFIG += plugin no_plugin_name_prefix
 TEMPLATE = lib
 
 SOURCES += mega_ext_module.c \
@@ -52,5 +53,4 @@ INSTALLS += emblems32  emblems64
 !contains(DEFINES, no_desktop) {
     update_cache.commands = gtk-update-icon-cache -f -t $${HICOLOR} || true
     update_cache.path = $${HICOLOR}
-    INSTALLS += update_cache
 }
--- a/src/MEGASync/MEGASync.pro
+++ b/src/MEGASync/MEGASync.pro
@@ -75,7 +75,6 @@ TEMPLATE = app
 #DEFINES += LOG_TO_LOGGER
 #DEFINES += LOG_TO_FILE
 #DEFINES += LOG_TO_STDOUT
-DEFINES += ENABLE_LOG_PERFORMANCE
 
 debug {
     CONFIG += console
--- a/src/MEGASync/control/control.pri
+++ b/src/MEGASync/control/control.pri
@@ -2,6 +2,7 @@ DEPENDPATH += $$PWD
 INCLUDEPATH += $$PWD
 
 QT       += network
+LIBS+=-lz
 
 SOURCES += $$PWD/HTTPServer.cpp \
     $$PWD/Preferences.cpp \
--- a/src/MEGASync/platform/platform.pri
+++ b/src/MEGASync/platform/platform.pri
@@ -49,7 +49,8 @@ unix:!macx {
     system(command -v lsb_release): version.commands = lsb_release -rs > $$version.target
 	version.files = $$version.target
 
-	INSTALLS += distro version
+	target.path = $$PREFIX/bin
+	INSTALLS += target
 
     SOURCES += $$PWD/linux/LinuxPlatform.cpp \
         $$PWD/linux/ExtServer.cpp \
@@ -82,7 +83,6 @@ unix:!macx {
         # desktop
         desktop.path = $$DESKTOP_DESTDIR/share/applications
         desktop.files = $$PWD/linux/data/megasync.desktop
-        desktop.commands = update-desktop-database &> /dev/null || true
         INSTALLS += desktop
 
         HICOLOR = $$DESKTOP_DESTDIR/share/icons/hicolor
