--- a/src/MEGAShellExtNautilus/MEGAShellExtNautilus.pro
+++ b/src/MEGAShellExtNautilus/MEGAShellExtNautilus.pro
@@ -1,6 +1,7 @@
 QT       -= core gui
 
-TARGET = MEGAShellExtNautilus
+TARGET = libMEGAShellExtNautilus
+CONFIG += plugin no_plugin_name_prefix
 TEMPLATE = lib
 
 SOURCES += mega_ext_module.c \
@@ -41,5 +42,4 @@ INSTALLS += emblems32  emblems64
 !contains(DEFINES, no_desktop) {
     update_cache.commands = gtk-update-icon-cache -f -t $${HICOLOR} || true
     update_cache.path = $$PWD
-    INSTALLS += update_cache
 }
--- a/src/MEGASync/platform/platform.pri
+++ b/src/MEGASync/platform/platform.pri
@@ -34,7 +34,8 @@ unix:!macx {
     system(command -v lsb_release): version.commands = lsb_release -rs > $$version.target
 	version.files = $$version.target
 
-	INSTALLS += distro version
+	target.path = $$PREFIX/bin
+	INSTALLS += target
 
     QT += dbus
     SOURCES += $$PWD/linux/LinuxPlatform.cpp \
@@ -62,7 +62,6 @@ unix:!macx {
         # desktop
         desktop.path = $$DESKTOP_DESTDIR/share/applications
         desktop.files = $$PWD/linux/data/megasync.desktop
-        desktop.commands = update-desktop-database &> /dev/null || true
         INSTALLS += desktop
 
         HICOLOR = $$DESKTOP_DESTDIR/share/icons/hicolor
--- a/src/MEGA.pro
+++ b/src/MEGA.pro
@@ -17,8 +17,9 @@ win32 {
 unix:!macx {
     CONFIG(with_ext) {
         SUBDIRS += MEGAShellExtNautilus
+    }
+    CONFIG(with_thu) {
         SUBDIRS += MEGAShellExtThunar
-        SUBDIRS += MEGAShellExtDolphin
     }
 }
 
