Index: Makefile
===================================================================
RCS file: /cvsroot/dvd-audio/tools/Makefile,v
retrieving revision 1.5
diff -u -r1.5 Makefile
--- Makefile	5 Apr 2005 18:47:29 -0000	1.5
+++ Makefile	13 Jul 2006 17:29:53 -0000
@@ -1,5 +1,5 @@
 CC=gcc
-CFLAGS=-Wall
+CFLAGS+=-Wall
 TARGETS=dump_audio_pp dump_aob dump_atsi vts2wav ats2wav wav2lpcm
 
 all: $(TARGETS)
Index: vts2wav.c
===================================================================
RCS file: /cvsroot/dvd-audio/tools/vts2wav.c,v
retrieving revision 1.1
diff -u -r1.1 vts2wav.c
--- vts2wav.c	4 Apr 2005 19:26:09 -0000	1.1
+++ vts2wav.c	13 Jul 2006 17:29:56 -0000
@@ -22,7 +22,7 @@
  */
 
 #include <stdio.h>
-#include <stdlib.h>
+#include <stdint.h>
 #include <string.h>
 #include <assert.h>
 #include <sys/types.h>
