--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,6 +38,11 @@ elseif(UNIX)
             DESTINATION "${CMAKE_INSTALL_LIBDIR}/mozilla/pkcs11-modules")
     install(FILES ${PROJECT_SOURCE_DIR}/idemiaawppkcs11.json
             DESTINATION "${CMAKE_INSTALL_LIBDIR}/mozilla/pkcs11-modules")
+    add_custom_target(ff ALL COMMAND ${CMAKE_COMMAND} -E make_directory "extensions")
+    add_custom_target(symlink ALL COMMAND ${CMAKE_COMMAND} -E create_symlink
+            "../../../../${CMAKE_INSTALL_DATADIR}/mozilla/extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}/${UUID}.xpi"
+            "${PROJECT_BINARY_DIR}/extensions/${UUID}.xpi")
+    install(DIRECTORY "${PROJECT_BINARY_DIR}/extensions" DESTINATION "${CMAKE_INSTALL_LIBDIR}/firefox/distribution")
     add_custom_target(installer
         COMMENT "Creating installer"
         WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
