--- a/4. Glyphs Source Files/CormorantItalic.glyphs	
+++ b/4. Glyphs Source Files/CormorantItalic.glyphs	
@@ -151,7 +151,6 @@ feature ss08;
 feature ss09;
 feature ss10;
 feature ss11;
-feature ss13;
 feature hlig;
 ";
 name = aalt;
--- a/4. Glyphs Source Files/CormorantRoman.glyphs	
+++ b/4. Glyphs Source Files/CormorantRoman.glyphs	
@@ -133,11 +133,9 @@ feature lnum;
 feature tnum;
 feature onum;
 feature calt;
-feature c2sc;
 feature case;
 feature hlig;
 feature zero;
-feature salt;
 feature cv01;
 feature cv02;
 feature cv05;
@@ -145,14 +143,10 @@ feature cv06;
 feature cv07;
 feature ss01;
 feature ss02;
-feature ss03;
 feature ss04;
-feature ss05;
 feature ss06;
 feature ss09;
 feature ss10;
-feature ss12;
-feature ss13;
 feature ss14;
 feature smcp;
 ";
