--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,25 +8,10 @@ set(HARFBUZZ_VERSION "2.4.0")
 set(HARFBUZZ_SHA256 "9413b8d96132d699687ef914ebb8c50440efc87b3f775d25856d7ec347c03c12")
 
 # download, configure, build and install harbfuzz static library
-ExternalProject_Add(
-    harfbuzz
-    URL "https://github.com/harfbuzz/harfbuzz/releases/download/${HARFBUZZ_VERSION}/harfbuzz-${HARFBUZZ_VERSION}.tar.xz"
-    URL_HASH SHA256=${HARFBUZZ_SHA256}
-    CMAKE_CACHE_ARGS
-      -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_INSTALL_PREFIX}
-      -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
-      -DCMAKE_CONFIGURATION_TYPES:STRING=${CMAKE_CONFIGURATION_TYPES}
-      -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
-      -DHB_BUILD_TESTS:BOOL=OFF
-      -DHB_BUILD_SUBSET:BOOL=OFF
-    # use a shorter prefix to fit in the Windows paths' 260-char limit
-    PREFIX hb
-)
 
 # this allows to use a step from an external project as a dependency of
 # another target. We want to install the harbuzz library before the
 # uharfbuzz module reaches its configure step
-ExternalProject_Add_StepTargets(harfbuzz install)
 
 # build the Cython wrapper as an external project, and pass down to it all
 # the cmake variables set by scikit-build in the top-level CMakeLists.txt
@@ -46,7 +31,6 @@ ExternalProject_Add(
       -DCYTHON_LINETRACE:BOOL=${CYTHON_LINETRACE}
       -DSKBUILD:BOOL=${SKBUILD}
       -DCMAKE_MODULE_PATH:PATH=${CMAKE_MODULE_PATH}
-    DEPENDS harfbuzz-install
     BUILD_ALWAYS 1
     DOWNLOAD_COMMAND ""
     UPDATE_COMMAND ""
--- a/src/uharfbuzz/CMakeLists.txt
+++ b/src/uharfbuzz/CMakeLists.txt
@@ -7,12 +7,10 @@ find_package(PythonLibs)
 find_package(PythonExtensions REQUIRED)
 find_package(Cython REQUIRED)
 
-find_library(HARFBUZZ_LIBRARY harfbuzz
-             PATHS ${CMAKE_INSTALL_PREFIX}/lib
-             NO_DEFAULT_PATH)
-
+include(FindPkgConfig)
+pkg_check_modules(HARFBUZZ REQUIRED harfbuzz)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}
-                    ${CMAKE_INSTALL_PREFIX}/include/harfbuzz)
+${HARFBUZZ_INCLUDE_DIRS})
 
 add_cython_target(_harfbuzz)
 
@@ -26,7 +24,7 @@ if (APPLE)
   endif ()
 endif ()
 
-target_link_libraries(_harfbuzz ${HARFBUZZ_LIBRARY})
+target_link_libraries(_harfbuzz ${HARFBUZZ_LIBRARIES})
 python_extension_module(_harfbuzz)
 
 install(TARGETS _harfbuzz LIBRARY DESTINATION src/uharfbuzz/)
