Index: configure.ac
===================================================================
--- configure.ac	(revision 13810)
+++ configure.ac	(working copy)
@@ -3190,11 +3227,18 @@
         VLC_ADD_PLUGINS([x264])
         VLC_ADD_LDFLAGS([x264],[-lx264])
       ],[
+  PKG_CHECK_MODULES(X264,
+  x264,
+    [
+      VLC_ADD_LDFLAGS([x264],[$X264_LIBS])
+      VLC_ADD_CFLAGS([x264],[$X264_CFLAGS])
+      VLC_ADD_PLUGINS([x264]) ],[
         if test "${enable_x264}" = "yes"; then
             AC_MSG_ERROR([Could not find libx264 on your system: you may get it from http://www.videolan.org/x264.html])
           fi
       ])
     ])
+  ])
     LDFLAGS="${LDFLAGS_save}"
   fi
 fi
