--- a/sources/OpenSans-Italic.designspace
+++ b/sources/OpenSans-Italic.designspace
@@ -14,7 +14,7 @@
   <rules processing="last">
     <rule name="bracket_redo">
       <conditionset>
-        <condition minimum="160" name="Weight" />
+        <condition minimum="160" maximum="800" name="Weight" />
       </conditionset>
       <sub name="dollar" with="dollar.rvrn"/>
       <sub name="alefdagesh-hb" with="alefdagesh-hb.rvrn"/>
--- a/sources/OpenSans-Roman.designspace
+++ b/sources/OpenSans-Roman.designspace
@@ -14,7 +14,7 @@
   <rules processing="last">
     <rule name="bracket_redo">
       <conditionset>
-        <condition minimum="160" name="Weight" />
+        <condition minimum="160" maximum="800" name="Weight" />
       </conditionset>
       <sub name="dollar" with="dollar.rvrn"/>
       <sub name="alefdagesh-hb" with="alefdagesh-hb.rvrn"/>
