diff -ru pinkytagger-2.2.3_orig/src/comparetestmanager.cpp pinkytagger-2.2.3/src/comparetestmanager.cpp
--- pinkytagger-2.2.3_orig/src/comparetestmanager.cpp	2009-04-06 21:05:05.000000000 +0300
+++ pinkytagger-2.2.3/src/comparetestmanager.cpp	2013-10-17 20:29:11.239176574 +0300
@@ -25,7 +25,7 @@
 
 namespace AutoTagger {
 
-   CompareTestManager::CompareTestManager* CompareTestManager::_instance = 0;
+   CompareTestManager* CompareTestManager::_instance = 0;
 
    CompareTestManager::CompareTestManager() throw()
    {
@@ -35,7 +35,7 @@
    {
    } // CompareTestManager::~CompareTestManager()
 
-   CompareTestManager::CompareTestManager&
+   CompareTestManager&
    CompareTestManager::instance() throw()
    {
       if ( !_instance )
diff -ru pinkytagger-2.2.3_orig/src/diriconprovider.h pinkytagger-2.2.3/src/diriconprovider.h
--- pinkytagger-2.2.3_orig/src/diriconprovider.h	2009-04-06 21:05:05.000000000 +0300
+++ pinkytagger-2.2.3/src/diriconprovider.h	2013-10-17 19:45:52.148983432 +0300
@@ -45,7 +45,7 @@
    virtual QIcon icon ( const QFileInfo & info ) const;
 
 private:
-   mutable DirStateCache& cache;
+   DirStateCache& cache;
 
    typedef std::vector<QIcon> TIcons;
    TIcons icons;
diff -ru pinkytagger-2.2.3_orig/src/dlgaboutimpl.cpp pinkytagger-2.2.3/src/dlgaboutimpl.cpp
--- pinkytagger-2.2.3_orig/src/dlgaboutimpl.cpp	2009-04-06 21:05:05.000000000 +0300
+++ pinkytagger-2.2.3/src/dlgaboutimpl.cpp	2013-10-16 20:46:02.210986679 +0300
@@ -45,7 +45,6 @@
 {
 	QString text = _text; 
     text.replace(QString("$PT_VERSION_STRING$"), QString(PT_VERSION_STRING));
-    text.replace(QString("$SVN_REVISION$"), QString(SVN_REVISION));
     text.replace(QString("$PT_COPYRIGHT$"), QString(PT_COPYRIGHT));
     text.replace(QString("$PT_AUTHORS$"), QString(PT_AUTHORS));
     text.replace(QString("$PT_EMAIL$"), QString(PT_EMAIL));
diff -ru pinkytagger-2.2.3_orig/src/taginfo.cpp pinkytagger-2.2.3/src/taginfo.cpp
--- pinkytagger-2.2.3_orig/src/taginfo.cpp	2009-04-06 21:05:05.000000000 +0300
+++ pinkytagger-2.2.3/src/taginfo.cpp	2013-10-17 19:40:22.728466871 +0300
@@ -22,6 +22,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstdio>
 
 #include <taglib/mpegfile.h>
 #include <taglib/mpcfile.h>
