Index: src/fingerprintgenerator.cpp
===================================================================
--- src/fingerprintgenerator.cpp	(revision 28)
+++ src/fingerprintgenerator.cpp	(working copy)
@@ -37,13 +37,13 @@
 #ifdef HAVE_FFMPEG
 extern "C"
 {
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 #ifdef BROKEN_LIBAVFORMAT_DEV
     // On FC6 we deal with a broken avformat package. INT64_C was not defined.
 #define INT64_C int64_t
 #endif
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 }
 #endif
 
Index: CheckSystem.cmake
===================================================================
--- CheckSystem.cmake	(revision 28)
+++ CheckSystem.cmake	(working copy)
@@ -47,8 +47,8 @@
 
 
 # ffmpeg finden
-CHECK_INCLUDE_FILE_CXX(ffmpeg/avcodec.h HAVE_LIBAVCODEC_DEV)
-CHECK_INCLUDE_FILE_CXX(ffmpeg/avformat.h HAVE_LIBAVFORMAT_DEV)
+CHECK_INCLUDE_FILE_CXX(libavcodec/avcodec.h HAVE_LIBAVCODEC_DEV)
+CHECK_INCLUDE_FILE_CXX(libavformat/avformat.h HAVE_LIBAVFORMAT_DEV)
 IF(NOT HAVE_LIBAVFORMAT_DEV)
    # On Fedora Core 6, the avformat.h was broken 
    # INT64_C not defined??? 
