Index: src/decoder/dec_lavc.h
===================================================================
--- src/decoder/dec_lavc.h	(revision 1015)
+++ src/decoder/dec_lavc.h	(working copy)
@@ -23,8 +23,8 @@
 #define _DEC_LAVC_H
 
 #ifdef HAVE_LAVC
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <avcodec.h>
+#include <avformat.h>
 #endif /* HAVE_LAVC */
 
 #include "file_decoder.h"
Index: configure.ac
===================================================================
--- configure.ac	(revision 1015)
+++ configure.ac	(working copy)
@@ -556,6 +556,7 @@
 	AC_CHECK_LIB(avcodec, avcodec_open, [avc_lib=yes], [avc_lib=no], [-lavformat -lavutil -lz])
 
 	if test "$avf_lib" = "yes" -a "$avc_lib" = "yes" ; then
+	        lavc_CFLAGS="`pkg-config --cflags libavcodec libavformat`"
 	        lavc_LIBS="-lavformat -lavcodec -lavutil -lz"
 		AC_DEFINE([HAVE_LAVC], [1], [Defined if compile with LAVC support])
 	fi
@@ -822,7 +823,7 @@
 
 CFLAGS="$CFLAGS $BUILD_CFLAGS -Wall $PLATFORM_CFLAGS $AQUALUNG_SKINDIR $AQUALUNG_LOCALEDIR $AQUALUNG_DATADIR -D_GNU_SOURCE"
 CXXFLAGS="$CFLAGS"
-CPPFLAGS="$gtk_CFLAGS $glib_CFLAGS $xml_CFLAGS $alsa_CFLAGS $jack_CFLAGS $cdda_CFLAGS"
+CPPFLAGS="$gtk_CFLAGS $glib_CFLAGS $xml_CFLAGS $alsa_CFLAGS $jack_CFLAGS $cdda_CFLAGS $lavc_CFLAGS"
 LIBS="decoder/libdecoder.a encoder/libencoder.a $gtk_LIBS $glib_LIBS $xml_LIBS $jack_LIBS $lrdf_LIBS $src_LIBS $alsa_LIBS $oss_LIBS $sndfile_LIBS $flac_LIBS $ogg_LIBS $wavpack_LIBS $speex_LIBS $mad_LIBS $mod_LIBS $mpc_LIBS $mac_LIBS $lavc_LIBS $vorbisenc_LIBS $lame_LIBS $cdda_LIBS $cddb_LIBS $ifp_LIBS $PLATFORM_LIBS $z_LIBS $bz2_LIBS"
 
 
