diff -ur aqualung-0.9beta10.orig/configure.ac aqualung-0.9beta10/configure.ac
--- aqualung-0.9beta10.orig/configure.ac	2009-02-08 12:54:25.000000000 +0200
+++ aqualung-0.9beta10/configure.ac	2009-07-26 15:45:15.000000000 +0300
@@ -526,9 +526,9 @@
 if test "$mpc" = "no"; then
 	AC_MSG_RESULT(no)
 else
-	AC_CHECK_LIB(mpcdec, mpc_streaminfo_init, [lib=yes], [lib=no], [-lstdc++])
+	AC_CHECK_LIB(mpcdecsv7, mpc_streaminfo_init, [lib=yes], [lib=no], [-lstdc++])
 	if test "$lib" = "yes"; then
-	        mpc_LIBS="-lmpcdec -lstdc++"
+	        mpc_LIBS="-lmpcdecsv7 -lstdc++"
 		AC_DEFINE([HAVE_MPC], [1], [Defined if compile with Musepack support])
 	fi
 	if test "$lib" = "no" -a "$mpc" = "yes"; then
diff -ur aqualung-0.9beta10.orig/src/decoder/dec_mpc.h aqualung-0.9beta10/src/decoder/dec_mpc.h
--- aqualung-0.9beta10.orig/src/decoder/dec_mpc.h	2009-02-08 12:55:56.000000000 +0200
+++ aqualung-0.9beta10/src/decoder/dec_mpc.h	2009-07-26 15:45:29.000000000 +0300
@@ -23,7 +23,7 @@
 #define _DEC_MPC_H
 
 #ifdef HAVE_MPC
-#include <mpcdec/mpcdec.h>
+#include <mpcdecsv7/mpcdec.h>
 #endif /* HAVE_MPC */
 
 #include "file_decoder.h"
