--- a/Lib/glyphsLib/builder/custom_params.py
+++ b/Lib/glyphsLib/builder/custom_params.py
@@ -92,7 +92,8 @@ class GlyphsObjectProxy:
         self._glyphs_module = glyphs_module
         self._lookup = defaultdict(list)
         for param in glyphs_object.customParameters:
-            self._lookup[param.name].append(param.value)
+            if not param.disabled:
+                self._lookup[param.name].append(param.value)
         self._handled = set()
 
     def get_attribute_value(self, key):
