--- a/Lib/glyphsLib/interpolation.py
+++ b/Lib/glyphsLib/interpolation.py
@@ -46,6 +46,7 @@ WEIGHT_CODES = {
     'ExtraLight': 250,
     'UltraLight': 250,
     'Light': 300,
+    'SemiLight': 350,
     None: 400,  # default value normally omitted in source
     'Normal': 400,
     'Regular': 400,
@@ -65,6 +66,8 @@ WIDTH_CODES = {
     'Condensed': 3,
     'SemiCondensed': 4,
     None: 5,  # default value normally omitted in source
+    'Regular': 5,
+    'SemiExtended': 6,
     'Medium (normal)': 5,
     'Semi Expanded': 6,
     'Expanded': 7,
@@ -106,6 +108,10 @@ def build_designspace(masters, master_dir, out_dir, instance_data):
         'Masters must all have same family'
 
     for font in masters:
+        font.info.openTypeOS2WeightClass = \
+            WEIGHT_CODES[font.lib.get(GLYPHS_PREFIX + 'weight')]
+        font.info.openTypeOS2WidthClass = \
+            WIDTH_CODES[font.lib.get(GLYPHS_PREFIX + 'width')]
         write_ufo(font, master_dir)
 
     # needed so that added masters and instances have correct relative paths
