--- a/Lib/glyphsLib/builder.py
+++ b/Lib/glyphsLib/builder.py
@@ -309,6 +309,14 @@ def generate_base_fonts(data, family_name):
         ufo.info.styleMapFamilyName = styleMapFamilyName
         ufo.info.styleMapStyleName = styleMapStyleName
 
+        from glyphsLib.interpolation import WEIGHT_CODES, WIDTH_CODES
+        if not width: width=None
+        if not weight: weight=None
+        if ufo.info.openTypeOS2WidthClass is None:
+            ufo.info.openTypeOS2WidthClass = WIDTH_CODES[width]
+        if ufo.info.openTypeOS2WeightClass is None:
+            ufo.info.openTypeOS2WeightClass = WEIGHT_CODES[weight]
+
         set_blue_values(ufo, master.pop('alignmentZones', []))
         set_family_user_data(ufo, user_data)
         set_master_user_data(ufo, master.pop('userData', {}))
--- a/Lib/glyphsLib/interpolation.py
+++ b/Lib/glyphsLib/interpolation.py
@@ -46,6 +46,7 @@ WEIGHT_CODES = {
     'ExtraLight': 250,
     'UltraLight': 250,
     'Light': 300,
+    'SemiLight': 350,
     None: 400,  # default value normally omitted in source
     'Normal': 400,
     'Regular': 400,
