--- a/Lib/glyphsLib/builder.py
+++ b/Lib/glyphsLib/builder.py
@@ -695,7 +695,7 @@ def build_gdef(ufo):
         for anchor in glyph.anchors:
             name = anchor.get('name')
             if name and name.startswith('caret_') and 'x' in anchor:
-                carets.setdefault(glyph.name, []).append(str(anchor['x']))
+                carets.setdefault(glyph.name, []).append('%.f' % (anchor['x']))
     if not carets:
         return None
     lines = ['table GDEF {', '# automatic']
