--- a/Lib/glyphsLib/builder/custom_params.py
+++ b/Lib/glyphsLib/builder/custom_params.py
@@ -103,7 +103,9 @@ class GlyphsObjectProxy:
         self._handled.add(key)
         values = self._lookup[key]
         if len(values) > 1:
-            raise RuntimeError(f"More than one value for this customParameter: {key}")
+            import logging
+            logging.getLogger(__name__).warning(\
+                "More than one value for this customParameter: %s", key)
         if values:
             return values[0]
         return None
