--- a/source/glyphs/FiraMono.glyphs
+++ b/source/glyphs/FiraMono.glyphs
@@ -43,7 +43,7 @@ value = (
 "{689, 14}",
 "{527, 14}",
 "{0, -14}",
-"{-202, -14}",
+"{-202, -14}"
 );
 },
 {
--- a/source/glyphs/FiraSans.glyphs
+++ b/source/glyphs/FiraSans.glyphs
@@ -27,7 +27,7 @@ value = (
 "{689, 14}",
 "{564, 14}",
 "{0, -14}",
-"{-202, -14}",
+"{-202, -14}"
 );
 },
 {
--- a/source/glyphs/FiraSansItalic.glyphs
+++ b/source/glyphs/FiraSansItalic.glyphs
@@ -27,7 +27,7 @@ value = (
 "{689, 14}",
 "{564, 14}",
 "{0, -14}",
-"{-202, -14}",
+"{-202, -14}"
 );
 },
 {
