Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 264)
+++ CMakeLists.txt	(working copy)
@@ -172,6 +172,8 @@
 SET_TARGET_PROPERTIES(flake_static PROPERTIES LINKER_LANGUAGE C)
 SET_TARGET_PROPERTIES(flake_static PROPERTIES COMPILE_FLAGS -DFLAKE_BUILD_LIBRARY)
 TARGET_LINK_LIBRARIES(flake_static ${LIBM} ${ADD_LIBS})
+SET_TARGET_PROPERTIES(flake_static PROPERTIES OUTPUT_NAME flake)
+SET(INSTALL_TARGETS ${INSTALL_TARGETS} flake_static)
 
 # building a separate static lib for the pcm audio decoder
 IF(NOT USE_LIBSNDFILE)
