Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 202)
+++ CMakeLists.txt	(working copy)
@@ -130,11 +130,13 @@
 CHECK_FUNCTION_DEFINE("#include <string.h>" "strnlen" "(\"help\", 6)" HAVE_STRNLEN)
 
 # output SVN version to config.h
+IF(NOT SVN_VERSION)
 EXECUTE_PROCESS(COMMAND svn info --xml WORKING_DIRECTORY ${Flake_SOURCE_DIR}
 OUTPUT_VARIABLE SVN_INFO ERROR_QUIET)
 STRING(REGEX MATCH "<entry[^>]+" SVN_VERSION "${SVN_INFO}")
 STRING(REGEX MATCH "revision=\"[0123456789]+\"" SVN_VERSION "${SVN_VERSION}")
 STRING(REGEX MATCH "[0123456789]+" SVN_VERSION "${SVN_VERSION}")
+ENDIF(NOT SVN_VERSION)
 IF(SVN_VERSION)
   ADD_DEFINE("SVN_VERSION \"${SVN_VERSION}\"")
 ENDIF(SVN_VERSION)
