--- ./CommandLine.cc.orig	2008-12-16 13:20:56.000000000 +0200
+++ ./CommandLine.cc	2010-02-28 20:11:06.000000000 +0200
@@ -29,6 +29,7 @@
 #include <sstream>
 
 #include <getopt.h>
+#include <stdio.h>
 
 #include "ErrorLog.h"
 
--- ./ConfigFile.cc.orig	2008-12-16 13:20:56.000000000 +0200
+++ ./ConfigFile.cc	2010-02-28 20:07:16.000000000 +0200
@@ -25,6 +25,7 @@
 ----------------------------------------------------------------------------*/
 
 #include "ConfigFile.h"
+#include <stdio.h>
 
 CConfigFile::CConfigFile()
 {
--- ./ErrorLog.cc.orig	2008-12-16 13:20:56.000000000 +0200
+++ ./ErrorLog.cc	2010-02-28 20:10:23.000000000 +0200
@@ -25,6 +25,7 @@
 ----------------------------------------------------------------------------*/
 
 #include "ErrorLog.h"
+#include <stdio.h>
 
 const std::vector<std::string>::size_type CErrorLog::m_cMaxSize=100;
 	
--- ./FileNameBuilder.cc.orig	2008-12-16 13:20:56.000000000 +0200
+++ ./FileNameBuilder.cc	2010-02-28 20:09:03.000000000 +0200
@@ -27,6 +27,7 @@
 #include "FileNameBuilder.h"
 
 #include <stdlib.h>
+#include <stdio.h>
 
 #include <unac.h>
 
--- ./UTF8Tag.cc.orig	2008-12-16 13:20:56.000000000 +0200
+++ ./UTF8Tag.cc	2010-02-28 20:11:43.000000000 +0200
@@ -32,6 +32,7 @@
 #include <errno.h>
 #include <string.h>
 #include <stdlib.h>
+#include <stdio.h>
 
 #include <iconv.h>
 
