Index: src/screen.c
===================================================================
--- src/screen.c	(revision 5350)
+++ src/screen.c	(working copy)
@@ -369,15 +369,15 @@
 	  if (c->status->totalTime > 3600)
 	    {
 	      g_snprintf(screen->buf, screen->buf_size, 
-		      " [%i:%02i:%02i/%i:%02i:%02i]",
-		      elapsedTime/3600, (elapsedTime%3600)/60, elapsedTime%60,
+		      " [%d kbps] [%i:%02i:%02i/%i:%02i:%02i]",
+		      status->bitRate, elapsedTime/3600, (elapsedTime%3600)/60, elapsedTime%60,
 		      status->totalTime/3600, (status->totalTime%3600)/60,  status->totalTime%60);
 	    }
 	  else
 	    {
 	      g_snprintf(screen->buf, screen->buf_size, 
-		      " [%i:%02i/%i:%02i]",
-		      elapsedTime/60, elapsedTime%60,
+		      " [%d kbps] [%i:%02i/%i:%02i]",
+		      status->bitRate, elapsedTime/60, elapsedTime%60,
 		      status->totalTime/60,   status->totalTime%60 );
 	    }
 	}
