Index: Makefile.am
===================================================================
--- Makefile.am	(revision 5834)
+++ Makefile.am	(working copy)
@@ -1,8 +1,6 @@
 
 # $Id$
 
-docdir = $(prefix)/share/doc/$(PACKAGE)
-
 doc_DATA = \
   AUTHORS\
   README\
Index: doc/Makefile.am
===================================================================
--- doc/Makefile.am	(revision 5834)
+++ doc/Makefile.am	(working copy)
@@ -3,6 +3,5 @@
 #
 
 man_MANS = ncmpc.1
-docdir = $(prefix)/share/doc/$(PACKAGE)
 doc_DATA = config.sample keys.sample ncmpc.lirc
 EXTRA_DIST = $(man_MANS) $(doc_DATA)
