Index: configure.in
===================================================================
--- configure.in	(revision 12482)
+++ configure.in	(working copy)
@@ -201,7 +201,7 @@
 AC_CHECK_TAGLIB(1.4, have_taglib=yes, have_taglib=no)
 
 dnl Check for libmpcdec
-AC_CHECK_LIB(mpcdec, mpc_decoder_decode, have_mpcdec=yes,)
+AC_CHECK_LIB(mpcdecsv7, mpc_decoder_decode, have_mpcdec=yes,)
 
 dnl Check if we can build TagLib-based plugins
 if test "x$have_taglib" = "xyes"; then
Index: plugins/mpc/Makefile.am
===================================================================
--- plugins/mpc/Makefile.am	(revision 12482)
+++ plugins/mpc/Makefile.am	(working copy)
@@ -31,7 +31,7 @@
 if HAVE_MPCDEC
 mpc_la_CXXFLAGS = $(TAGLIB_CFLAGS) -DHAVE_MPCDEC
 mpc_la_SOURCES = mpc.cpp mpcdecode.cpp
-mpc_la_LIBADD = ../../lib/utf8/libutf8.la ../../lib/libpluginsupport.la $(TAGLIB_LIBS) -lmpcdec -lmusicbrainz
+mpc_la_LIBADD = ../../lib/utf8/libutf8.la ../../lib/libpluginsupport.la $(TAGLIB_LIBS) -lmpcdecsv7 -lmusicbrainz
 else
 mpc_la_CXXFLAGS = $(TAGLIB_CFLAGS)
 mpc_la_SOURCES = mpc.cpp
Index: plugins/mpc/mpcdecode.cpp
===================================================================
--- plugins/mpc/mpcdecode.cpp	(revision 12482)
+++ plugins/mpc/mpcdecode.cpp	(working copy)
@@ -30,7 +30,7 @@
 #include <time.h>
 #include "fileio.h"
 
-#include <mpcdec/mpcdec.h> 
+#include <mpcdecsv7/mpcdec.h> 
 
 extern char *mpcErrorString;
 
