Index: plugins/mp4/mp4.cpp
===================================================================
--- plugins/mp4/mp4.cpp	(revision 12482)
+++ plugins/mp4/mp4.cpp	(working copy)
@@ -27,7 +27,8 @@
 
 #include <string.h>
 #include <stdio.h>
-#include <mp4.h>
+#include <stdlib.h>
+#include <mp4v2/mp4v2.h>
 #include "metadata.h"
 #include "plugin.h"
 #ifndef WIN32
@@ -231,7 +232,7 @@
   
   strcpy(mdata->fileFormat, fileName + strlen(fileName) - 3); 
   
-  if (!MP4Close(mp4file))
+  MP4Close(mp4file);
     return 0;
   
   return 1;
@@ -315,7 +316,7 @@
   sprintf(temp, "%d", mdata->nonAlbum);  
   MP4SetMetadataFreeForm(mp4file, "MusicBrainz Non-Album", (u_int8_t *)temp, strlen(temp) + 1);
   
-  if (!MP4Close(mp4file))
+  MP4Close(mp4file);
     return 0;
 
 #ifndef WIN32
