diff -ru makefiles.orig/gtksubdir.mk makefiles/gtksubdir.mk
--- makefiles.orig/gtksubdir.mk	2007-12-24 11:28:01.000000000 +0200
+++ makefiles/gtksubdir.mk	2007-12-27 09:55:21.000000000 +0200
@@ -11,14 +11,10 @@
 .SUFFIXES: .c .cpp .o .h
 
 .c.o:
-	@echo -n "Compiling $@ ..."
-	@$(CC) -MMD -c $(CFLAGS) $(INCLUDE) $<
-	@echo " OK"
+	$(CC) -MMD -c $(CFLAGS) $(INCLUDE) $<
 
 .cpp.o:
-	@echo -n "Compiling $@ ..."
-	@$(CC) -MMD -c $(CFLAGS) $(INCLUDE) $<
-	@echo " OK"
+	$(CC) -MMD -c $(CFLAGS) $(INCLUDE) $<
 
 all: $(OBJECTS)
 
diff -ru makefiles.orig/qsubdir.mk makefiles/qsubdir.mk
--- makefiles.orig/qsubdir.mk	2007-12-24 11:28:01.000000000 +0200
+++ makefiles/qsubdir.mk	2007-12-27 09:55:32.000000000 +0200
@@ -11,9 +11,7 @@
 .SUFFIXES: .cpp .moc.cpp .moc.o .o .h
 
 .cpp.o:
-	@echo -n "Compiling $@ ..."
-	@$(CC) -MMD -c $(CFLAGS) $(INCLUDE) $<
-	@echo " OK"
+	$(CC) -MMD -c $(CFLAGS) $(INCLUDE) $<
 
 .h.moc.cpp:
 	@echo -n "Generating $@ ..."
diff -ru makefiles.orig/subdir.mk makefiles/subdir.mk
--- makefiles.orig/subdir.mk	2007-12-24 11:28:01.000000000 +0200
+++ makefiles/subdir.mk	2007-12-27 09:54:54.000000000 +0200
@@ -9,9 +9,7 @@
 .SUFFIXES: .cpp .o .h
 
 .cpp.o:
-	@echo -n "Compiling $@ ..."
-	@$(CC) -MMD -c $(CFLAGS) $(INCLUDE) $<
-	@echo " OK"
+	$(CC) -MMD -c $(CFLAGS) $(INCLUDE) $<
 
 all: $(OBJECTS)
 
