--- a/sources/Podkova.glyphs
+++ b/sources/Podkova.glyphs
@@ -60600,7 +60600,7 @@ paths = (
 closed = 1;
 nodes = (
 "183 69 OFFCURVE",
-"127 69 LINE SMOOTH",
+"127 69 CURVE SMOOTH",
 "104 69 OFFCURVE",
 "79 75 OFFCURVE",
 "58 95 CURVE",
@@ -60790,7 +60790,7 @@ paths = (
 closed = 1;
 nodes = (
 "162 69 OFFCURVE",
-"106 69 LINE SMOOTH",
+"106 69 CURVE SMOOTH",
 "83 69 OFFCURVE",
 "58 75 OFFCURVE",
 "37 95 CURVE",
