Index: configure.in
===================================================================
RCS file: /cvsroot/faac/faac/configure.in,v
retrieving revision 1.18
diff -u -r1.18 configure.in
--- configure.in	2 Aug 2004 20:40:30 -0000	1.18
+++ configure.in	22 May 2006 10:10:21 -0000
@@ -35,6 +35,7 @@
 
 if test x$external_mp4v2 = xyes; then
   AC_MSG_NOTICE([*** Building with external mp4v2 ***])
+     MY_DEFINE(HAVE_LIBMP4V2)
 else
   if test x$WITHMP4V2 = xyes; then
      AC_MSG_NOTICE([*** Building with internal mp4v2 ***])
Index: frontend/Makefile.am
===================================================================
RCS file: /cvsroot/faac/faac/frontend/Makefile.am,v
retrieving revision 1.7
diff -u -r1.7 Makefile.am
--- frontend/Makefile.am	28 Jul 2004 08:18:21 -0000	1.7
+++ frontend/Makefile.am	22 May 2006 10:10:22 -0000
@@ -7,5 +7,5 @@
 LDADD = $(top_builddir)/libfaac/libfaac.la $(top_srcdir)/common/mp4v2/libmp4v2.la -lm -lstdc++
 else
 INCLUDES = -I$(top_srcdir)/include 
-LDADD = $(top_builddir)/libfaac/libfaac.la -lm
+LDADD = $(top_builddir)/libfaac/libfaac.la -lmp4v2 -lm
 endif
