--- a/fontFeatures/__init__.py
+++ b/fontFeatures/__init__.py
@@ -279,7 +279,7 @@ class FontFeatures:
 
     def setGlyphClassesFromFont(self, font):
         """Loads glyph classes from the font."""
-        for g in font.exportedGlyphs():
+        for g in font.exported_glyphs():
             if hasattr(font, "glyphs"):
                 self.glyphclasses[g] = font.glyphs[g].category
             else:
--- a/fontFeatures/shaperLib/Buffer.py
+++ b/fontFeatures/shaperLib/Buffer.py
@@ -67,8 +67,8 @@ class BufferItem:
 
     def prep_glyph(self, font):
         if "pytest" in sys.modules:
-            if self.glyph in font.exportedGlyphs():
-                self.gid = font.exportedGlyphs().index(self.glyph)
+            if self.glyph in font.exported_glyphs():
+                self.gid = font.exported_glyphs().index(self.glyph)
             else:
                 self.gid = -1  # ?
         self.substituted = False
