--- ./Makefile.orig	2007-09-27 11:23:58.000000000 +0300
+++ ./Makefile	2007-09-27 11:44:11.000000000 +0300
@@ -17,12 +17,11 @@
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
-CC := $(shell which colorgcc || which cc)
 
 TAGSPRG := ctags
 
 CFLAGS += -Wall
-LDFLAGS += -lao -lmpcdec
+LDADD += -lao -lmpcdec
 
 # in case of DEBUG, we shall add these flags
 ifneq ($(DEBUG), )
@@ -31,8 +30,6 @@
 	LDFLAGS += -rdynamic
 #	LDFLAGS += -lefence
 #	LDFLAGS += -lmudflap
-else
-	CFLAGS += -O2
 endif
 
 TARGET := mpc123
@@ -50,7 +47,7 @@
 
 $(TARGET): $(DEPS) $(TARGET).c $(MOS)
 	@echo Building mpc123 version $(MAJOR).$(MINOR) ...
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $(DEPS) $(TARGET).c
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $(DEPS) $(TARGET).c $(LDADD)
 
 .PHONY: clean tarball
 
