diff --git a/FDK/Tools/linux/setFDKPaths b/FDK/Tools/linux/setFDKPaths
index e26b67a..38ed509 100755
--- a/FDK/Tools/linux/setFDKPaths
+++ b/FDK/Tools/linux/setFDKPaths
@@ -5,7 +5,7 @@ AFDKO_EXE_PATH="${cmdDirPath}"
 AFDKO_Python="${cmdDirPath}"/Python/bin/python2.7
 
 # uncomment this to point to the system python
-# AFDKO_Python="/usr/bin/env python"
+AFDKO_Python="/usr/bin/env python2"
 
 AFDKO_Scripts="${AFDKO_EXE_PATH}"/../SharedData/FDKScripts
 
