Index: misc/dvdnav-config.in
===================================================================
--- misc/dvdnav-config.in	(revision 1109)
+++ misc/dvdnav-config.in	(working copy)
@@ -80,7 +80,7 @@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdnav -ldvdread @THREAD_LIBS@
+      echo -L@libdir@ -ldvdnav -ldvdread-mplayer @THREAD_LIBS@
 fi
 
 if test "$echo_minilibs" = "yes"; then
