Index: Makefile
===================================================================
--- Makefile	(revision 976)
+++ Makefile	(working copy)
@@ -184,7 +184,7 @@
 	cat $(SRC_PATH_BARE)/misc/dvdnav-config2.sh >> $(.OBJDIR)/dvdnav-config
 	chmod 0755 $(SRC_PATH_BARE)/$(.OBJDIR)/dvdnav-config
 
-install-dvdnav-config: dvdnav-config
+install-dvdnav-config:
 	install -d $(PREFIX)/bin
 	install -m 0755 $(.OBJDIR)/dvdnav-config $(PREFIX)/bin/dvdnav-config
 
