--- encode_sv7.c	2005-09-05 00:37:41.000000000 +0300
+++ encode_sv7_nu.c	2006-03-29 13:01:08.000000000 +0300
@@ -108,7 +108,7 @@
 
 #define ENCODE_SCF1( new, old, rll )                         \
         d = new - old + 7;                                   \
-        if ( d <= 14u  &&  (rll < 32  || (int)new < 0 ) ) {  \
+        if ( d <= 14u  && rll < 32) {                        \
             WriteBits ( Table[d].Code, Table[d].Length );    \
         }                                                    \
         else {                                               \
@@ -241,7 +241,7 @@
                 break;
             case 2:
                 ENCODE_SCF1 ( SCF_Index_L[n][0], SCF_Last_L[n]    , DSCF_RLL_L[n] );
-                ENCODE_SCFn ( SCF_Index_L[n][2], SCF_Index_L[n][1], DSCF_RLL_L[n] );
+                ENCODE_SCFn ( SCF_Index_L[n][2], SCF_Index_L[n][0], DSCF_RLL_L[n] );
                 SCF_Last_L[n] = SCF_Index_L[n][2];
                 break;
             case 3:
@@ -250,6 +250,7 @@
                 break;
             }
         }
+		if (DSCF_RLL_L[n] < 33)
         DSCF_RLL_L[n]++;        // Increased counters for SCF that haven't been initialized again
 
         if ( Res_R[n] ) {
@@ -267,7 +268,7 @@
                 break;
             case 2:
                 ENCODE_SCF1 ( SCF_Index_R[n][0], SCF_Last_R[n]    , DSCF_RLL_R[n] );
-                ENCODE_SCFn ( SCF_Index_R[n][2], SCF_Index_R[n][1], DSCF_RLL_R[n] );
+                ENCODE_SCFn ( SCF_Index_R[n][2], SCF_Index_R[n][0], DSCF_RLL_R[n] );
                 SCF_Last_R[n] = SCF_Index_R[n][2];
                 break;
             case 3:
@@ -276,6 +277,7 @@
                 break;
             }
         }
+		if (DSCF_RLL_R[n] < 33)
         DSCF_RLL_R[n]++;          // Increased counters for SCF that haven't been freshly initialized
     }
 
