--- mppdec/replaygain.c.orig	2002-08-11 20:14:12.000000000 +0300
+++ mppdec/replaygain.c	2005-03-07 22:35:10.000000000 +0200
@@ -454,6 +454,7 @@
     gain_info_t  Gain;
     int          i;
     int          ilast;
+    int          smartskip;
 
 #ifdef USE_ARGV
     mysetargv ( &argc, &argv, extentions );
@@ -518,11 +519,43 @@
 
 repeat:
     ilast = -1;
+    smartskip = 0;
+    if ( smart ) {
+        for ( i = 0; argv[i]  &&  (mode != AUTO || (AlbumNameLen (argv[0])==AlbumNameLen (argv[i])  &&  0 == memcmp (argv[0], argv[i], AlbumNameLen (argv[0])))); i++ ) {
+            ilast = i;
+            name = argv [i];
+            Gain.FileName = name;
+            if ( mode == AUTO ) {
+                fd = OPEN (name);
+                if ( fd == INVALID_FILEDESC ) {
+                    stderr_printf ("Can't open: %s\n", name );
+                    //break;
+                }
+                if ( READ ( fd, buff, sizeof(buff) ) != sizeof(buff) ) {
+                    stderr_printf ("Can't read header: %s\n", name );
+                    //break;
+                }
+                if ( CLOSE (fd) < 0 ) {
+                    stderr_printf ("Error closing file: %s\n", name );
+                    //break;
+                }
+                if ( buff[12] || buff[13] || buff[14] || buff[15] || buff[16] || buff[17] || buff[18] || buff[19]) {
+                    smartskip = 1;
+                    //break;
+                } else {
+                    smartskip = 0;
+                    break;
+                }
+            }
+        }
+    }
     for ( i = 0; argv[i]  &&  (mode != AUTO || (AlbumNameLen (argv[0])==AlbumNameLen (argv[i])  &&  0 == memcmp (argv[0], argv[i], AlbumNameLen (argv[0])))); i++ ) {
         ilast = i;
         name = argv [i];
         Gain.FileName = name;
         if ( mode == AUTO ) {
+            if ( smartskip == 0 ) {
+            /*
             if (smart) {
                 fd = OPEN (name);
                 if ( fd == INVALID_FILEDESC ) {
@@ -540,6 +573,7 @@
                 if ( buff[12] || buff[13] || buff[14] || buff[15] || buff[16] || buff[17] || buff[18] || buff[19])
                     continue;
             }
+            */
             if ( ISATTY (FILENO(STDOUT)) ) {
                 printf ("%s\r", name );
                 FLUSH (stdout);
@@ -559,6 +593,7 @@
                 Gain.TitleGain = 0.;
             if ( Gain.AlbumGain > +24. )
                 Gain.AlbumGain = 0.;
+            }
         }
         fd = OPEN (name);
         if ( fd == INVALID_FILEDESC ) {
@@ -604,13 +639,26 @@
             break;
         default:
             Gain.TitlePeak = title_peak;
+            if ( mode == 0 ) {
+                Gain.TitleGain = 0;
+                Gain.TitlePeak = 0;
+                Gain.AlbumPeak = 0;
+                Gain.AlbumGain = 0;
+            }
         case AUTO:
-            ModifyFile ( &Gain, mode==AUTO  ?  4|1  :  4);
-            printf ("%+6.2f dB =>%+6.2f dB | %s => %s (%s)| %s\n",
-                    0.01 * title_gain, Gain.TitleGain,
-                    dB(title_peak), dB(Gain.TitlePeak),
-                    dB(Gain.TitlePeak * pow (10., Gain.TitleGain/20.)),
-                    name );
+            if ( !smartskip ) {
+                if ( mode == 0 )
+                    ModifyFile ( &Gain, 15);
+                else
+                    ModifyFile ( &Gain, mode==AUTO  ?  4|1  :  4);
+                printf ("%+6.2f dB =>%+6.2f dB | %s => %s (%s)| %s\n",
+                        0.01 * title_gain, Gain.TitleGain,
+                        dB(title_peak), dB(Gain.TitlePeak),
+                        dB(Gain.TitlePeak * pow (10., Gain.TitleGain/20.)),
+                        name );
+            } else {
+                printf ("skipped: %s\n", name );
+            }
             break;
         }
         fflush (stdout);
@@ -626,15 +674,20 @@
 
     if ( mode == AUTO ) {
         int len = AlbumNameLen (argv[0]);
-        printf ("          =>%+6.2f dB |       => %s (%s)| %*.*s\n",
-                Gain.AlbumGain,
-                dB(Gain.AlbumPeak),
-                dB(Gain.AlbumPeak * pow (10., Gain.AlbumGain/20.)), len, len, *argv );
+        if ( !smartskip ) {
+            printf ("          =>%+6.2f dB |       => %s (%s)| %*.*s\n",
+                    Gain.AlbumGain,
+                    dB(Gain.AlbumPeak),
+                    dB(Gain.AlbumPeak * pow (10., Gain.AlbumGain/20.)), len, len, *argv );
+        } else {
+            printf ( "          skipped...\n" );
+        }
         fflush (stdout);
         for ( i = 0; i <= ilast; i++ ) {
             name = argv [i];
             Gain.FileName = name;
-            ModifyFile ( &Gain, 2|8 );
+            if ( !smartskip )
+                ModifyFile ( &Gain, 2|8 );
         }
         argv += ilast + 1;
         if ( argv[0] != NULL ) {
