Index: libcuefile/src/CMakeLists.txt
===================================================================
--- libcuefile/src/CMakeLists.txt	(revision 452)
+++ libcuefile/src/CMakeLists.txt	(working copy)
@@ -4,7 +4,3 @@
 add_library(cuefile-static STATIC cd cdtext cue_parse cue_print cue_scan cuefile time toc toc_parse toc_print toc_scan)
 set_target_properties(cuefile-static PROPERTIES OUTPUT_NAME cuefile CLEAN_DIRECT_OUTPUT 1)
 
-add_library(cuefile-shared SHARED cd cdtext cue_parse cue_print cue_scan cuefile time toc toc_parse toc_print toc_scan)
-set_target_properties(cuefile-shared PROPERTIES OUTPUT_NAME cuefile CLEAN_DIRECT_OUTPUT 1 VERSION 0.0.0 SOVERSION 0)
-
-install(TARGETS cuefile-static cuefile-shared LIBRARY DESTINATION "lib${LIB_SUFFIX}" ARCHIVE DESTINATION "lib${LIB_SUFFIX}")
Index: mpcchap/CMakeLists.txt
===================================================================
--- mpcchap/CMakeLists.txt	(revision 452)
+++ mpcchap/CMakeLists.txt	(working copy)
@@ -1,5 +1,5 @@
 FIND_PATH(CUEFILE_INCLUDE_DIR cuetools/cuefile.h
-	${libmpc_SOURCE_DIR}/../libcuefile/include
+	${libmpc_SOURCE_DIR}/libcuefile/include
 )
 
 FIND_LIBRARY(CUEFILE_LIBRARY NAMES cuefile PATHS
@@ -11,6 +11,7 @@
 
 link_directories(${libmpc_BINARY_DIR}/libmpcdec)
 link_directories(${libmpc_BINARY_DIR}/libmpcenc)
+link_directories(${libmpc_BINARY_DIR}/libcuefile/src)
 
 add_executable(mpcchap mpcchap dictionary iniparser ${libmpc_SOURCE_DIR}/common/tags)
 
@@ -21,7 +22,7 @@
 endif(SHARED)
 
 target_link_libraries(mpcchap mpcenc_static)
-target_link_libraries(mpcchap ${CUEFILE_LIBRARY})
+target_link_libraries(mpcchap cuefile-static)
 
 if(NOT MSVC)
 target_link_libraries(mpcchap m)
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 452)
+++ CMakeLists.txt	(working copy)
@@ -27,6 +27,7 @@
 add_subdirectory(mpcenc)
 add_subdirectory(mpc2sv8)
 add_subdirectory(mpccut)
+add_subdirectory(libcuefile)
 add_subdirectory(mpcchap)
 add_subdirectory(wavcmp)
 add_subdirectory(include)
