diff -ruN musepack-tools-444.orig/libmpcdec/CMakeLists.txt musepack-tools-444/libmpcdec/CMakeLists.txt
--- musepack-tools-444.orig/libmpcdec/CMakeLists.txt	2009-07-25 16:49:10.000000000 +0300
+++ musepack-tools-444/libmpcdec/CMakeLists.txt	2009-07-31 16:05:03.000000000 +0300
@@ -1,2 +1,21 @@
 include_directories(${libmpc_SOURCE_DIR}/include)
+
 add_library(mpcdec_static STATIC huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
+set_target_properties(mpcdec_static PROPERTIES OUTPUT_NAME mpcdec CLEAN_DIRECT_OUTPUT 1)
+if(NOT MSVC)
+  target_link_libraries(mpcdec_static m)
+endif(NOT MSVC)
+
+if(SHARED)
+  add_library(mpcdec_shared SHARED huffman mpc_decoder mpc_reader streaminfo mpc_bits_reader mpc_demux requant synth_filter ${libmpc_SOURCE_DIR}/common/crc32)
+  set_target_properties(mpcdec_shared PROPERTIES OUTPUT_NAME mpcdec CLEAN_DIRECT_OUTPUT 1 VERSION 6.1.0 SOVERSION 6)
+  if(NOT MSVC)
+    target_link_libraries(mpcdec_shared m)
+  endif(NOT MSVC)
+endif(SHARED)
+
+install(TARGETS mpcdec_static LIBRARY DESTINATION "lib${LIB_SUFFIX}" ARCHIVE DESTINATION "lib${LIB_SUFFIX}")
+
+if(SHARED)
+  install(TARGETS mpcdec_shared LIBRARY DESTINATION "lib${LIB_SUFFIX}" ARCHIVE DESTINATION "lib${LIB_SUFFIX}")
+endif(SHARED)
