Index: tags.c
===================================================================
--- tags.c	(revision 1)
+++ tags.c	(working copy)
@@ -999,6 +999,7 @@
     return 0;
 }
 
+#if 0
 static void
 FullPathName ( char* dst, size_t dstlen, const char* filename )         // Can contain stuff like ".." and "."
 {
@@ -1307,6 +1308,7 @@
     Parser        ( buff );
     return 0;
 }
+#endif
 
 
 int
@@ -1323,7 +1325,6 @@
 
     CopyTags_APE  (fp);                 // APE tags have higher priority than ID3V1 tags
     CopyTags_ID3  (fp);
-    CopyTags_Name (filename);
 
     fclose (fp);
     return 0;
