diff --git a/configure.ac b/configure.ac
index 65df2a6..7fd2f6d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,6 +3,7 @@ AC_INIT(rtorrent, 0.9.0, jaris@ifi.uio.no)
 AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 AM_PATH_CPPUNIT(1.9.6)
+AC_CONFIG_SUBDIRS(src/libtorrent)
 
 AC_PROG_CXX
 AC_PROG_LIBTOOL
diff --git a/src/Makefile.am b/src/Makefile.am
index c28e255..a33f843 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,4 +1,5 @@
 SUBDIRS = \
+	libtorrent \
 	core \
 	display \
 	input \
@@ -46,6 +47,7 @@ rtorrent_LDADD = \
 	display/libsub_display.a \
 	input/libsub_input.a \
 	rpc/libsub_rpc.a \
+	libtorrent/src/libtorrent.la \
 	utils/libsub_utils.a \
 	@PTHREAD_LIBS@
 
diff --git a/src/libtorrent/Makefile.am b/src/libtorrent/Makefile.am
index f175e63..aec3f83 100644
--- a/src/libtorrent/Makefile.am
+++ b/src/libtorrent/Makefile.am
@@ -1,7 +1,5 @@
 SUBDIRS = src test
 
-pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = libtorrent.pc
 
 EXTRA_DIST= \
 	autogen.sh \
diff --git a/src/libtorrent/src/Makefile.am b/src/libtorrent/src/Makefile.am
index 64413d1..9ffe869 100644
--- a/src/libtorrent/src/Makefile.am
+++ b/src/libtorrent/src/Makefile.am
@@ -8,7 +8,7 @@ SUBDIRS = \
 	tracker \
 	utils
 
-lib_LTLIBRARIES = libtorrent.la
+noinst_LTLIBRARIES = libtorrent.la
 
 libtorrent_la_LDFLAGS = -version-info $(LIBTORRENT_INTERFACE_VERSION_INFO)
 libtorrent_la_LIBADD = \
diff --git a/src/libtorrent/src/torrent/Makefile.am b/src/libtorrent/src/torrent/Makefile.am
index 0c5de3d..c3d54ea 100644
--- a/src/libtorrent/src/torrent/Makefile.am
+++ b/src/libtorrent/src/torrent/Makefile.am
@@ -59,8 +59,7 @@ libsub_torrent_la_SOURCES = \
 
 INCLUDES = -I$(srcdir) -I$(srcdir)/.. -I$(top_srcdir)
 
-libtorrentincludedir = $(includedir)/torrent
-libtorrentinclude_HEADERS = \
+noinst_HEADERS = \
 	bitfield.h \
 	chunk_manager.h \
 	common.h \
diff --git a/src/libtorrent/src/torrent/data/Makefile.am b/src/libtorrent/src/torrent/data/Makefile.am
index ae27aef..51667ab 100644
--- a/src/libtorrent/src/torrent/data/Makefile.am
+++ b/src/libtorrent/src/torrent/data/Makefile.am
@@ -27,8 +27,7 @@ libsub_torrentdata_la_SOURCES = \
 
 INCLUDES = -I$(srcdir) -I$(srcdir)/.. -I$(srcdir)/../.. -I$(top_srcdir)
 
-libtorrentincludedir = $(includedir)/torrent/data
-libtorrentinclude_HEADERS = \
+noinst_HEADERS = \
 	block.h \
 	block_list.h \
 	block_transfer.h \
diff --git a/src/libtorrent/src/torrent/download/Makefile.am b/src/libtorrent/src/torrent/download/Makefile.am
index f97ffae..800e969 100644
--- a/src/libtorrent/src/torrent/download/Makefile.am
+++ b/src/libtorrent/src/torrent/download/Makefile.am
@@ -13,8 +13,7 @@ libsub_torrentdownload_la_SOURCES = \
 
 INCLUDES = -I$(srcdir) -I$(srcdir)/.. -I$(srcdir)/../.. -I$(top_srcdir)
 
-libtorrentincludedir = $(includedir)/torrent/download
-libtorrentinclude_HEADERS = \
+noinst_HEADERS = \
 	choke_group.h \
 	choke_queue.h \
 	download_manager.h \
diff --git a/src/libtorrent/src/torrent/peer/Makefile.am b/src/libtorrent/src/torrent/peer/Makefile.am
index 824209d..53e3354 100644
--- a/src/libtorrent/src/torrent/peer/Makefile.am
+++ b/src/libtorrent/src/torrent/peer/Makefile.am
@@ -17,8 +17,7 @@ libsub_torrentpeer_la_SOURCES = \
 
 INCLUDES = -I$(srcdir) -I$(srcdir)/.. -I$(srcdir)/../.. -I$(top_srcdir)
 
-libtorrentincludedir = $(includedir)/torrent/peer
-libtorrentinclude_HEADERS = \
+noinst_HEADERS = \
 	choke_status.h \
 	client_info.h \
 	client_list.h \
diff --git a/src/libtorrent/src/torrent/utils/Makefile.am b/src/libtorrent/src/torrent/utils/Makefile.am
index 8f7d9e8..165838d 100644
--- a/src/libtorrent/src/torrent/utils/Makefile.am
+++ b/src/libtorrent/src/torrent/utils/Makefile.am
@@ -21,7 +21,7 @@ libsub_torrentutils_la_SOURCES = \
 INCLUDES = -I$(srcdir) -I$(srcdir)/.. -I$(srcdir)/../.. -I$(top_srcdir)
 
 libtorrentincludedir = $(includedir)/torrent/utils
-libtorrentinclude_HEADERS = \
+noinst_HEADERS = \
 	directory_events.h \
 	extents.h \
 	log.h \
