# Fixes a segfault when configure finds execinfo.h
Index: rtorrent/scripts/common.m4
===================================================================
--- rtorrent/scripts/common.m4	(revision 1062)
+++ rtorrent/scripts/common.m4	(working copy)
@@ -183,7 +183,7 @@
 AC_DEFUN([TORRENT_CHECK_EXECINFO], [
   AC_MSG_CHECKING(for execinfo.h)
 
-  AC_RUN_IFELSE(
+  AC_LINK_IFELSE(
     [[#include <execinfo.h>
       int main() { backtrace((void**)0, 0); backtrace_symbols((char**)0, 0); return 0;}
     ]],
