Index: configure.ac
===================================================================
--- configure.ac	(revision 1092)
+++ configure.ac	(working copy)
@@ -3,6 +3,7 @@
 AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 AM_PATH_CPPUNIT(1.9.6)
+AC_CONFIG_SUBDIRS(src/libtorrent)
 
 AC_PROG_CXX
 AC_PROG_LIBTOOL
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 1092)
+++ src/Makefile.am	(working copy)
@@ -1,4 +1,5 @@
 SUBDIRS = \
+	libtorrent \
 	core \
 	display \
 	input \
@@ -14,6 +15,7 @@
 	display/libsub_display.a \
 	input/libsub_input.a \
 	rpc/libsub_rpc.a \
+	libtorrent/src/libtorrent.la \
 	utils/libsub_utils.a
 
 rtorrent_SOURCES = \
Index: src/libtorrent/src/torrent/peer/Makefile.am
===================================================================
--- src/libtorrent/src/torrent/peer/Makefile.am	(revision 1092)
+++ src/libtorrent/src/torrent/peer/Makefile.am	(working copy)
@@ -16,8 +16,8 @@
 
 INCLUDES = -I$(srcdir) -I$(srcdir)/.. -I$(srcdir)/../.. -I$(top_srcdir)
 
-libtorrentincludedir = $(includedir)/torrent/peer
-libtorrentinclude_HEADERS = \
+dnl libtorrentincludedir = $(includedir)/torrent/peer
+dnl libtorrentinclude_HEADERS = \
 	client_info.h \
 	client_list.h \
 	connection_list.h \
Index: src/libtorrent/src/torrent/data/Makefile.am
===================================================================
--- src/libtorrent/src/torrent/data/Makefile.am	(revision 1092)
+++ src/libtorrent/src/torrent/data/Makefile.am	(working copy)
@@ -23,8 +23,8 @@
 
 INCLUDES = -I$(srcdir) -I$(srcdir)/.. -I$(srcdir)/../.. -I$(top_srcdir)
 
-libtorrentincludedir = $(includedir)/torrent/data
-libtorrentinclude_HEADERS = \
+dnl libtorrentincludedir = $(includedir)/torrent/data
+dnl libtorrentinclude_HEADERS = \
 	block.h \
 	block_list.h \
 	block_transfer.h \
Index: src/libtorrent/src/torrent/Makefile.am
===================================================================
--- src/libtorrent/src/torrent/Makefile.am	(revision 1092)
+++ src/libtorrent/src/torrent/Makefile.am	(working copy)
@@ -52,8 +52,8 @@
 
 INCLUDES = -I$(srcdir) -I$(srcdir)/.. -I$(top_srcdir)
 
-libtorrentincludedir = $(includedir)/torrent
-libtorrentinclude_HEADERS = \
+dnl libtorrentincludedir = $(includedir)/torrent
+dnl libtorrentinclude_HEADERS = \
 	bitfield.h \
 	chunk_manager.h \
 	common.h \
Index: src/libtorrent/src/Makefile.am
===================================================================
--- src/libtorrent/src/Makefile.am	(revision 1092)
+++ src/libtorrent/src/Makefile.am	(working copy)
@@ -8,7 +8,7 @@
 	tracker \
 	utils
 
-lib_LTLIBRARIES = libtorrent.la
+noinst_LTLIBRARIES = libtorrent.la
 
 libtorrent_la_LDFLAGS = -version-info $(LIBTORRENT_INTERFACE_VERSION_INFO)
 libtorrent_la_LIBADD = \
Index: src/libtorrent/Makefile.am
===================================================================
--- src/libtorrent/Makefile.am	(revision 1092)
+++ src/libtorrent/Makefile.am	(working copy)
@@ -1,7 +1,5 @@
 SUBDIRS = src test
 
-pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = libtorrent.pc
 
 EXTRA_DIST= \
 	autogen.sh \
