--- src/libtorrent/src/torrent/data/file_list.cc (revision 1259)
+++ src/libtorrent/src/torrent/data/file_list.cc (revision 1257)
@@ -632,11 +637,7 @@
   inc_completed(begin(), index);
 
   // TODO: Remember to validate 'wanted_chunks'.
+  if (m_data.normal_priority()->has(index) || m_data.high_priority()->has(index))
-  if (m_data.normal_priority()->has(index) || m_data.high_priority()->has(index)) {
-    if (m_data.wanted_chunks() == 0)
-      throw internal_error("FileList::mark_completed(...) m_data.wanted_chunks() == 0.");
-    
     m_data.set_wanted_chunks(m_data.wanted_chunks() - 1);
-  }
 }
 
