Index: configure.ac
===================================================================
--- configure.ac	(revision 944)
+++ configure.ac	(working copy)
@@ -2,6 +2,7 @@
 
 AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
+AC_CONFIG_SUBDIRS(libtorrent)
 
 TORRENT_CHECK_CXXFLAGS()
 TORRENT_ENABLE_DEBUG()
@@ -25,7 +26,7 @@
 TORRENT_WITHOUT_STATVFS()
 TORRENT_WITHOUT_STATFS()
 
-PKG_CHECK_MODULES(STUFF, sigc++-2.0 libcurl >= 7.12.0 libtorrent >= 0.11.5,
+PKG_CHECK_MODULES(STUFF, sigc++-2.0 libcurl >= 7.12.0,
 	          CXXFLAGS="$CXXFLAGS $STUFF_CFLAGS";
 		  LIBS="$LIBS $STUFF_LIBS")
 
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 944)
+++ src/Makefile.am	(working copy)
@@ -14,7 +14,8 @@
 	$(top_srcdir)/src/display/libsub_display.a \
 	$(top_srcdir)/src/input/libsub_input.a \
 	$(top_srcdir)/src/rpc/libsub_rpc.a \
-	$(top_srcdir)/src/utils/libsub_utils.a
+	$(top_srcdir)/src/utils/libsub_utils.a \
+	$(top_srcdir)/libtorrent/src/libtorrent.la
 
 rtorrent_SOURCES = \
 	command_download.cc \
