Index: modules/ffmpeg_in/ffmpeg_decode.c
===================================================================
RCS file: /cvsroot/gpac/gpac/modules/ffmpeg_in/ffmpeg_decode.c,v
retrieving revision 1.12
diff -u -r1.12 ffmpeg_decode.c
--- modules/ffmpeg_in/ffmpeg_decode.c	24 May 2006 09:52:56 -0000	1.12
+++ modules/ffmpeg_in/ffmpeg_decode.c	14 Jul 2006 14:06:28 -0000
@@ -659,7 +659,7 @@
 {
 	FFDec *ffd = dec->privateStack;
 	if (ffd->codec) {
-		sprintf(szCodec, "FFMPEG %s (v%s)", ffd->codec->name ? ffd->codec->name : "unknown", FFMPEG_VERSION);
+		sprintf(szCodec, "FFMPEG %s", ffd->codec->name ? ffd->codec->name : "unknown");
 		return szCodec;
 	}
 	return NULL;
