Index: modules/oss_audio/oss.h
===================================================================
RCS file: /cvsroot/gpac/gpac/modules/oss_audio/oss.h,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 oss.h
--- modules/oss_audio/oss.h	13 Jul 2005 14:35:51 -0000	1.1.1.1
+++ modules/oss_audio/oss.h	14 Jul 2005 17:52:36 -0000
@@ -31,7 +31,7 @@
 #endif
 
 
-#include <gpac/module/av_hw.h>
+#include <gpac/modules/audio_out.h>
 
 #define OSS_AUDIO_DEVICE	"/dev/dsp"
 
