--- a/xpcom/io/TorFileUtils.cpp
+++ b/xpcom/io/TorFileUtils.cpp
@@ -15,7 +15,7 @@ nsresult TorBrowser_GetUserDataDir(nsIFi
   NS_ENSURE_ARG_POINTER(aFile);
   nsCOMPtr<nsIFile> tbDataDir;
 
-#ifdef ANDROID
+#if defined(ANDROID) || defined(TOR_BROWSER_DATA_IN_HOME_DIR)
   const char* homeDir = getenv("HOME");
   if (!homeDir || !*homeDir) return NS_ERROR_FAILURE;
   nsresult rv = NS_NewNativeLocalFile(nsDependentCString(homeDir), true,
--- a/toolkit/xre/nsXREDirProvider.cpp
+++ b/toolkit/xre/nsXREDirProvider.cpp
@@ -1414,7 +1414,12 @@ nsresult nsXREDirProvider::GetUserDataDi
 #  ifdef TOR_BROWSER_DATA_OUTSIDE_APP_DIR
   rv = GetTorBrowserUserDataDir(getter_AddRefs(localDir));
   NS_ENSURE_SUCCESS(rv, rv);
+#ifdef TOR_BROWSER_DATA_IN_HOME_DIR
+  rv = localDir->AppendNative(nsDependentCString(MOZ_USER_DIR));
+  rv = localDir->AppendNative("torbrowser"_ns);
+#else
   rv = localDir->AppendNative("Browser"_ns);
+#endif
 #  else
   RefPtr<nsXREDirProvider> singleton = GetSingleton();
   if (!singleton) {
@@ -1427,7 +1433,19 @@ nsresult nsXREDirProvider::GetUserDataDi
   NS_ENSURE_SUCCESS(rv, rv);
 #endif
   if (aLocal) {
+#ifndef TOR_BROWSER_DATA_IN_HOME_DIR
     rv = localDir->AppendNative("Caches"_ns);
+#else
+    const char* cacheHome = getenv("XDG_CACHE_HOME");
+    if (cacheHome && *cacheHome) {
+      rv = NS_NewNativeLocalFile(nsDependentCString(cacheHome), true,
+                                 getter_AddRefs(localDir));
+    } else {
+      rv = GetTorBrowserUserDataDir(getter_AddRefs(localDir));
+      if (NS_SUCCEEDED(rv)) rv = localDir->AppendNative(".cache"_ns);
+    }
+    rv = localDir->AppendNative("torbrowser"_ns);
+#endif
     NS_ENSURE_SUCCESS(rv, rv);
   }
 #elif defined(XP_MACOSX)
@@ -1534,7 +1552,11 @@ nsresult nsXREDirProvider::GetSysUserExt
 
 nsresult nsXREDirProvider::GetSysUserExtensionsDevDirectory(nsIFile** aFile) {
   nsCOMPtr<nsIFile> localDir;
+#ifndef TOR_BROWSER_DATA_IN_HOME_DIR
   nsresult rv = GetUserDataDirectoryHome(getter_AddRefs(localDir), false);
+#else
+  nsresult rv = GetTorBrowserUserDataDir(getter_AddRefs(localDir));
+#endif
   NS_ENSURE_SUCCESS(rv, rv);
 
   rv = AppendSysUserExtensionsDevPath(localDir);
