--- a/security/sandbox/linux/moz.build
+++ b/security/sandbox/linux/moz.build
@@ -98,7 +98,7 @@ if CONFIG['CC_TYPE'] in ('clang', 'gcc')
 # gcc lto likes to put the top level asm in syscall.cc in a different partition
 # from the function using it which breaks the build.  Work around that by
 # forcing there to be only one partition.
-for f in CONFIG['OS_CXXFLAGS']:
+for f in CONFIG['MOZ_LTO_CFLAGS']:
     if f.startswith('-flto') and CONFIG['CC_TYPE'] != 'clang':
         LDFLAGS += ['--param', 'lto-partitions=1']
 
