--- a/toolkit/xre/nsAppRunner.cpp	2016-05-01 10:22:35.624335413 +0300
+++ b/toolkit/xre/nsAppRunner.cpp	2016-05-01 10:23:04.281579824 +0300
@@ -4372,6 +4372,7 @@
     return 1;
   }
 
+  nsCOMPtr<nsIFile> exeDir;
   rv = SelectProfile(getter_AddRefs(mProfileLock), mProfileSvc, exeDir,
                      mNativeApp, &mStartOffline, &mProfileName);
   if (rv == NS_ERROR_LAUNCHED_CHILD_PROCESS ||
