--- synthasm.nas
+++ synthasm.nas
@@ -2210,3 +2210,6 @@
 ;
 ; end of synthasm.nas
 ;
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- cpu_feat.nas
+++ cpu_feat.nas
@@ -123,3 +123,6 @@
 endproc
 
         end
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- endianasm.nas
+++ endianasm.nas
@@ -30,3 +30,6 @@
 ;
 ; end of endianasm.nas
 ;
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
