--- a/afdko/Tools/SharedData/FDKScripts/FDKUtils.py
+++ b/afdko/Tools/SharedData/FDKScripts/FDKUtils.py
@@ -32,7 +32,9 @@ def findFDKDirs():
     fdkToolsDir = None
     _dir = os.path.dirname(__file__)
 
-    while _dir:
+    fdkSharedDataDir = _dir
+    fdkToolsDir = os.path.join(fdkSharedDataDir.split('/usr/')[0],'usr/lib/afdko/bin')
+    while False:
         if os.path.basename(_dir) == "Tools":
             fdkScriptsDir = os.path.join(_dir, "SharedData", "FDKScripts")
             if curSystem == "Darwin":
