--- a/FDK/Tools/Programs/public/lib/config/linux/gcc/gcc.mak
+++ b/FDK/Tools/Programs/public/lib/config/linux/gcc/gcc.mak
@@ -28,7 +28,7 @@ default: $(LIB_TARGET)
 
 $(LIB_TARGET): $(LIB_OBJS)
 	mkdir -p $(LIB_DIR)
-	$(AR) -rvs $@ $?
+	$(AR) rvs $@ $?
 
 clean:
 	if [ "$(LIB_OBJS)" ]; then \
--- a/FDK/Tools/Programs/public/lib/config/linux/gcc/gcc_tx.mak
+++ b/FDK/Tools/Programs/public/lib/config/linux/gcc/gcc_tx.mak
@@ -32,7 +32,7 @@ default: $(TARGETS)
 
 $(LIB_TARGET): $(LIB_OBJS)
 	mkdir -p $(LIB_DIR)
-	$(AR) -rvs $@ $?
+	$(AR) rvs $@ $?
 
 clean:
 	if [ "$(LIB_OBJS)" ]; then \
