Index: Plugins/Input/aac/Makefile.in
===================================================================
--- Plugins/Input/aac/Makefile.in	(revision 685)
+++ Plugins/Input/aac/Makefile.in	(working copy)
@@ -1,3 +1,3 @@
 include ../../../mk/objective.mk
 
-SUBDIRS = libmp4v2 libfaad2 src
+SUBDIRS = src
Index: Plugins/Input/aac/src/Makefile.in
===================================================================
--- Plugins/Input/aac/src/Makefile.in	(revision 685)
+++ Plugins/Input/aac/src/Makefile.in	(working copy)
@@ -8,10 +8,10 @@
 # override objective make's LIBDIR
 LIBDIR = $(plugindir)/$(INPUT_PLUGIN_DIR)
 
-LDFLAGS += ../libmp4v2/libmp4v2.a ../libfaad2/libfaad.a -lstdc++
+LDFLAGS += -lmp4v2 -lfaad -lstdc++
 
 SOURCES = libmp4.c mp4_utils.c aac_utils.c tagging_mp4.c fileinfo.c
 
 OBJECTS = ${SOURCES:.c=.o}
 
-CFLAGS += $(GTK_CFLAGS) -I../../../../intl -I../../../.. -I../include -I../libmp4v2 -I../libfaad
+CFLAGS += $(GTK_CFLAGS) -I../../../../intl -I../../../..
Index: Plugins/Input/adplug/Makefile.in
===================================================================
--- Plugins/Input/adplug/Makefile.in	(revision 685)
+++ Plugins/Input/adplug/Makefile.in	(working copy)
@@ -1,17 +1,17 @@
 include ../../../mk/rules.mk
 include ../../../mk/objective.mk
 
-SUBDIRS = core
+SUBDIRS = 
 OBJECTIVE_LIBS = libadplug.so
 
 LIBDIR = $(plugindir)/$(INPUT_PLUGIN_DIR)
 
-LIBADD = -lstdc++ $(BINIO_LIBS) ./core/libadplugcore.a
+LIBADD = -lstdc++ $(BINIO_LIBS) -ladplug
 
 SOURCES = adplug-xmms.cc
 
 OBJECTS = ${SOURCES:.cc=.o}
 
 CFLAGS += -fPIC -DPIC $(GTK_CFLAGS) $(BINIO_CFLAGS) \
-	-I../../../intl -I../../.. -I./core
+	-I../../../intl -I../../.. -I/usr/include/adplug
 CXXFLAGS = $(CFLAGS)
