Index: configure.ac
===================================================================
--- configure.ac	(revision 685)
+++ configure.ac	(working copy)
@@ -336,7 +336,7 @@
     [enable_aac=yes]
 )
 
-if test "$have_aac" = "yes"; then
+if test "$enable_aac" = "yes"; then
 	INPUT_PLUGINS="$INPUT_PLUGINS aac"
 fi
 
Index: Makefile.in
===================================================================
--- Makefile.in	(revision 685)
+++ Makefile.in	(working copy)
@@ -4,11 +4,6 @@
 SUBDIRS = Plugins libaudacious audacious po icons skin intl
 
 install-posthook:
-	@if test `whoami` = 'root'; then \
-		echo "[running ldconfig to update system library cache]"; \
-		ldconfig; \
-		echo "[system library cache updated]"; \
-	fi
 	@mkdir -p $(DESTDIR)/$(LIBDIR)/pkgconfig
 	@mkdir -p $(DESTDIR)/$(mandir)/man1
 	@mkdir -p $(DESTDIR)/$(datadir)/applications
Index: mk/objective.mk
===================================================================
--- mk/objective.mk	(revision 685)
+++ mk/objective.mk	(working copy)
@@ -38,6 +38,7 @@
 	@if test "$(OBJECTIVE_BINS)" != "none"; then \
 		for i in $(OBJECTIVE_BINS); do \
 			printf "%10s     %-20s\n" INSTALL $$i; \
+			$(INSTALL) $(INSTALL_OVERRIDE) -d $(DESTDIR)/$(BINDIR); \
 			$(INSTALL) $(INSTALL_OVERRIDE) $$i $(DESTDIR)/$(BINDIR)/$$i; \
 		done; \
 	fi;
