--- configure.ac	2009-11-02 08:31:58.000000000 +0200
+++ configure.ac	2009-11-10 17:15:28.377119108 +0200
@@ -104,7 +104,7 @@
 
 AC_ARG_ENABLE(man,
               [AC_HELP_STRING([--enable-man],
-                              [regenerate man pages from Docbook [default=no]])],enable_man=yes,
+                              [regenerate man pages from Docbook [default=no]])],enable_man=$enableval,
               enable_man=no)
 
 if test "${enable_man}" != no; then
--- man/Makefile.am	2009-10-09 00:22:33.000000000 +0300
+++ man/Makefile.am	2009-11-10 17:23:21.387372149 +0200
@@ -1,9 +1,9 @@
+if ENABLE_MAN
 man_MANS = wavpack.1 wvgain.1 wvunpack.1
+endif
 
-if ENABLE_MAN
 %.1: %.xml
 	$(XSLTPROC) -nonet http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl $<
-endif
 
 EXTRA_DIST = $(man_MANS)
 
