Index: common.mak
===================================================================
RCS file: /cvsroot/ffmpeg/ffmpeg/common.mak,v
retrieving revision 1.1
diff -u -r1.1 common.mak
--- common.mak	13 Feb 2006 12:53:25 -0000	1.1
+++ common.mak	14 Feb 2006 13:17:37 -0000
@@ -86,6 +86,7 @@
 	install -m 644 $(LIB) "$(libdir)"
 
 install-headers:
+	install -d "$(incdir)"
 	install -m 644 $(addprefix "$(SRC_DIR)"/,$(HEADERS)) "$(incdir)"
 	install -m 644 $(BUILD_ROOT)/lib$(NAME).pc "$(libdir)/pkgconfig"
 
