diff --git a/wafadmin/Tools/gcc.py b/wafadmin/Tools/gcc.py
index 28f2b4c..90b3c48 100644
--- a/wafadmin/Tools/gcc.py
+++ b/wafadmin/Tools/gcc.py
@@ -81,7 +81,7 @@ def detect(conf):
 		if not dest: dest=orig
 		try: conf.env[dest] = ron[orig]
 		except KeyError: pass
-	addflags('CCFLAGS', 'CFLAGS')
+	addflags('CFLAGS', 'CCFLAGS')
 	addflags('CPPFLAGS')
 	addflags('LINKFLAGS')
 
diff --git a/wscript b/wscript
index ac1a840..7d6d645 100644
--- a/wscript
+++ b/wscript
@@ -203,8 +203,8 @@ def configure(conf):
     conf.check_tool('man', tooldir=os.path.abspath('waftools'))
 
     conf.env["VERSION"] = VERSION
-    conf.env["CCFLAGS"] = Utils.to_list(conf.env["CCFLAGS"]) + ['-g', '-O0']
-    conf.env["CXXFLAGS"] = Utils.to_list(conf.env["CXXFLAGS"]) + ['-g', '-O0']
+    conf.env["CCFLAGS"] = Utils.to_list(conf.env["CCFLAGS"]) + ['-g']
+    conf.env["CXXFLAGS"] = Utils.to_list(conf.env["CXXFLAGS"]) + ['-g']
     conf.env['XMMS_PKGCONF_FILES'] = []
     conf.env['XMMS_OUTPUT_PLUGINS'] = []
 
