diff -ru libtool-1.5.2/libtool.m4 libtool-1.5.2_nu/libtool.m4
--- libtool-1.5.2/libtool.m4	2004-01-23 08:07:04.000000000 +0200
+++ libtool-1.5.2_nu/libtool.m4	2004-11-11 14:22:29.731145112 +0200
@@ -2125,8 +2125,9 @@
 mingw* | pw32*)
   # Base MSYS/MinGW do not provide the 'file' command needed by
   # func_win32_libid shell function, so use a weaker test based on 'objdump'.
-  lt_cv_deplibs_check_method='file_magic file format pei*-i386(.*architecture: i386)?'
-  lt_cv_file_magic_cmd='$OBJDUMP -f'
+  #lt_cv_deplibs_check_method='file_magic file format pei*-i386(.*architecture: i386)?'
+  #lt_cv_file_magic_cmd='$OBJDUMP -f'
+  lt_cv_deplibs_check_method=pass_all
   ;;
 
 darwin* | rhapsody*)
@@ -2870,12 +2871,13 @@
     # _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1) is actually meaningless,
     # as there is no search path for DLLs.
     _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
-    _LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
+    #_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
+    _LT_AC_TAGVAR(no_undefined_flag, $1)=' -no-undefined'
     _LT_AC_TAGVAR(always_export_symbols, $1)=no
     _LT_AC_TAGVAR(enable_shared_with_static_runtimes, $1)=yes
 
     if $LD --help 2>&1 | grep 'auto-import' > /dev/null; then
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -shared -nostdlib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags -o $output_objdir/$soname ${wl}--image-base=0x10000000 ${wl}--out-implib,$lib'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -shared $libobjs $deplibs $compiler_flags -o $output_objdir/$soname ${wl}--image-base=0x10000000 ${wl}--out-implib,$lib'
       # If the export-symbols file already is a .def file (1st line
       # is EXPORTS), use it as is; otherwise, prepend...
       _LT_AC_TAGVAR(archive_expsym_cmds, $1)='if test "x`$SED 1q $export_symbols`" = xEXPORTS; then
@@ -2884,7 +2886,7 @@
 	echo EXPORTS > $output_objdir/$soname.def;
 	cat $export_symbols >> $output_objdir/$soname.def;
       fi~
-      $CC -shared -nostdlib $output_objdir/$soname.def $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags -o $output_objdir/$soname ${wl}--image-base=0x10000000 ${wl}--out-implib,$lib'
+      $CC -shared $output_objdir/$soname.def $libobjs $deplibs $compiler_flags -o $output_objdir/$soname ${wl}--image-base=0x10000000 ${wl}--out-implib,$lib'
     else
       _LT_AC_TAGVAR(ld_shlibs, $1)=no
     fi
diff -ru libtool-1.5.2/ltmain.in libtool-1.5.2_nu/ltmain.in
--- libtool-1.5.2/ltmain.in	2004-01-25 08:50:41.000000000 +0200
+++ libtool-1.5.2_nu/ltmain.in	2004-11-11 14:19:50.790307792 +0200
@@ -2314,7 +2314,7 @@
 	      case $host in
 	      *cygwin* | mingw*)
 		major=`expr $current - $age`
-		versuffix="-$major"
+		versuffix=""
 		;;
 	      esac
 	      eval soname=\"$soname_spec\"
@@ -3037,7 +3037,7 @@
 	  # Use '-' rather than '.', since we only want one
 	  # extension on DOS 8.3 filesystems.
 	  major=`expr $current - $age`
-	  versuffix="-$major"
+	  versuffix=""
 	  ;;
 
 	*)
