--- libtool-1.5.10/ltmain.in.old 2004-11-16 22:21:32.849312264 -0800
+++ libtool-1.5.10/ltmain.in	2004-11-16 22:26:47.899417368 -0800
@@ -1005,7 +1005,7 @@
     # Lock this critical section if it is needed
     # We use this script file to make the link, it avoids creating a new file
     if test "$need_locks" = yes; then
-      until $run ln "$progpath" "$lockfile" 2>/dev/null; do
+      until $run ln "$srcfile" "$lockfile" 2>/dev/null; do
 	$show "Waiting for $lockfile to be removed"
 	sleep 2
       done
