Index: Makefile.in
===================================================================
--- Makefile.in	(revision 3896)
+++ Makefile.in	(working copy)
@@ -156,7 +156,7 @@
 USE_NLS = @USE_NLS@
 VERSION = @VERSION@
 
-SUBDIRS = intl po macros src pixmaps
+SUBDIRS = macros src pixmaps
 
 mpd_appletdocdir = ${prefix}/doc/mpd-applet
 mpd_appletdoc_DATA =  	README 	COPYING 	AUTHORS 	ChangeLog 	INSTALL 	NEWS 	TODO 	ABOUT-NLS
@@ -168,7 +168,7 @@
 mpd_applet_glade_file_DATA = mpd-applet.glade
 
 ACLOCAL_AMFLAGS = -I m4
-ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
+ACLOCAL_M4 = 
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = config.h
 CONFIG_CLEAN_FILES = 
