--- devede.py	2006-12-20 00:29:09.000000000 +0200
+++ devede	2007-02-04 02:23:28.000000000 +0200
@@ -81,18 +81,18 @@
 	sys.exit(1)
 
 try:
-	from devede_other import *
+	from DeVeDe.devede_other import *
 except:
 	print "Failed to load modules 2. Exiting"
 	sys.exit(1)
 try:
-	from devede_gtk_helper import *
+	from DeVeDe.devede_gtk_helper import *
 except:
 	print "Failed to load modules 3. Exiting"
 	sys.exit(1)
 
 try:
-	import devede_convert
+	from DeVeDe import devede_convert
 except:
 	print "Failed to load modules. Exiting"
 	sys.exit(1)
--- /dev/null	2000-10-22 01:01:00.000000000 +0200
+++ setup.py	2007-02-04 01:35:38.000000000 +0200
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+
+import os, glob
+from distutils.core import setup
+
+name='DeVeDe'
+data_files=[
+    ('share/devede',        ['devede.glade']),
+    ('share/devede',        glob.glob('pixmaps/*.png')),
+    ('share/devede',        ['devedesans.ttf']),
+    ('share/applications',  ['devede.desktop']),
+    ('share/pixmaps',       ['devede.png']),
+    ('share/doc/@PF@/html', glob.glob('docs/*'))
+    ]
+
+for po_file in glob.glob('po/*.po'):
+    lang = os.path.basename(po_file[:-3])
+    mo_dir = 'share/locale/%s/LC_MESSAGES' % lang
+    mo_file = mo_dir + '/devede.mo'
+    if not os.path.isdir(mo_dir):
+        os.makedirs(mo_dir)
+        os.system('msgfmt %s -o %s' % (po_file, mo_file))
+    data_files.append((mo_dir, [mo_file]))
+
+setup(name=name,
+    version='@PV@',
+    description='A program to create video DVD suitable for home DVD players',
+    url='http://www.rastersoft.com/programas/devede.html',
+    packages=[name],
+    scripts = ['devede'],
+    data_files = data_files
+    )
