Index: lib/Makefile.am
===================================================================
--- lib/Makefile.am	(revision 14226)
+++ lib/Makefile.am	(working copy)
@@ -144,6 +144,7 @@
 libtheoraenc_la_LDFLAGS = \
   -version-info @THENC_LIB_CURRENT@:@THENC_LIB_REVISION@:@THENC_LIB_AGE@ \
   @THEORAENC_LDFLAGS@
+libtheoraenc_la_LIBADD = $(OGG_LIBS)
 
 libtheora_la_SOURCES = \
   cpu.c \
@@ -155,6 +156,7 @@
 libtheora_la_LDFLAGS = \
   -version-info @TH_LIB_CURRENT@:@TH_LIB_REVISION@:@TH_LIB_AGE@ \
   @THEORA_LDFLAGS@
+libtheora_la_LIBADD = $(OGG_LIBS)
 
 debug:
        $(MAKE) all CFLAGS="@DEBUG@" 
