--- libtool-1.5.10/ltmain.in.orig	2004-11-09 13:43:24.290875562 -0500
+++ libtool-1.5.10/ltmain.in	2004-11-09 13:45:51.250048464 -0500
@@ -242,7 +242,13 @@
 	    esac
 	    CC_quoted="$CC_quoted $arg"
 	  done
+	    # user sometimes does CC=<HOST>-gcc so we need to match that to 'gcc'
+	    trimedcc="${CC/${host}-}"
 	    case "$@ " in
+	      "gcc "* | " gcc "* | "${host}-gcc "* | " ${host}-gcc "*)
+	      tagname=CC
+	      break ;;
+	      "$trimedcc "* | " $trimedcc "* | "`$echo $trimedcc` "* | " `$echo $trimedcc` "*|\
 	      " $CC "* | "$CC "* | " `$echo $CC` "* | "`$echo $CC` "* | " $CC_quoted"* | "$CC_quoted "* | " `$echo $CC_quoted` "* | "`$echo $CC_quoted` "*)
 	      # The compiler in the base compile command matches
 	      # the one in the tagged configuration.
