--- configure.ac.orig	2004-07-17 06:03:20.309921376 +0300
+++ configure.ac	2004-07-17 06:29:14.763608496 +0300
@@ -55,12 +55,15 @@
 
 
 AC_CHECK_LIB(faad,faacDecClose, [
-  AC_CHECK_HEADER(faad.h, [HAVE_FAAD=yes])
+  AC_CHECK_HEADER(faad.h, [FFMPEG_EXTLIBS="$FFMPEG_EXTLIBS -lfaad"])
     ],,[-lm -lz])
-if test "x$HAVE_FAAD" = xyes; then
-	FFMPEG_EXTLIBS="-lfaad"
-	FFMPEG_LIBS="$FFMPEG_LIBS $FFMPEG_EXTLIBS"
-fi
+AC_CHECK_LIB(faad,NeAACDecClose, [
+  AC_CHECK_HEADER(faad.h, [FFMPEG_EXTLIBS="$FFMPEG_EXTLIBS -lfaad"])
+    ],,[-lm])
+AC_CHECK_LIB(faac,faacEncClose,[FFMPEG_EXTLIBS="$FFMPEG_EXTLIBS -lfaac"])
+AC_CHECK_LIB(mp3lame,lame_init,[FFMPEG_EXTLIBS="$FFMPEG_EXTLIBS -lmp3lame"
+    ],,[-lm])
+FFMPEG_LIBS="$FFMPEG_LIBS $FFMPEG_EXTLIBS"
 AC_SUBST(FFMPEG_CFLAGS)
 AC_SUBST(FFMPEG_LIBS)
 
--- Makefile.am	2004-07-13 21:48:13.432947872 +0300
+++ Makefile.am	2004-07-13 20:13:54.055304048 +0300
@@ -9,5 +9,5 @@
 bin_PROGRAMS = ffmpeg2theora
 
 ffmpeg2theora_SOURCES = ffmpeg2theora.c  theorautils.c
-ffmpeg2theora_LDFLAGS = -static -L$(prefix)/lib @FFMPEG_LIBS@ @XIPH_LIBS@
+ffmpeg2theora_LDFLAGS = @FFMPEG_LIBS@ @XIPH_LIBS@
 ffmpeg2theora_CFLAGS = $(EXTRA) @XIPH_CFLAGS@ @FFMPEG_CFLAGS@
