Index: flac-1.1.2/doc/html/Makefile.am
===================================================================
--- flac-1.1.2.orig/doc/html/Makefile.am
+++ flac-1.1.2/doc/html/Makefile.am
@@ -19,9 +19,9 @@ AUTOMAKE_OPTIONS = foreign
 
 SUBDIRS = ru images
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/html
+htmldir = $(docdir)/html
 
-doc_DATA =  \
+html_DATA =  \
 	changelog.html \
 	comparison.html \
 	developers.html \
@@ -40,9 +40,9 @@ doc_DATA =  \
 	ogg_mapping.html
 
 if FLaC__HAS_DOXYGEN
-EXTRA_DIST = $(doc_DATA) api
+EXTRA_DIST = $(html_DATA) api
 else
-EXTRA_DIST = $(doc_DATA)
+EXTRA_DIST = $(html_DATA)
 endif
 
 if FLaC__HAS_DOXYGEN
Index: flac-1.1.2/doc/html/ru/Makefile.am
===================================================================
--- flac-1.1.2.orig/doc/html/ru/Makefile.am
+++ flac-1.1.2/doc/html/ru/Makefile.am
@@ -17,9 +17,9 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/html/ru
+rudir = $(docdir)/html/ru
 
-doc_DATA = \
+ru_DATA = \
 	authors.html \
 	comparison.html \
 	developers.html \
@@ -33,4 +33,4 @@ doc_DATA = \
 	links.html \
 	news.html
 
-EXTRA_DIST = $(doc_DATA)
+EXTRA_DIST = $(ru_DATA)
Index: flac-1.1.2/doc/html/images/Makefile.am
===================================================================
--- flac-1.1.2.orig/doc/html/images/Makefile.am
+++ flac-1.1.2/doc/html/images/Makefile.am
@@ -17,12 +17,12 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/html/images
+imagedir = $(docdir)/html/images
 
-doc_DATA = \
+image_DATA = \
 	1x1.gif \
 	cafebug.gif \
 	logo100.gif \
 	logo130.gif
 
-EXTRA_DIST = $(doc_DATA)
+EXTRA_DIST = $(image_DATA)
