Index: flac-1.1.2/src/plugin_xmms/plugin.c
===================================================================
--- flac-1.1.2.orig/src/plugin_xmms/plugin.c
+++ flac-1.1.2/src/plugin_xmms/plugin.c
@@ -180,6 +180,9 @@ static const decoder_funcs_t* DECODER_FU
 
 static decoder_funcs_t const * decoder_func_table_;
 
+#ifdef SUPPORT_ATTRIBUTE_VISIBILITY
+InputPlugin *get_iplugin_info() __attribute__((visibility("default")));
+#endif
 
 InputPlugin *get_iplugin_info()
 {
