--- configure.in.orig	2006-11-21 15:37:30.000000000 +0200
+++ configure.in	2006-11-25 10:17:42.000000000 +0200
@@ -269,7 +269,7 @@
 else
 	OUR_CFLAGS_HEAD="$OUR_CFLAGS_HEAD -DNDEBUG"
 	if test "x$GCC" = xyes; then
-		OUR_CFLAGS_HEAD="$OUR_CFLAGS_HEAD -O3 -funroll-loops -finline-functions -Wall -W -Winline -DFLaC__INLINE=__inline__"
+		OUR_CFLAGS_HEAD="$OUR_CFLAGS_HEAD -Wall -W -Winline -DFLaC__INLINE=__inline__"
 	fi
 fi
 CFLAGS="$OUR_CFLAGS_HEAD $CFLAGS"
@@ -327,7 +327,6 @@
 	doc/html/Makefile \
 	doc/html/images/Makefile \
 	doc/html/ru/Makefile \
-	m4/Makefile \
 	man/Makefile \
 	test/Makefile \
 	test/cuesheets/Makefile \
--- Makefile.am.orig	2006-11-21 15:37:30.000000000 +0200
+++ Makefile.am	2006-11-25 10:05:18.000000000 +0200
@@ -30,7 +30,7 @@
 
 AUTOMAKE_OPTIONS = foreign 1.7
 
-SUBDIRS = doc include m4 man src test build obj
+SUBDIRS = doc include man src test build obj
 
 DISTCLEANFILES = libtool-disable-static
 
