--- configure.in.orig	2006-11-25 09:40:26.000000000 +0200
+++ configure.in	2006-11-25 10:11:56.000000000 +0200
@@ -184,6 +184,15 @@
 esac],[FLAC__TEST_WITH_VALGRIND=no])
 AC_SUBST(FLAC__TEST_WITH_VALGRIND)
 
+AC_ARG_ENABLE(doc,
+AC_HELP_STRING([--enable-doc], [Enable documentation building]),
+[case "${enableval}" in
+	yes) enable_doc=true ;;
+	no)  enable_doc=false ;;
+	*) AC_MSG_ERROR(bad value ${enableval} for --enable-doc) ;;
+esac],[enable_doc=false])
+AM_CONDITIONAL(FLaC__BUILD_DOCS, test "x$enable_doc" = xtrue)
+
 AC_ARG_ENABLE(doxygen-docs,
 AC_HELP_STRING([--disable-doxygen-docs], [Disable API documentation building via Doxygen]),
 [case "${enableval}" in
--- Makefile.am.orig	2006-11-21 15:37:30.000000000 +0200
+++ Makefile.am	2006-11-25 10:05:18.000000000 +0200
@@ -30,7 +30,11 @@
 
 AUTOMAKE_OPTIONS = foreign 1.7
 
-SUBDIRS = doc include man src examples test build obj
+SUBDIRS = include man src examples test build obj
+
+if FLaC__BUILD_DOCS
+SUBDIRS += doc
+endif
 
 DISTCLEANFILES = libtool-disable-static
 
--- ./doc/html/Makefile.am.orig	2009-10-15 23:13:45.000000000 +0300
+++ ./doc/html/Makefile.am	2009-10-15 23:13:27.000000000 +0300
@@ -71,6 +71,7 @@
 
 # The install targets don't copy whole directories so we have to
 # handle 'api/' specially:
+if FLaC__HAS_DOXYGEN
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(docdir)/api
 	(cd $(srcdir)/api && $(INSTALL_DATA) * $(DESTDIR)$(docdir)/api)
@@ -78,3 +79,4 @@
 	rm -rf $(DESTDIR)$(docdir)/api
 maintainer-clean-local:
 	rm -rf api
+endif
