--- ./SConstruct.orig	2007-06-17 15:57:00.000000000 +0300
+++ ./SConstruct	2007-06-17 15:59:53.000000000 +0300
@@ -883,7 +883,7 @@
 
 #    libraries['flowcanvas'] = LibraryInfo(LIBS='flowcanvas', LIBPATH='#/libs/flowcanvas', CPPPATH='#libs/flowcanvas')
     libraries['soundtouch'] = LibraryInfo()
-    libraries['soundtouch'].ParseConfig ('pkg-config --cflags --libs libSoundTouch')
+    libraries['soundtouch'].ParseConfig ('pkg-config --cflags --libs soundtouch-1.0')
     # Comment the previous line and uncomment this for Debian:
     #libraries['soundtouch'].ParseConfig ('pkg-config --cflags --libs libSoundTouch')
 
