--- a/libs/surfaces/control_protocol/wscript
+++ b/libs/surfaces/control_protocol/wscript
@@ -34,6 +34,7 @@ def build(bld):
     obj.target       = 'ardourcp'
     obj.use          = 'libardour libpbd'
     obj.uselib       = 'GLIBMM SIGCPP XML OSX'
+    obj.vnum         = '0.0.0'
     obj.install_path = bld.env['LIBDIR']
 
 def shutdown():
--- a/libs/surfaces/frontier/wscript
+++ b/libs/surfaces/frontier/wscript
@@ -27,7 +27,7 @@ def build(bld):
     obj.name         = 'libgeneric_midi'
     obj.target       = 'generic_midi'
     obj.uselib       = 'XML OSX'
-    obj.use          = 'libardour libardourcp'
+    obj.use          = 'libardour libardour_cp'
     obj.install_path = os.path.join(bld.env['LIBDIR'], 'surfaces')
 
 def shutdown():
