Index: SConstruct
===================================================================
--- SConstruct	(revision 10031)
+++ SConstruct	(working copy)
@@ -47,6 +47,7 @@
     BoolVariable('DMALLOC', 'Compile and link using the dmalloc library', 0),
     BoolVariable('EXTRA_WARN', 'Compile with -Wextra, -ansi, and -pedantic.  Might break compilation.  For pedants', 0),
     BoolVariable('FFT_ANALYSIS', 'Include FFT analysis window', 1),
+    BoolVariable('RUBBERBAND', 'Use rubberband instead of soundtouch for timefx', 1),
     BoolVariable('FREESOUND', 'Include Freesound database lookup', 0),
     BoolVariable('FPU_OPTIMIZATION', 'Build runtime checked assembler code', 1),
     BoolVariable('LIBLO', 'Compile with support for liblo library', 1),
@@ -901,11 +902,10 @@
                                  LIBPATH='#libs/vamp-sdk',
                                  CPPPATH='#libs/vamp-sdk')
 
-env['RUBBERBAND'] = False
 
 conf = Configure (env)
 
-if conf.CheckHeader ('fftw3.h'):
+if env['RUBBERBAND'] and conf.CheckHeader ('fftw3.h'):
     env['RUBBERBAND'] = True
     libraries['rubberband'] = LibraryInfo (LIBS='rubberband',
                                            LIBPATH='#libs/rubberband',
@@ -1105,9 +1105,11 @@
 
 #    libraries['flowcanvas'] = LibraryInfo(LIBS='flowcanvas', LIBPATH='#/libs/flowcanvas', CPPPATH='#libs/flowcanvas')
     libraries['soundtouch'] = LibraryInfo()
-    libraries['soundtouch'].ParseConfig ('pkg-config --cflags --libs soundtouch-1.0')
+    libraries['soundtouch'].ParseConfig ('pkg-config --cflags --libs soundtouch')
     # Comment the previous line and uncomment this for old versions of Debian:
     #libraries['soundtouch'].ParseConfig ('pkg-config --cflags --libs libSoundTouch')
+    libraries['rubberband'] = LibraryInfo()
+    libraries['rubberband'].ParseConfig ('pkg-config --cflags --libs rubberband')
 
     libraries['appleutility'] = LibraryInfo(LIBS='libappleutility',
                                             LIBPATH='#libs/appleutility',
@@ -1254,6 +1256,9 @@
 timefx_subdirs = ['libs/soundtouch']
 if env['RUBBERBAND']:
     timefx_subdirs += ['libs/rubberband']
+    env.Append(CCFLAGS="-DUSE_RUBBERBAND")
+if env['SYSLIBS']:
+    timefx_subdirs = []
 
 #
 # Tools
