Index: gtk2_ardour/ardour_ui_options.cc
===================================================================
--- gtk2_ardour/ardour_ui_options.cc	(revision 2952)
+++ gtk2_ardour/ardour_ui_options.cc	(working copy)
@@ -24,7 +24,9 @@
 
 #include <ardour/configuration.h>
 #include <ardour/session.h>
+#ifdef HAVE_LIBLO
 #include <ardour/osc.h>
+#endif
 #include <ardour/audioengine.h>
 
 #include "ardour_ui.h"
