Index: SConstruct
===================================================================
--- SConstruct	(revision 1525)
+++ SConstruct	(working copy)
@@ -771,13 +771,16 @@
 # prepend boiler plate optimization flags
 #
 
-opt_flags[:0] = [
+if 0:
+    opt_flags[:0] = [
     "-O3",
     "-fomit-frame-pointer",
     "-ffast-math",
     "-fstrength-reduce",
     "-pipe"
     ]
+else:
+    opt_flags[:0] = os.environ['CFLAGS'].split()
 
 if env['DEBUG'] == 1:
     env.Append(CCFLAGS=" ".join (debug_flags))
