Index: SConstruct
===================================================================
--- SConstruct	(revision 2952)
+++ SConstruct	(working copy)
@@ -782,10 +782,6 @@
 prep_libcheck(env, env)
 
 
-libraries['vamp'] = LibraryInfo (LIBS='vampsdk',
-                                 LIBPATH='#libs/vamp-sdk',
-                                 CPPPATH='#libs/vamp-sdk/vamp')
-
 env['RUBBERBAND'] = False
 
 conf = Configure (env)
@@ -984,6 +980,9 @@
                                             LIBPATH='#libs/appleutility',
                                             CPPPATH='#libs/appleutility')
     
+    libraries['vamp'] = LibraryInfo ()
+    libraries['vamp'].ParseConfig ('pkg-config --cflags --libs vamp-sdk vamp-hostsdk')
+
     coredirs = [
         'templates'
     ]
@@ -993,7 +992,6 @@
         'libs/pbd',
         'libs/midi++2',
         'libs/ardour',
-        'libs/vamp-sdk',
     # these are unconditionally included but have
     # tests internally to avoid compilation etc
     # if VST is not set
@@ -1048,6 +1046,10 @@
                                             LIBPATH='#libs/appleutility',
                                             CPPPATH='#libs/appleutility')
 
+    libraries['vamp'] = LibraryInfo (LIBS='vampsdk',
+                                     LIBPATH='#libs/vamp-sdk',
+                                     CPPPATH='#libs/vamp-sdk/vamp')
+
     coredirs = [
         'templates'
     ]
