Index: wscript
===================================================================
--- wscript	(revision 13043)
+++ wscript	(working copy)
@@ -95,6 +95,8 @@
         rev = fetch_git_revision('.');
     elif os.path.exists('.bzr'):
         rev = fetch_bzr_revision('.');
+    elif os.environ['ESVN_WC_REVISION']:
+        rev = os.environ['ESVN_WC_REVISION']
         print("Revision: %s", rev)
     elif os.path.exists('libs/ardour/svn_revision.cc'):
         print("Using packaged svn revision")
