Index: wscript
===================================================================
--- wscript	(revision 13053)
+++ wscript	(working copy)
@@ -19,11 +19,7 @@
         'libs/pbd',
         'libs/midi++2',
         'libs/evoral',
-        'libs/vamp-sdk',
         'libs/qm-dsp',
-        'libs/vamp-plugins',
-        'libs/taglib',
-        'libs/rubberband',
         'libs/surfaces',
         'libs/panners',
         'libs/timecode',
Index: libs/ardour/wscript
===================================================================
--- libs/ardour/wscript	(revision 13053)
+++ libs/ardour/wscript	(working copy)
@@ -282,7 +282,9 @@
 
     # we don't try to detect this, since its part of our source tree
 
-    conf.define('HAVE_RUBBERBAND', 1) # controls whether we think we have it
+    autowaf.check_pkg(conf, 'vamp-sdk vamp-hostsdk', uselib_store='VAMP', atleast_version='2.0')
+    autowaf.check_pkg(conf, 'taglib', uselib_store='TAGLIB', atleast_version='1.7')
+    autowaf.check_pkg(conf, 'rubberband', uselib_store='RUBBERBAND', atleast_version='1.4')
     conf.define('USE_RUBBERBAND', 1)  # controls whether we actually use it
 
     conf.define('CURRENT_SESSION_FILE_VERSION', CURRENT_SESSION_FILE_VERSION)
@@ -360,7 +362,7 @@
     obj.target       = 'ardour'
     obj.uselib       = ['GLIBMM','GTHREAD','AUBIO','SIGCPP','XML','UUID',
                         'JACK','SNDFILE','SAMPLERATE','LRDF','AUDIOUNITS',
-                        'OSX','BOOST','CURL','DL']
+                        'OSX','BOOST','CURL','DL','VAMP','TAGLIB','RUBBERBAND']
     obj.use          = ['libpbd','libmidipp','libevoral','libvamphost',
                         'libvampplugin','libtaglib','librubberband',
                         'libaudiographer']
Index: libs/ardour/rb_effect.cc
===================================================================
--- libs/ardour/rb_effect.cc	(revision 13053)
+++ libs/ardour/rb_effect.cc	(working copy)
@@ -21,7 +21,7 @@
 #include <cmath>
 
 #include "pbd/error.h"
-#include "rubberband/RubberBandStretcher.h"
+#include <rubberband/RubberBandStretcher.h>
 
 #include "ardour/audioregion.h"
 #include "ardour/audiosource.h"
