--- ./SConstruct.orig	2010-10-23 21:20:28.057578917 +0300
+++ ./SConstruct	2010-10-23 21:40:10.219397777 +0300
@@ -264,8 +264,8 @@
     return commands.getoutput (cmd)
 
 def create_stored_revision (target = None, source = None, env = None):
-    if os.path.exists('.svn'):    
-        rev = fetch_svn_revision ('.');
+    try:
+        rev = os.environ['ESVN_WC_REVISION']
         try:
             text  = "#include <ardour/svn_revision.h>\n"
             text += "namespace ARDOUR {\n";
@@ -278,7 +278,7 @@
         except IOError:
             print "Could not open libs/ardour/svn_revision.cc for writing\n"
             sys.exit (-1)
-    else:
+    except KeyError:
         if not os.path.exists('libs/ardour/ardour/svn_revision.h'):    
             print "This release of ardour is missing libs/ardour/ardour/svn_revision.h. Blame the packager."
             sys.exit (-1)
@@ -1344,8 +1344,10 @@
 subst_dict['%VERSION%'] = ardour_version[0:3]
 subst_dict['%EXTRA_VERSION%'] = ardour_version[3:]
 subst_dict['%REVISION_STRING%'] = ''
-if os.path.exists('.svn'):
-    subst_dict['%REVISION_STRING%'] = '.' + fetch_svn_revision ('.') + 'svn'
+try:
+    subst_dict['%REVISION_STRING%'] = '.' + os.environ['ESVN_WC_REVISION'] + 'svn'
+except KeyError:
+    pass
 
 # specbuild = env.SubstInFile ('ardour.spec','ardour.spec.in', SUBST_DICT = subst_dict)
 
