Index: SConstruct
===================================================================
--- SConstruct	(revision 2979)
+++ SConstruct	(working copy)
@@ -866,7 +866,7 @@
 #
 
 conf = Configure (libraries['flac'])
-if conf.CheckLib ('FLAC', 'FLAC__seekable_stream_decoder_init', language='CXX'):
+if conf.CheckLib ('FLAC', 'FLAC__stream_decoder_init_file', language='CXX'):
     conf.env.Append(CCFLAGS='-DHAVE_FLAC')
     use_flac = True
 else:
