diff --git a/libs/ardour/rb_effect.cc b/libs/ardour/rb_effect.cc
index a5d47d2..a9b10d0 100644
--- a/libs/ardour/rb_effect.cc
+++ b/libs/ardour/rb_effect.cc
@@ -21,7 +21,7 @@
 #include <cmath>
 
 #include "pbd/error.h"
-#include "rubberband/RubberBandStretcher.h"
+#include <rubberband/RubberBandStretcher.h>
 
 #include "ardour/audioregion.h"
 #include "ardour/audiosource.h"
diff --git a/libs/ardour/wscript b/libs/ardour/wscript
index b4a50a8..667cba8 100644
--- a/libs/ardour/wscript
+++ b/libs/ardour/wscript
@@ -287,7 +287,9 @@ def configure(conf):
 
     # we don't try to detect this, since its part of our source tree
 
-    conf.define('HAVE_RUBBERBAND', 1) # controls whether we think we have it
+    autowaf.check_pkg(conf, 'vamp-sdk vamp-hostsdk', mandatory=True, uselib_store='VAMP', atleast_version='2.0')
+    autowaf.check_pkg(conf, 'taglib', mandatory=True, uselib_store='TAGLIB', atleast_version='1.7')
+    autowaf.check_pkg(conf, 'rubberband', mandatory=True, uselib_store='RUBBERBAND', atleast_version='1.4')
     conf.define('USE_RUBBERBAND', 1)  # controls whether we actually use it
 
     conf.define('CURRENT_SESSION_FILE_VERSION', CURRENT_SESSION_FILE_VERSION)
@@ -378,7 +380,7 @@ def build(bld):
     obj.target       = 'ardour'
     obj.uselib       = ['GLIBMM','GTHREAD','AUBIO','SIGCPP','XML','UUID',
                         'JACK','SNDFILE','SAMPLERATE','LRDF','AUDIOUNITS',
-                        'OSX','BOOST','CURL','DL']
+                        'OSX','BOOST','CURL','DL','VAMP','TAGLIB','RUBBERBAND']
     obj.use          = ['libpbd','libmidipp','libevoral','libvamphost',
                         'libvampplugin','libtaglib','librubberband',
                         'libaudiographer','libltc']
diff --git a/wscript b/wscript
index 794c2f1..a625d53 100644
--- a/wscript
+++ b/wscript
@@ -28,12 +28,8 @@ children = [
         'libs/pbd',
         'libs/midi++2',
         'libs/evoral',
-        'libs/vamp-sdk',
         'libs/qm-dsp',
-        'libs/vamp-plugins',
-        'libs/taglib',
         'libs/libltc',
-        'libs/rubberband',
         'libs/surfaces',
         'libs/panners',
         'libs/timecode',
@@ -724,14 +720,11 @@ def build(bld):
     # add directories that contain only headers, to workaround an issue with waf
 
     bld.path.find_dir ('libs/evoral/evoral')
-    bld.path.find_dir ('libs/vamp-sdk/vamp-sdk')
     bld.path.find_dir ('libs/surfaces/control_protocol/control_protocol')
     bld.path.find_dir ('libs/timecode/timecode')
     bld.path.find_dir ('libs/libltc/ltc')
-    bld.path.find_dir ('libs/rubberband/rubberband')
     bld.path.find_dir ('libs/gtkmm2ext/gtkmm2ext')
     bld.path.find_dir ('libs/ardour/ardour')
-    bld.path.find_dir ('libs/taglib/taglib')
     bld.path.find_dir ('libs/pbd/pbd')
 
     autowaf.set_recursive()
