Index: manual/Makefile
===================================================================
--- manual/Makefile	(revision 1407)
+++ manual/Makefile	(working copy)
@@ -20,7 +20,9 @@
 
 html:: xml
 	# generate html
-	LANG=en_US.UTF-8 $(XMLTO) -o tmp/ html $(XMLFILE)
+	-@mkdir -p html
+	-@cp -rf images html/
+	LANG=en_US.UTF-8 $(XMLTO) -o html/ html $(XMLFILE)
 #	LANG=en_US.UTF-8 $(XMLTO) -x $(XSLFILE) -o tmp/ html $(XMLFILE)
 	# copy css file to html directory
 #	-@cp -f css/$(DOCNAME).css $(CSSFILE)
