Index: SConstruct
===================================================================
--- SConstruct	(revision 2336)
+++ SConstruct	(working copy)
@@ -231,7 +231,9 @@
         installenv.Alias ('install', buildenv.MoBuild (mo_file, po_file))
     
     for lang in languages:
-        modir = (os.path.join (install_prefix, 'share/locale/' + lang + '/LC_MESSAGES/'))
+        mosub = lang
+        if lang != 'pt_BR': mosub = mosub.split('_')[0]
+        modir = (os.path.join (install_prefix, 'share/locale/' + mosub + '/LC_MESSAGES/'))
         moname = domain + '.mo'
         installenv.Alias('install', installenv.InstallAs (os.path.join (modir, moname), lang + '.mo'))
 
