--- a/libs/ardour/wscript
+++ b/libs/ardour/wscript
@@ -333,7 +333,7 @@ def configure(conf):
     autowaf.check_pkg(conf, 'libcurl', uselib_store='CURL',
                       atleast_version='7.0.0')
 
-    autowaf.check_pkg(conf, 'fftw3f', uselib_store='FFTW35F',
+    autowaf.check_pkg(conf, 'fftw3f_threads', uselib_store='FFTW35F',
                       atleast_version='3.3.5', mandatory=False)
 
     # controls whether we actually use it in preference to soundtouch
