diff --exclude-from=/home/dang/.diffrc -u -ruN evince-0.4.0.orig/data/evince.desktop.in.in evince-0.4.0/data/evince.desktop.in.in
--- evince-0.4.0.orig/data/evince.desktop.in.in	2005-08-12 23:08:25.000000000 -0400
+++ evince-0.4.0/data/evince.desktop.in.in	2005-09-08 21:50:03.000000000 -0400
@@ -9,7 +9,6 @@
 Terminal=false
 Type=Application
 Icon=postscript-viewer.png
-NoDisplay=true
 X-GNOME-DocPath=
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=evince
