diff --exclude-from=/home/dang/.diffrc -u -ruN evince-0.4.0.orig/configure.ac evince-0.4.0/configure.ac
--- evince-0.4.0.orig/configure.ac	2005-08-26 00:04:22.000000000 -0400
+++ evince-0.4.0/configure.ac	2005-08-29 22:20:12.000000000 -0400
@@ -234,7 +234,7 @@
 
 if test "x$enable_dvi" = "xyes"; then
     if test "x$enable_type1_fonts" = "xyes"; then
-	AC_CHECK_LIB([t1lib],T1_InitLib,enable_type1_fonts=yes,enable_type1_fonts=no)
+	AC_CHECK_LIB([t1],T1_InitLib,enable_type1_fonts=yes,enable_type1_fonts=no)
     fi
 
     if test "x$enable_type1_fonts" = xyes; then
diff --exclude-from=/home/dang/.diffrc -u -ruN evince-0.4.0.orig/dvi/mdvi-lib/Makefile.am evince-0.4.0/dvi/mdvi-lib/Makefile.am
--- evince-0.4.0.orig/dvi/mdvi-lib/Makefile.am	2005-08-25 18:40:58.000000000 -0400
+++ evince-0.4.0/dvi/mdvi-lib/Makefile.am	2005-08-29 22:21:44.000000000 -0400
@@ -42,5 +42,5 @@
 	-lkpathsea
 
 if WITH_TYPE1_FONTS
-libmdvi_la_LIBADD += -lt1lib
+libmdvi_la_LIBADD += -lt1
 endif
