Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 606)
+++ CMakeLists.txt	(working copy)
@@ -170,11 +170,13 @@
 CHECK_INCLUDE_FILE_DEFINE(byteswap.h HAVE_BYTESWAP_H)
 
 # output SVN version to config.h
+IF(NOT SVN_VERSION)
 EXECUTE_PROCESS(COMMAND svn info --xml WORKING_DIRECTORY ${Aften_SOURCE_DIR}
 OUTPUT_VARIABLE SVN_INFO ERROR_QUIET)
 STRING(REGEX MATCH "<entry[^>]+" SVN_VERSION "${SVN_INFO}")
 STRING(REGEX MATCH "revision=\"[0123456789]+\"" SVN_VERSION "${SVN_VERSION}")
 STRING(REGEX MATCH "[0123456789]+" SVN_VERSION "${SVN_VERSION}")
+ENDIF(NOT SVN_VERSION)
 IF(SVN_VERSION)
   ADD_DEFINE("SVN_VERSION \"${SVN_VERSION}\"")
 ENDIF(SVN_VERSION)
