Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 544)
+++ CMakeLists.txt	(working copy)
@@ -263,10 +263,11 @@
 ENDIF(SHARED)
 
 ADD_LIBRARY(aften_static STATIC ${LIBAFTEN_SRCS})
-#SET_TARGET_PROPERTIES(aften_static PROPERTIES OUTPUT_NAME aften)
+SET_TARGET_PROPERTIES(aften_static PROPERTIES OUTPUT_NAME aften)
 SET_TARGET_PROPERTIES(aften_static PROPERTIES LINKER_LANGUAGE C)
 SET_TARGET_PROPERTIES(aften_static PROPERTIES COMPILE_FLAGS -DAFTEN_BUILD_LIBRARY)
 TARGET_LINK_LIBRARIES(aften_static ${LIBM} ${ADD_LIBS})
+SET(INSTALL_TARGETS ${INSTALL_TARGETS} aften_static)
 
 # building a separate static lib for the pcm audio decoder
 ADD_LIBRARY(aften_pcm STATIC ${PCM_SRCS})
