--- old-dvdauthor/src/subfont.c	2006-07-27 03:09:22.000000000 +0300
+++ new-dvdauthor/src/subfont.c	2006-07-27 03:09:22.000000000 +0300
@@ -150,7 +150,7 @@
 		      int unicode) {
     FT_Error	error;
     FT_Face face = desc->faces[pic_idx];
-    int	const	load_flags = FT_LOAD_DEFAULT | FT_LOAD_NO_HINTING | FT_LOAD_MONOCHROME | FT_LOAD_RENDER;
+    int	const	load_flags = FT_LOAD_DEFAULT | FT_LOAD_TARGET_NORMAL | FT_LOAD_MONOCHROME | FT_LOAD_RENDER;
 	/* int	const	load_flags = FT_LOAD_DEFAULT | FT_LOAD_NO_HINTING; */ /* Anti-aliasing */
 	int		ymin = INT_MAX, ymax = INT_MIN;
     int		space_advance = 20;
@@ -516,7 +516,7 @@
     int width, height, stride, maxw, off;
     unsigned char *abuffer, *bbuffer;
 
-    int	const	load_flags = FT_LOAD_DEFAULT | FT_LOAD_NO_HINTING | FT_LOAD_MONOCHROME | FT_LOAD_RENDER;
+    int	const	load_flags = FT_LOAD_DEFAULT | FT_LOAD_TARGET_NORMAL | FT_LOAD_MONOCHROME | FT_LOAD_RENDER;
 	/* int	const	load_flags = FT_LOAD_DEFAULT | FT_LOAD_NO_HINTING; */ /* Anti-aliasing */
 
     int		pen_xa;

