Index: version.sh
===================================================================
--- version.sh	(revision 8645)
+++ version.sh	(working copy)
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-svn_revision=`cd "$1" && LC_ALL=C svn info 2> /dev/null | grep Revision | cut -d' ' -f2`
+svn_revision="${ESVN_WC_REVISION}"
 test $svn_revision || svn_revision=`cd "$1" && grep revision .svn/entries | \
                                     cut -d '"' -f2 2> /dev/null`
 test $svn_revision || svn_revision=UNKNOWN
