Index: configure
===================================================================
--- configure	(revision 8645)
+++ configure	(working copy)
@@ -1593,7 +1593,7 @@
 
 # these are off by default, so fail if requested and not available
 enabled liba52     && require liba52 a52dec/a52.h a52_init -la52
-enabled libdts     && require libdts dts.h dts_init -ldts -lm
+enabled libdts     && require libdts dca.h dts_init -ldca -lm
 enabled libgsm     && require libgsm gsm.h gsm_create -lgsm
 enabled libmp3lame && require LAME lame/lame.h lame_init -lmp3lame -lm
 enabled libtheora  && require libtheora theora/theora.h theora_info_init -ltheora -logg
