Index: version.sh
===================================================================
--- version.sh	(revision 8645)
+++ version.sh	(working copy)
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 # check for SVN revision number
-revision=`cd "$1" && LC_ALL=C svn info 2> /dev/null | grep Revision | cut -d' ' -f2`
+revision="${ESVN_WC_REVISION}"
 test $revision || revision=`cd "$1" && grep revision .svn/entries 2>/dev/null | cut -d '"' -f2`
 test $revision || revision=`cd "$1" && sed -n -e '/^dir$/{n;p;q}' .svn/entries 2>/dev/null`
 test $revision && revision=SVN-r$revision
