diff -ru faad2/configure.in faad2_nu/configure.in
--- faad2/configure.in	2006-08-13 17:13:31.000000000 +0300
+++ faad2_nu/configure.in	2007-03-04 14:33:17.000000000 +0200
@@ -114,12 +114,6 @@
  AM_CONDITIONAL(HAVE_XMMS, false)
 fi
 
-if test x$WITHBMP = xyes; then
-  AC_DEFINE([HAVE_BMP], 1, [User wants beep media player plugin built])
-  AM_CONDITIONAL([HAVE_XMMS], true)
-  AM_CONDITIONAL([HAVE_BMP], true)
-fi
-
 if test x$WITHDRM = xyes; then
   AC_DEFINE(DRM, 1, [Define if you want to use libfaad together with Digital Radio Mondiale (DRM)])
   AC_DEFINE(DRM_PS, 1, [Define if you want support for Digital Radio Mondiale (DRM) parametric stereo])
@@ -128,10 +122,6 @@
 AC_CONFIG_FILES(libfaad/Makefile)
 AC_CONFIG_FILES(common/Makefile)
 AC_CONFIG_FILES(common/mp4ff/Makefile)
-AC_CONFIG_FILES(plugins/Makefile)
-AC_CONFIG_FILES(plugins/xmms/Makefile)
-AC_CONFIG_FILES(plugins/xmms/src/Makefile)
-AC_CONFIG_FILES(plugins/mpeg4ip/Makefile)
 AC_CONFIG_FILES(faad2.spec)
 AC_CONFIG_FILES(frontend/Makefile)
 AC_CONFIG_FILES(Makefile)
diff -ru faad2/Makefile.am faad2_nu/Makefile.am
--- faad2/Makefile.am	2004-07-27 11:52:20.000000000 +0300
+++ faad2_nu/Makefile.am	2007-03-04 14:33:28.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = libfaad common frontend plugins
+SUBDIRS = libfaad common frontend
 
 EXTRA_DIST = faad2.spec
 
