--- ./snd-file.c.orig	2007-06-11 14:33:09.000000000 +0300
+++ ./snd-file.c	2007-09-04 00:13:26.000000000 +0300
@@ -2639,31 +2639,31 @@
     {
 #if HAVE_OGG
     case MUS_OGG:
-      command = mus_format("%s %s -o %s", PATH_OGGENC, input_filename, output_filename);
+      command = mus_format("%s \"%s\" -o \"%s\"", PATH_OGGENC, input_filename, output_filename);
       break;
 #endif
 
 #if HAVE_FLAC
     case MUS_FLAC: 
-      command = mus_format("%s %s -o %s", PATH_FLAC, input_filename, output_filename);
+      command = mus_format("%s \"%s\" -o \"%s\"", PATH_FLAC, input_filename, output_filename);
       break;
 #endif

 #if HAVE_SPEEX
     case MUS_SPEEX:
-      command = mus_format("%s %s %s", PATH_SPEEXENC, input_filename, output_filename);
+      command = mus_format("%s \"%s\" \"%s\"", PATH_SPEEXENC, input_filename, output_filename);
       break;
 #endif

 #if HAVE_TTA
     case MUS_TTA:
-      command = mus_format("%s -e %s -o %s", PATH_TTA, input_filename, output_filename);
+      command = mus_format("%s -e \"%s\" -o \"%s\"", PATH_TTA, input_filename, output_filename);
       break;
 #endif

 #if HAVE_WAVPACK
     case MUS_WAVPACK:
-      command = mus_format("%s %s -o %s", PATH_WAVPACK, input_filename, output_filename);
+      command = mus_format("%s \"%s\" -o \"%s\"", PATH_WAVPACK, input_filename, output_filename);
       break;
 #endif

@@ -2693,53 +2693,53 @@
     {
 #if HAVE_OGG
     case MUS_OGG:
-      command = mus_format("%s %s -b 16 -o %s", PATH_OGGDEC, input_filename, output_filename);
+      command = mus_format("%s \"%s\" -b 16 -o \"%s\"", PATH_OGGDEC, input_filename, output_filename);
       break;
 #endif
 
 #if HAVE_FLAC
     case MUS_FLAC: 
-      command = mus_format("%s -d %s -o %s", PATH_FLAC, input_filename, output_filename);
+      command = mus_format("%s -d -s \"%s\" -o \"%s\"", PATH_FLAC, input_filename, output_filename);
       break;
 #endif

 #if HAVE_SPEEX
     case MUS_SPEEX:
-      command = mus_format("%s %s %s", PATH_SPEEXDEC, input_filename, output_filename);
+      command = mus_format("%s \"%s\" \"%s\"", PATH_SPEEXDEC, input_filename, output_filename);
       break;
 #endif

 #if HAVE_MPEG
     case MUS_MPEG:
 #if HAVE_MPG321
-      command = mus_format("%s -q -w %s %s", PATH_MPG321, output_filename, input_filename);
+      command = mus_format("%s -q -w \"%s\" \"%s\"", PATH_MPG321, output_filename, input_filename);
 #else
-      command = mus_format("%s -q -w %s %s", PATH_MPG123, output_filename, input_filename);
+      command = mus_format("%s -q -w \"%s\" \"%s\"", PATH_MPG123, output_filename, input_filename);
 #endif
       break;
 #endif

 #if HAVE_TIMIDITY
     case MUS_MIDI:
-      command = mus_format("%s %s -Ou -o %s", PATH_TIMIDITY, input_filename, output_filename);
+      command = mus_format("%s \"%s\" -Ou -o \"%s\"", PATH_TIMIDITY, input_filename, output_filename);
       break;
 #endif

 #if HAVE_SHORTEN
     case MUS_SHORTEN:
-      command = mus_format("%s -x %s %s", PATH_SHORTEN, input_filename, output_filename);
+      command = mus_format("%s -x \"%s\" \"%s\"", PATH_SHORTEN, input_filename, output_filename);
       break;
 #endif

 #if HAVE_TTA
     case MUS_TTA:
-      command = mus_format("%s -d %s -o %s", PATH_TTA, input_filename, output_filename);
+      command = mus_format("%s -d \"%s\" -o \"%s\"", PATH_TTA, input_filename, output_filename);
       break;
 #endif

 #if HAVE_WAVPACK
     case MUS_WAVPACK:
-      command = mus_format("%s %s -o %s", PATH_WVUNPACK, input_filename, output_filename);
+      command = mus_format("%s -q \"%s\" -o \"%s\"", PATH_WVUNPACK, input_filename, output_filename);
       break;
 #endif
     default: 
