--- ./makefile.in.orig	2010-04-20 10:06:25.508958802 +0300
+++ ./makefile.in	2010-04-20 13:37:39.674990149 +0300
@@ -182,13 +182,13 @@
 	$(CC) $(LDFLAGS) $(CFLAGS) headers.o io.o audio.o sound.o audinfo.o -o audinfo $(JACK_LIBS) $(AUDIO_LIB) $(LIBS)
 
 install: snd
-	${SHELL} ${top_srcdir}/mkinstalldirs ${bindir}
-	${SHELL} ${top_srcdir}/mkinstalldirs ${mandir}
-	${SHELL} ${top_srcdir}/mkinstalldirs ${mandir}/man1
-	${SHELL} ${top_srcdir}/mkinstalldirs ${pkgdatadir}
-	$(INSTALL) snd ${bindir}/snd
-	$(INSTALL_DATA) ${top_srcdir}/snd.1 ${mandir}/man1
-	(cd ${top_srcdir} && for f in ${SND_SCRIPTS}; do ${INSTALL_DATA} ${top_srcdir}/$${f} ${pkgdatadir}/$${f}; done)
+	${SHELL} ${top_srcdir}/mkinstalldirs ${DESTDIR}/${bindir}
+	${SHELL} ${top_srcdir}/mkinstalldirs ${DESTDIR}/${mandir}
+	${SHELL} ${top_srcdir}/mkinstalldirs ${DESTDIR}/${mandir}/man1
+	${SHELL} ${top_srcdir}/mkinstalldirs ${DESTDIR}/${pkgdatadir}
+	$(INSTALL) snd ${DESTDIR}/${bindir}/snd
+	$(INSTALL_DATA) ${top_srcdir}/snd.1 ${DESTDIR}/${mandir}/man1
+	(cd ${top_srcdir} && for f in ${SND_SCRIPTS}; do ${INSTALL_DATA} ${top_srcdir}/$${f} ${DESTDIR}/${pkgdatadir}/$${f}; done)
 
 uninstall:
 	rm -f ${bindir}/snd
