--- ./makefile.in.orig	2007-04-15 21:57:50.000000000 +0300
+++ ./makefile.in	2007-09-04 11:43:01.000000000 +0300
@@ -64,7 +64,7 @@
 GX_HEADERS = $(@GX_HEADERS@)
 
 .SUFFIXES:
-.SUFFIXES: .c .o
+.SUFFIXES: .c .o .obj
 
 .c.o:
 	$(CC) -c $(DEFS) $(GTK_CFLAGS) $(CAIRO_CFLAGS) $(CFLAGS) $(XEN_CFLAGS) $(MOTIF_FLAGS) $(GSL_FLAGS) $(GL_FLAGS) $(JACK_FLAGS) $(FFTW_CFLAGS) $<
@@ -96,7 +96,9 @@
 GG_SO_FILE = xg.so
 NO_FILES =
 
-main_target: @MAKE_TARGET@
+UTILS = sndplay audinfo sndinfo
+
+main_target: @MAKE_TARGET@ $(UTILS)
 
 snd: $(SNDLIB_HEADERS) $(SND_HEADERS) $(GX_HEADERS) $(S7_HEADERS) $(S7_O_FILES) $(SNDLIB_FILES) $(O_FILES) $(GX_FILES) $(GL_FILES)
 	$(CC) $(LDFLAGS) $(CFLAGS) $(S7_O_FILES) $(SNDLIB_FILES) $(O_FILES) $(GX_FILES) $(GL_FILES) -o snd $(SNDLIB_LIB) $(XEN_LIBS) $(GTK_LIBS) $(GL_LIBS) $(MOTIF_LIBS) $(JACK_LIBS) $(AUDIO_LIB) $(FFTW_LIBS) $(GSL_LIBS) $(FAM_LIB) $(GMP_LIBS) $(LIBS)
@@ -157,29 +161,22 @@
 	rm -f *.a
 	rm -f snd sndplay audinfo sndinfo
 
-sndplay: $(SNDLIB_HEADERS) $(S7_HEADERS)
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 headers.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 io.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 audio.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 sound.c
+.c.obj:
+	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 -DHAVE_RUBY=0 $< -o $@
+
+UTILS_OBJ = headers.obj io.obj audio.obj sound.obj
+
+sndplay: $(SNDLIB_HEADERS) $(S7_HEADERS) $(UTILS_OBJ)
 	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 sndplay.c
-	$(CC) $(LDFLAGS) $(CFLAGS) headers.o io.o audio.o sound.o sndplay.o -o sndplay $(JACK_LIBS) $(AUDIO_LIB) $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) $(UTILS_OBJ) sndplay.o -o sndplay $(JACK_LIBS) $(AUDIO_LIB) $(LIBS)
 
-sndinfo: $(SNDLIB_HEADERS) $(S7_HEADERS)
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 headers.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 io.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 audio.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 sound.c
+sndinfo: $(SNDLIB_HEADERS) $(S7_HEADERS) $(UTILS_OBJ)
 	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 sndinfo.c
-	$(CC) $(LDFLAGS) $(CFLAGS) headers.o io.o audio.o sound.o sndinfo.o -o sndinfo $(JACK_LIBS) $(AUDIO_LIB) $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) $(UTILS_OBJ) sndinfo.o -o sndinfo $(JACK_LIBS) $(AUDIO_LIB) $(LIBS)
 
-audinfo: $(SNDLIB_HEADERS) $(S7_HEADERS)
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 headers.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 io.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 audio.c
-	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 sound.c
+audinfo: $(SNDLIB_HEADERS) $(S7_HEADERS) $(UTILS_OBJ)
 	$(CC) -c $(DEFS) $(CFLAGS) -DUSE_SND=0 audinfo.c
-	$(CC) $(LDFLAGS) $(CFLAGS) headers.o io.o audio.o sound.o audinfo.o -o audinfo $(JACK_LIBS) $(AUDIO_LIB) $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) $(UTILS_OBJ) audinfo.o -o audinfo $(JACK_LIBS) $(AUDIO_LIB) $(LIBS)
 
 install: snd
 	${SHELL} ${top_srcdir}/mkinstalldirs ${DESTDIR}/${bindir}
@@ -195,7 +188,7 @@
 	${SHELL} ${top_srcdir}/mkinstalldirs ${DESTDIR}/${mandir}
 	${SHELL} ${top_srcdir}/mkinstalldirs ${DESTDIR}/${mandir}/man1
 	${SHELL} ${top_srcdir}/mkinstalldirs ${DESTDIR}/${pkgdatadir}
-	$(INSTALL) snd ${DESTDIR}/${bindir}/snd
+	for x in snd $(UTILS); do $(INSTALL) $${x} ${DESTDIR}/${bindir}; done
 	$(INSTALL_DATA) ${top_srcdir}/snd.1 ${DESTDIR}/${mandir}/man1
 	(cd ${top_srcdir} && for f in ${SND_SCRIPTS}; do ${INSTALL_DATA} ${top_srcdir}/$${f} ${DESTDIR}/${pkgdatadir}/$${f}; done)
 
