Index: emesene/setup.py
===================================================================
--- emesene/setup.py	(revision 992)
+++ emesene/setup.py	(working copy)
@@ -16,9 +16,9 @@
       package_data = {'emesene': ['conversation_themes/*/*', 'po/*/LC_MESSAGES/*.mo',
                                   'smilies/*/*', 'sound_themes/*/*', 'themes/*/*',
                                   'emesene.bat', 'hotmlog.htm', 'setup']},
-      #scripts      = ['emesene'],
-      data_files   = [('share/pixmaps', ['misc/emesene.png']),
-                      ('share/icons/hicolor/scalable/apps', ['misc/emesene.svg']),
-                      ('share/man/man1', ['misc/emesene.1']),
-                      ('share/applications', ['misc/emesene.desktop'])]
+      scripts      = ['emesene/emesene'],
+      data_files   = [('share/pixmaps', ['emesene/misc/emesene.png']),
+                      ('share/icons/hicolor/scalable/apps', ['emesene/misc/emesene.svg']),
+                      ('share/man/man1', ['emesene/misc/emesene.1']),
+                      ('share/applications', ['emesene/misc/emesene.desktop'])]
       )
