Index: cmake/admGetRevision.cmake
===================================================================
--- cmake/admGetRevision.cmake	(revision 5122)
+++ cmake/admGetRevision.cmake	(working copy)
@@ -16,7 +16,7 @@
                 SET(${_rev} ${ADM_GIT_SVN_REVISION})
         else (EXISTS "${_dir}/.git")
                 MESSAGE(STATUS "Dont know what SCM is used")
-                SET(${_rev} "0")
+                SET(${_rev} ${Project_WC_REVISION})
         endif (EXISTS "${_dir}/.git")
 endif (EXISTS "${_dir}/.svn")
                 #MESSAGE( STATUS "revision : ${${_rev}}" )
