diff --git a/CMakeLists.txt b/CMakeLists.txt
index e01e1ab..49f2eec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,6 +200,7 @@ MESSAGE("")
 # Directories to build
 ########################################
 ADD_SUBDIRECTORY(avidemux)
+ADD_SUBDIRECTORY(plugins)
 ADD_SUBDIRECTORY(po)
 
 ########################################
diff --git a/cmake/admPluginVideoEncoder.cmake b/cmake/admPluginVideoEncoder.cmake
index 8f82419..3303dfe 100644
--- a/cmake/admPluginVideoEncoder.cmake
+++ b/cmake/admPluginVideoEncoder.cmake
@@ -1,8 +1,8 @@
 MACRO(INIT_VIDEO_ENCODER_PLUGIN _lib)
 	INCLUDE_DIRECTORIES("${CMAKE_BINARY_DIR}/config")
-	INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/../avidemux/ADM_core/include")
-	INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/../avidemux/ADM_coreUI/include")
-	INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/../avidemux/ADM_plugin")
+	INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/avidemux/ADM_core/include")
+	INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/avidemux/ADM_coreUI/include")
+	INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/avidemux/ADM_plugin")
 ENDMACRO(INIT_VIDEO_ENCODER_PLUGIN)
 
 MACRO(INSTALL_VIDEO_ENCODER _lib)
diff --git a/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt b/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt
index cc564b1..66a5a33 100644
--- a/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt
+++ b/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt
@@ -1,7 +1,7 @@
 	INCLUDE(ae_plugin)
-        INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/../avidemux/")
-        INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/../avidemux/ADM_libraries/ADM_ffmpeg/ADM_lavcodec/")
-        INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/../avidemux/ADM_libraries/ADM_ffmpeg/ADM_lavutil/")
+        INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/avidemux/")
+        INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/avidemux/ADM_libraries/ADM_ffmpeg/ADM_lavcodec/")
+        INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/avidemux/ADM_libraries/ADM_ffmpeg/libavutil/")
 
 	SET(ADM_ae_lav_mp2_SRCS audioencoder_lavcodec.cpp)
 
