--- configure.in.in.orig	2006-10-24 04:23:33.000000000 +0300
+++ configure.in.in	2006-10-24 14:00:39.000000000 +0300
@@ -53,6 +53,7 @@
 dnl Ranlib
 AC_PROG_RANLIB
 AC_PROG_CXX
+AM_PROG_AS
 AC_PROG_LIBTOOL
 dnl without this order in this file, automake will be confused!
 dnl
@@ -115,6 +116,7 @@
 CPPFLAGS="  $USER_INCLUDES  $CPPFLAGS"
 dnl __________gettext_________________
 AM_GNU_GETTEXT( [external])
+AM_GNU_GETTEXT_VERSION([0.11.5])
 GETTEXT_PACKAGE=avidemux
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE], "$GETTEXT_PACKAGE",
                    [gettext package name])
