Index: avidemux/common/ADM_render/CMakeLists.txt
===================================================================
--- avidemux/common/ADM_render/CMakeLists.txt	(revision 4945)
+++ avidemux/common/ADM_render/CMakeLists.txt	(working copy)
@@ -21,4 +21,8 @@
         TARGET_LINK_LIBRARIES(${ADM_LIB}  ${SDL_LIBRARY})
 ENDIF (USE_SDL)
 
+IF (USE_XV)
+        TARGET_LINK_LIBRARIES(${ADM_LIB}  ${XVIDEO_LIBRARY_DIR})
+ENDIF (USE_XV)
+
 ADM_INSTALL_LIB(${ADM_LIB})
