Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 3679)
+++ CMakeLists.txt	(working copy)
@@ -42,8 +42,6 @@
 	FIND_PACKAGE(Subversion)
 	IF(Subversion_FOUND)
 		Subversion_WC_INFO(${PROJECT_SOURCE_DIR} Project)
-		MESSAGE(STATUS "Current revision is ${Project_WC_REVISION}")
-		SET(ADM_SUBVERSION ${Project_WC_REVISION})
 	ELSE(Subversion_FOUND)
 		MESSAGE(STATUS "Could not find Subversion")
 	ENDIF(Subversion_FOUND)
@@ -51,6 +49,9 @@
 
 IF(NOT Subversion_FOUND)
 	SET(ADM_SUBVERSION 0)
+ELSE(NOT Subversion_FOUND)
+	MESSAGE(STATUS "Current revision is ${Project_WC_REVISION}")
+	SET(ADM_SUBVERSION ${Project_WC_REVISION})
 ENDIF(NOT Subversion_FOUND)
 
 #**************************************************
