Index: po/CMakeLists.txt
===================================================================
--- po/CMakeLists.txt	(revision 4708)
+++ po/CMakeLists.txt	(working copy)
@@ -6,8 +6,9 @@
 ENDIF (HAVE_GETTEXT)
 
 IF (QT4_FOUND)
-	INCLUDE(Ts)
-	COMPILE_TS_FILES(${CMAKE_SOURCE_DIR}/po QM_FILES)
+	FILE(GLOB TRANSLATIONS_FILES ${CMAKE_SOURCE_DIR}/po/*.ts)
+	QT4_ADD_TRANSLATION(QM_FILES ${TRANSLATIONS_FILES})
+	INSTALL(FILES ${QM_FILES} DESTINATION "${CMAKE_INSTALL_PREFIX}/bin/i18n")
 	
 	ADD_CUSTOM_TARGET(qmfiles ALL DEPENDS ${QM_FILES})
 ENDIF (QT4_FOUND)
