Index: avidemux/ADM_audiocodec/ADM_dca.cpp
===================================================================
--- avidemux/ADM_audiocodec/ADM_dca.cpp	(revision 2926)
+++ avidemux/ADM_audiocodec/ADM_dca.cpp	(working copy)
@@ -32,7 +32,6 @@
 extern "C"
 {
 #include "dts.h"
-#include "dts_internal.h"
 
 }
 #include "ADM_audio/ADM_dcainfo.h"
Index: configure.in.in
===================================================================
--- configure.in.in	(revision 2937)
+++ configure.in.in	(working copy)
@@ -854,7 +854,6 @@
     AC_CHECK_HEADERS([dts.h], , have_libdca=no)
     
     if test "x$have_libdca" = "xyes"; then
-    	AC_CHECK_HEADERS([dts_internal.h], , have_libdca=no)
     	
     	if test "x$have_libdca" = "xyes"; then
             AC_CHECK_LIB(dts, dts_init, , have_libdca=no)
