--- ./test/Makefile.orig	2007-06-07 15:36:39.000000000 +0300
+++ ./test/Makefile	2007-06-16 20:27:39.000000000 +0300
@@ -128,8 +128,6 @@
 	button$(EXEEXT) \
 	buttons$(EXEEXT) \
 	color_chooser$(EXEEXT) \
-	cube$(EXEEXT) \
-	CubeView$(EXEEXT) \
 	cursor$(EXEEXT) \
 	curve$(EXEEXT) \
 	demo$(EXEEXT) \
@@ -140,9 +138,6 @@
 	exception$(EXEEXT) \
 	file_chooser$(EXEEXT) \
 	fonts$(EXEEXT) \
-	fractals$(EXEEXT) \
-	glpuzzle$(EXEEXT) \
-	gl_overlay$(EXEEXT) \
 	hello$(EXEEXT) \
 	image$(EXEEXT) \
 	inactive$(EXEEXT) \
@@ -161,11 +156,9 @@
 	pixmap$(EXEEXT) \
 	progress$(EXEEXT) \
 	radio$(EXEEXT) \
-	qubix$(EXEEXT) \
 	resizable$(EXEEXT) \
 	resizealign$(EXEEXT) \
 	scroll$(EXEEXT) \
-	shape$(EXEEXT) \
 	sizes$(EXEEXT) \
 	symbols$(EXEEXT) \
 	threads$(EXEEXT) \
@@ -173,7 +166,6 @@
 	tile$(EXEEXT) \
 	timer$(EXEEXT) \
 	utf$(EXEEXT) \
-	fullscreen$(EXEEXT) \
 	valuators$(EXEEXT) \
 	list$(EXEEXT) \
 	wizard$(EXEEXT)
--- ./Makefile.orig	2006-07-11 12:32:27.000000000 +0300
+++ ./Makefile	2007-06-16 20:22:34.000000000 +0300
@@ -25,7 +25,7 @@
 
 include makeinclude
 
-DIRS = src $(LOCALIMAGES) images OpenGL fluid glut test
+DIRS = src $(LOCALIMAGES) images fluid test
 
 all: makeinclude
 	for dir in $(DIRS); do\
