Index: makeinclude.in
===================================================================
--- makeinclude.in	(revision 6006)
+++ makeinclude.in	(working copy)
@@ -115,31 +115,25 @@
 CAT3EXT		= @CAT3EXT@
 
 # Be quiet when building...
-.SILENT:
 
 # Build commands and filename extensions...
 .SUFFIXES:	.0 .1 .3 .c .cxx .h .fl .man .o .z $(EXEEXT)
 
 .o$(EXEEXT):
-	echo Linking $@...
 	$(CXX) -I.. -I../fltk/compat $(CXXFLAGS) $< $(LINKFLTK) $(LDLIBS) -o $@
 	$(POSTBUILD) $@ ../fltk/mac.r
 
 .c.o:
-	echo Compiling $<...
 	$(CC) -I.. -I../fltk/compat $(CFLAGS) -c $<
 
 .cxx.o:
-	echo Compiling $<...
 	$(CXX) -I.. -I../fltk/compat $(CXXFLAGS) -c $<
 
 .man.0 .man.1 .man.3:
-	echo Formatting $<...
 	rm -f $@
 	$(NROFF) -man $< >$@
 
 .man.z:
-	echo Formatting $<...
 	rm -f $@ t.z
 	$(NROFF) -man $< >t
 	pack -f t
