Index: src/utils.cpp
===================================================================
--- src/utils.cpp	(revision 27)
+++ src/utils.cpp	(working copy)
@@ -22,6 +22,7 @@
 #  include "config.h"
 #endif
 
+#include <cstdlib>
 #include <glib.h>
 #include <glib/gi18n.h>
 
Index: src/readline.cpp
===================================================================
--- src/readline.cpp	(revision 27)
+++ src/readline.cpp	(working copy)
@@ -23,6 +23,7 @@
 #endif
 
 #include <cstdio>
+#include <cstdlib>
 #ifdef WITH_READLINE
 #  include <readline/readline.h>
 #  include <readline/history.h>
Index: src/libwrapper.cpp
===================================================================
--- src/libwrapper.cpp	(revision 27)
+++ src/libwrapper.cpp	(working copy)
@@ -24,6 +24,7 @@
 
 #include <glib/gi18n.h>
 #include <map>
+#include <cstring>
 
 #include "utils.hpp"
 
