--- build/linux/Makefile.orig	2007-02-17 19:48:28.000000000 +0200
+++ build/linux/Makefile	2007-02-17 19:48:28.000000000 +0200
@@ -201,9 +201,9 @@
 	@echo "  I: $(DESTDIR)$(libdir)/$(SHARED_LIB)"
 	@$(INSTALL) -m 644 $(BUILD_DIR)/$(SHARED_LIB) $(DESTDIR)$(libdir)/$(SHARED_LIB)
 	@echo "  I: $(DESTDIR)$(libdir)/libavisynth.$(SHARED_EXTENSION).$(API_MAJOR)"
-	@$(LN_S) -f $(libdir)/$(SHARED_LIB) $(DESTDIR)$(libdir)/libavisynth.$(SHARED_EXTENSION).$(API_MAJOR)
+	@$(LN_S) -f $(SHARED_LIB) $(DESTDIR)$(libdir)/libavisynth.$(SHARED_EXTENSION).$(API_MAJOR)
 	@echo "  I: $(DESTDIR)$(libdir)/libavisynth.$(SHARED_EXTENSION)"
-	@$(LN_S) -f $(libdir)/$(SHARED_LIB) $(DESTDIR)$(libdir)/libavisynth.$(SHARED_EXTENSION)
+	@$(LN_S) -f $(SHARED_LIB) $(DESTDIR)$(libdir)/libavisynth.$(SHARED_EXTENSION)
 	@echo "  I: $(DESTDIR)$(libdir)/$(STATIC_LIB)"
 	@$(INSTALL) -m 644 $(BUILD_DIR)/$(STATIC_LIB) $(DESTDIR)$(libdir)/$(STATIC_LIB)
 	@echo "  D: $(DESTDIR)$(includedir)"
