Index: build/linux/configure.in
===================================================================
--- build/linux/configure.in	(revision 2345)
+++ build/linux/configure.in	(working copy)
@@ -434,7 +434,11 @@
       dnl Checking nasm patch version
       dnl
       AC_MSG_CHECKING([for nasm patch version])
-      nasm_patch=`nasm -r | cut -d '.' -f 3 | cut -d ' ' -f 1`
+      nasm_patch=`nasm -v | cut -d '.' -f 3 | cut -d ' ' -f 1`
+      nasm_major=`nasm -v | cut -d '.' -f 1 | cut -d ' ' -f 3`
+      if test "${nasm_major}" -ge 2 ; then
+        nasm_patch="${minimum_nasm_patch_version}"
+      fi
       if test -z $nasm_patch ; then
         nasm_patch=-1
       fi
