Index: build/linux/Makefile
===================================================================
RCS file: /cvsroot/avisynth2/avisynth/build/linux/Attic/Makefile,v
retrieving revision 1.1.2.4.2.14
diff -u -r1.1.2.4.2.14 Makefile
--- build/linux/Makefile	19 Dec 2005 22:50:27 -0000	1.1.2.4.2.14
+++ build/linux/Makefile	21 Jan 2006 15:47:27 -0000
@@ -159,7 +159,7 @@
 	@echo "  I: $(bindir)/$(BIN_OBJECT)"
 	@$(INSTALL) -m 755 $(BUILD_DIR)/$(PROG) $(DEST_DIR)$(bindir)/$(PROG)
 	@echo "  D: $(libdir)"
-	@$(INSTALL) -d $(libdir)
+	@$(INSTALL) -d $(DEST_DIR)$(libdir)
 	@echo "  I: $(libdir)/$(SHARED_LIB)"
 	@$(INSTALL) -m 644 $(BUILD_DIR)/$(SHARED_LIB) $(DEST_DIR)$(libdir)/$(SHARED_LIB)
 	@echo "  I: $(libdir)/libavisynth.$(SHARED_EXTENSION).$(API_MAJOR)"
@@ -178,12 +178,12 @@
 	    dir=""; \
 	  fi; \
 	  if test "$$dir" != "avisynth_c/private"; then \
-	    install -d $(includedir)/$$dir; \
+	    install -d $(DEST_DIR)$(includedir)/$$dir; \
 	    echo "  I: $(includedir)/$$filename"; \
 	    if test "$$filename" = "core.h" ; then \
-	      install -m 644 $$filename $(includedir)/avisynth.h; \
+	      install -m 644 $$filename $(DEST_DIR)$(includedir)/avisynth.h; \
 	    else \
-	      install -m 644 $$filename $(includedir)/$$filename; \
+	      install -m 644 $$filename $(DEST_DIR)$(includedir)/$$filename; \
 	    fi; \
 	  fi \
 	done
