Index: build/linux/m4/stlport.m4
===================================================================
RCS file: /cvsroot/avisynth2/avisynth/build/linux/m4/Attic/stlport.m4,v
retrieving revision 1.1.2.8
diff -u -r1.1.2.8 stlport.m4
--- build/linux/m4/stlport.m4	16 Dec 2005 20:51:02 -0000	1.1.2.8
+++ build/linux/m4/stlport.m4	20 Jan 2006 17:20:31 -0000
@@ -23,16 +23,16 @@
     case "$1" in
        *bsd* | linux* | irix* | solaris* )
           if test x"${core_debug_mode}" = x"yes" ; then
-             STLPORT_LIB_NAME="stlportg"
+             STLPORT_LIB_NAME="stlport_gcc_stldebug"
           else
-             STLPORT_LIB_NAME="stlport"
+             STLPORT_LIB_NAME="stlport_gcc"
           fi
        ;;
        [[cC]][[yY]][[gG]][[wW]][[iI]][[nN]]*|mingw32*|mks*)
           if test x"${core_debug_mode}" = x"yes" ; then
-             STLPORT_LIB_NAME="stlportg"
+             STLPORT_LIB_NAME="stlport_gcc_stldebug"
           else
-             STLPORT_LIB_NAME="stlport"
+             STLPORT_LIB_NAME="stlport_gcc"
           fi
        ;;
        darwin*|raphsody*)
