Index: build/linux/configure.in
===================================================================
--- build/linux/configure.in	(revision 2345)
+++ build/linux/configure.in	(working copy)
@@ -34,7 +34,7 @@
 
 dnl BOOST version requirement
 BOOST_VERSION_MAJOR="1"
-BOOST_VERSION_MINOR="33"
+BOOST_VERSION_MINOR="34"
 BOOST_VERSION="${BOOST_VERSION_MAJOR}_${BOOST_VERSION_MINOR}"
 
 dnl Gstreamer version requirement
