Index: src/core/cache/cachelogic.h
===================================================================
RCS file: /cvsroot/avisynth2/avisynth/src/core/cache/Attic/cachelogic.h,v
retrieving revision 1.1.2.5
diff -u -r1.1.2.5 cachelogic.h
--- src/core/cache/cachelogic.h	15 Mar 2005 09:44:39 -0000	1.1.2.5
+++ src/core/cache/cachelogic.h	20 Jan 2006 20:29:02 -0000
@@ -28,7 +28,7 @@
 #include "base.h"
 
 //boost include
-#include <boost/circular_buffer.hpp>
+#include <circular_buffer/circular_buffer.hpp>
 //actually a boost extension, grab it at http://groups.yahoo.com/group/boost/files/circular_buffer_v3.6.zip
 
 
Index: build/linux/platform.inc.in
===================================================================
RCS file: /cvsroot/avisynth2/avisynth/build/linux/Attic/platform.inc.in,v
retrieving revision 1.1.2.6.2.12
diff -u -r1.1.2.6.2.12 platform.inc.in
--- build/linux/platform.inc.in	29 Dec 2005 13:55:15 -0000	1.1.2.6.2.12
+++ build/linux/platform.inc.in	20 Jan 2006 20:29:02 -0000
@@ -35,7 +35,7 @@
 FREETYPE_CFLAGS=@FREETYPE_CFLAGS@
 CORE_DEBUG_MODE_CFLAGS=@CORE_DEBUG_MODE_CFLAGS@
 
-AVS3_CPPFLAGS=@STLPORT_CFLAGS@ @BOOST_CFLAGS@ @GST_CFLAGS@ @FREETYPE_CFLAGS@ @VISIBILITY_CPPFLAGS@
+AVS3_CPPFLAGS=@STLPORT_CFLAGS@ @BOOST_CFLAGS@ -I$(PWD) @GST_CFLAGS@ @FREETYPE_CFLAGS@ @VISIBILITY_CPPFLAGS@
 AVS3_DEBUG_CFLAGS=@CORE_DEBUG_MODE_CFLAGS@
 AVS3_CXXFLAGS=@CXXFLAGS@ @SPECIFIC_CFLAGS@ @VISIBILITY_CFLAGS@
 
Index: build/linux/m4/boost.m4
===================================================================
RCS file: /cvsroot/avisynth2/avisynth/build/linux/m4/Attic/boost.m4,v
retrieving revision 1.1.2.5
diff -u -r1.1.2.5 boost.m4
--- build/linux/m4/boost.m4	8 Oct 2005 15:39:52 -0000	1.1.2.5
+++ build/linux/m4/boost.m4	20 Jan 2006 20:29:03 -0000
@@ -25,7 +25,7 @@
           if test x"${core_debug_mode}" = x"yes" ; then
              BOOST_LIB_NAME="boost_thread-gcc-mt-d-$1"
           else
-             BOOST_LIB_NAME="boost_thread-gcc-mt-$1"
+             BOOST_LIB_NAME="boost_thread-gcc-mt"
           fi
           
        ;;
@@ -56,7 +56,7 @@
     dnl We check the headers, then the library.
     saved_CPPFLAGS="${CPPFLAGS}"
     saved_LDFLAGS="${LDFLAGS}"
-    CPPFLAGS="${CPPFLAGS} -I${boost_path}/include/boost-$1"
+    CPPFLAGS="${CPPFLAGS} -I${boost_path}/include/boost"
     AC_CHECK_HEADER(
        [boost/config.hpp],
        [
@@ -90,9 +90,9 @@
               BOOST_LIB_NAME=""
            ;;
         esac
-        BOOST_CFLAGS="-I${boost_path}/include/boost-$1"
+        BOOST_CFLAGS="-I${boost_path}/include/boost"
         m4_if([$3], [], [:], [$3])],
-       [AC_MSG_WARN(Boost headers not in ${boost_path}/include/boost-$1)
+       [AC_MSG_WARN(Boost headers not in ${boost_path}/include/boost)
         m4_if([$4], [], [:], [$4])])
     CPPFLAGS="${saved_CPPFLAGS}"
     LDFLAGS="${saved_LDFLAGS}"
