diff -r 582fb6f55103 Makefile
--- a/Makefile	Wed Feb 06 11:37:45 2008 -0500
+++ b/Makefile	Mon Feb 11 22:30:33 2008 +0200
@@ -1,6 +1,5 @@ CC=gcc
 CC=gcc
 AM_CFLAGS = -Wall -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2
-CFLAGS = -g -Werror
 objects = ctree.o disk-io.o radix-tree.o extent-tree.o print-tree.o \
 	  root-tree.o dir-item.o hash.o file-item.o inode-item.o \
 	  inode-map.o crc32c.o rbtree.o extent-cache.o extent_map.o \
