--- a/Lib/fontmake/font_project.py
+++ b/Lib/fontmake/font_project.py
@@ -262,7 +262,7 @@ class FontProject:
                     GLYPHS_PREFIX + "Don't use Production Names")
             compiler_options = dict(
                 featureCompilerClass=fea_compiler,
-                mtiFeaFiles=mti_paths[name] if mti_paths is not None else None,
+                mtiFeaFiles=mti_paths[name] if mti_paths is not None and name in mti_paths else None,
                 kernWriterClass=kern_writer_class, markWriterClass=mark_writer_class,
                 glyphOrder=ufo.lib.get(PUBLIC_PREFIX + 'glyphOrder'),
                 useProductionNames=use_production_names
