--- a/Source/Exo Pro Italic.glyphs	
+++ b/Source/Exo Pro Italic.glyphs	
@@ -41948,8 +41948,8 @@ components = (
 name = "Ii-cy";
 },
 {
-name = "breve-cy";
-transform = "{1, 0, 0, 1, 174, 186}";
+name = "brevecomb-cy";
+transform = "{1, 0, 0, 1, 60, 260}";
 }
 );
 layerId = "E21536AE-9819-4FE7-8A20-8865FE3C0515";
@@ -41961,8 +41961,8 @@ components = (
 name = "Ii-cy";
 },
 {
-name = "breve-cy";
-transform = "{1, 0, 0, 1, 177, 203}";
+name = "brevecomb-cy";
+transform = "{1, 0, 0, 1, 94, 235}";
 }
 );
 layerId = "924B7471-43FB-4CAD-AF7E-45215B64D997";
@@ -41974,8 +41974,8 @@ components = (
 name = "Ii-cy";
 },
 {
-name = "breve-cy";
-transform = "{1, 0, 0, 1, 175, 188}";
+name = "brevecomb-cy";
+transform = "{1, 0, 0, 1, 82, 243}";
 }
 );
 layerId = "77EC9CE6-E33A-4866-9665-0506F15B57B2";
@@ -42044,8 +42044,8 @@ components = (
 name = "Ii-cy";
 },
 {
-name = "breve-cy";
-transform = "{1, 0, 0, 1, 174, 186}";
+name = "brevecomb-cy";
+transform = "{1, 0, 0, 1, 68, 264}";
 }
 );
 layerId = "E21536AE-9819-4FE7-8A20-8865FE3C0515";
@@ -42071,8 +42071,8 @@ components = (
 name = "Ii-cy";
 },
 {
-name = "breve-cy";
-transform = "{1, 0, 0, 1, 179, 188}";
+name = "brevecomb-cy";
+transform = "{1, 0, 0, 1, 60, 260}";
 }
 );
 layerId = "77EC9CE6-E33A-4866-9665-0506F15B57B2";
@@ -42098,8 +42098,8 @@ components = (
 name = "Ii-cy";
 },
 {
-name = "breve-cy";
-transform = "{1, 0, 0, 1, 177, 203}";
+name = "brevecomb-cy";
+transform = "{1, 0, 0, 1, 89, 232}";
 }
 );
 layerId = "924B7471-43FB-4CAD-AF7E-45215B64D997";
