Index: Makefile
===================================================================
RCS file: /cvsroot/mplayer/main/Makefile,v
retrieving revision 1.355
diff -u -r1.355 Makefile
--- Makefile	26 Mar 2006 21:44:55 -0000	1.355
+++ Makefile	31 Mar 2006 16:44:40 -0000
@@ -61,8 +61,13 @@
 SRCS_COMMON += unrarlib.c
 endif
 
-OBJS_MENCODER = $(SRCS_MENCODER:.c=.o)
-OBJS_MPLAYER = $(SRCS_MPLAYER:.c=.o)
+SRCS = $(SRCS_COMMON) $(SRCS_MPLAYER)
+
+ifeq ($(MENCODER),yes)
+SRCS += $(SRCS_MENCODER)
+endif
+
+OBJS = $(SRCS:.c=.o)
 
 VO_LIBS = $(AA_LIB) \
           $(X_LIB) \
@@ -352,17 +357,19 @@
 libmenu/libmenu.a:
 	$(MAKE) -C libmenu
 
-MPLAYER_DEP = $(OBJS_MPLAYER) $(COMMON_DEPS)
+DEP = $(OBJS) $(COMMON_DEPS)
 
 ifeq ($(LIBMENU),yes)
-MPLAYER_DEP += libmenu/libmenu.a
+DEP += libmenu/libmenu.a
 MENU_LIBS = libmenu/libmenu.a
 PARTS += libmenu
 else
 MENU_LIBS =
 endif
 
-MENCODER_DEP = $(OBJS_MENCODER) $(COMMON_DEPS) libmpcodecs/libmpencoders.a
+ifeq ($(MENCODER),yes)
+DEP += libmpcodecs/libmpencoders.a
+endif
 
 ifeq ($(VIDIX),yes)
 VIDIX_LIBS = vidix/libvidix.a
@@ -371,70 +378,65 @@
 endif
 
 ifeq ($(TARGET_WIN32),yes)
-OBJS_MPLAYER += osdep/mplayer-rc.o
+OBJS += osdep/mplayer-rc.o
 endif
 
-LIBS_MPLAYER = libvo/libvo.a \
-               libao2/libao2.a \
-               $(MENU_LIBS) \
-               $(VIDIX_LIBS) \
-               $(GUI_LIBS) \
-               $(COMMON_LIBS) \
-               $(GTK_LIBS) \
-               $(VO_LIBS) \
-               $(AO_LIBS) \
-               $(EXTRA_LIB)\
-               $(LIRC_LIB) \
-               $(LIRCC_LIB) \
-               $(STATIC_LIB) \
-               $(ARCH_LIB) \
-               $(I18NLIBS) \
-               $(MATH_LIB) \
-               $(LIBC_LIB) \
+ifeq ($(MENCODER),yes)
+LIBS += libmpcodecs/libmpencoders.a \
+        $(ENCORE_LIB) \
+        $(MLIB_LIB) \
+
+endif
 
-$(PRG):	$(MPLAYER_DEP)
+LIBS += libvo/libvo.a \
+        libao2/libao2.a \
+        $(MENU_LIBS) \
+        $(VIDIX_LIBS) \
+        $(GUI_LIBS) \
+        $(COMMON_LIBS) \
+        $(GTK_LIBS) \
+        $(VO_LIBS) \
+        $(AO_LIBS) \
+        $(EXTRA_LIB)\
+        $(LIRC_LIB) \
+        $(LIRCC_LIB) \
+        $(STATIC_LIB) \
+        $(ARCH_LIB) \
+        $(I18NLIBS) \
+        $(MATH_LIB) \
+        $(LIBC_LIB) \
+
+$(PRG):	$(DEP)
     ifeq ($(TARGET_WIN32),yes)
 	windres -o osdep/mplayer-rc.o osdep/mplayer.rc
     endif
-	$(CC) $(CFLAGS) -o $(PRG) $(OBJS_MPLAYER) $(LIBS_MPLAYER)
+	$(CC) $(CFLAGS) -o $(PRG) $(OBJS) $(LIBS)
 
 mplayer.exe.spec.c: libmpcodecs/libmpcodecs.a
 	winebuild -fPIC -o mplayer.exe.spec.c -exe mplayer.exe -mcui \
 	libmpcodecs/ad_qtaudio.o libmpcodecs/vd_qtvideo.o \
 	-L/usr/local/lib/wine -lkernel32
 
-mplayer.exe.so:	$(MPLAYER_DEP) mplayer.exe.spec.c
+mplayer.exe.so:	$(MPLAYER) mplayer.exe.spec.c
 	$(CC) $(CFLAGS) -Wall -shared \
 	-Wl,-rpath,/usr/local/lib -Wl,-Bsymbolic \
-	-o mplayer.exe.so $(OBJS_MPLAYER) mplayer.exe.spec.c \
+	-o mplayer.exe.so $(OBJS) mplayer.exe.spec.c \
 	libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) \
 	$(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) \
 	$(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) \
 	$(STATIC_LIB) $(ARCH_LIB) -lwine $(MATH_LIB) \
 
-mplayer_wine.so:	$(MPLAYER_DEP)
+mplayer_wine.so:	$(MPLAYER)
 	$(CC) $(CFLAGS) -shared -Wl,-Bsymbolic -o mplayer_wine.so \
-          mplayer_wine.spec.c $(OBJS_MPLAYER) libvo/libvo.a \
+          mplayer_wine.spec.c $(OBJS) libvo/libvo.a \
 	  libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) \
 	  $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) \
 	  $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) \
 	  -lwine $(ARCH_LIB) $(MATH_LIB) \
 
 ifeq ($(MENCODER),yes)
-LIBS_MENCODER = libmpcodecs/libmpencoders.a \
-                $(ENCORE_LIB) \
-                $(COMMON_LIBS) \
-                $(EXTRA_LIB) \
-                $(MLIB_LIB) \
-                $(LIRC_LIB) \
-                $(LIRCC_LIB) \
-                $(ARCH_LIB) \
-                $(I18NLIBS) \
-                $(MATH_LIB) \
-                $(LIBC_LIB) \
-
-$(PRG_MENCODER): $(MENCODER_DEP)
-	$(CC) $(CFLAGS) -o $(PRG_MENCODER) $(OBJS_MENCODER) $(LIBS_MENCODER)
+$(PRG_MENCODER): $(PRG)
+	-ln -sf $(PRG) $(PRG_MENCODER)
 endif
 
 codecs.conf.h: $(PRG_CFG) etc/codecs.conf
@@ -453,6 +455,7 @@
 # help_mp.h is also required by a lot of files, so force generating it early.
 $(MPLAYER_DEP): version.h help_mp.h
 $(MENCODER_DEP): version.h help_mp.h
+$(DEP): version.h help_mp.h
 
 $(PRG_CFG): version.h codec-cfg.c codec-cfg.h help_mp.h
 	$(HOST_CC) $(HOST_CFLAGS) -I. codec-cfg.c -o $(PRG_CFG) \
@@ -478,7 +481,7 @@
 		fi ; \
 	done
 ifeq ($(MENCODER),yes)
-	$(INSTALL) -m 755 $(INSTALLSTRIP) $(PRG_MENCODER) $(BINDIR)/$(PRG_MENCODER)
+	-ln -sf $(PRG) $(BINDIR)/$(PRG_MENCODER)
 	for i in $(MAN_LANG); do \
 		if test "$$i" = en ; then \
 			ln -sf mplayer.1 $(MANDIR)/man1/mencoder.1 ; \
@@ -550,7 +553,7 @@
 
 depend: help_mp.h
 	./version.sh `$(CC) -dumpversion`
-	$(CC) -MM $(CFLAGS) -DCODECS2HTML mplayer.c mencoder.c $(SRCS_MPLAYER) $(SRCS_MENCODER) 1>.depend
+	$(CC) -MM $(CFLAGS) -DCODECS2HTML mplayer.c mencoder.c $(SRCS) 1>.depend
 	@for a in $(PARTS); do $(MAKE) -C $$a dep; done
 
 # ./configure must be run if it changed in CVS
Index: cfg-common.h
===================================================================
RCS file: /cvsroot/mplayer/main/cfg-common.h,v
retrieving revision 1.159
diff -u -r1.159 cfg-common.h
--- cfg-common.h	26 Mar 2006 11:09:17 -0000	1.159
+++ cfg-common.h	31 Mar 2006 16:44:41 -0000
@@ -358,6 +358,9 @@
 #endif
 
 #ifdef USE_TV
+#ifdef FOR_MENCODER
+extern m_option_t tvopts_conf[];
+#else
 m_option_t tvopts_conf[]={
 	{"on", "-tv on is deprecated, use tv:// instead.\n", CONF_TYPE_PRINT, 0, 0, 0, NULL},
 	{"immediatemode", &tv_param_immediate, CONF_TYPE_INT, CONF_RANGE, 0, 1, NULL},
@@ -403,6 +406,7 @@
 	{NULL, NULL, 0, 0, 0, 0, NULL}
 };
 #endif
+#endif
 
 #ifdef HAS_DVBIN_SUPPORT
 #include "libmpdemux/dvbin.h"
@@ -424,6 +428,9 @@
 extern float sws_chr_sharpen;
 extern float sws_lum_sharpen;
 
+#ifdef FOR_MENCODER
+extern m_option_t scaler_filter_conf[];
+#else
 m_option_t scaler_filter_conf[]={
 	{"lgb", &sws_lum_gblur, CONF_TYPE_FLOAT, 0, 0, 100.0, NULL},
 	{"cgb", &sws_chr_gblur, CONF_TYPE_FLOAT, 0, 0, 100.0, NULL},
@@ -433,6 +440,7 @@
 	{"cs", &sws_chr_sharpen, CONF_TYPE_FLOAT, 0, -100.0, 100.0, NULL},
 	{NULL, NULL, 0, 0, 0, 0, NULL}
 };
+#endif
 
 /* VIVO demuxer options: */
 extern int vivo_param_version;
@@ -445,6 +453,9 @@
 extern int vivo_param_vformat;
 extern char *dvd_device, *cdrom_device;
 
+#ifdef FOR_MENCODER
+extern m_option_t vivoopts_conf[];
+#else
 m_option_t vivoopts_conf[]={
 	{"version", &vivo_param_version, CONF_TYPE_INT, 0, 0, 0, NULL},
 	/* audio options */
@@ -458,6 +469,7 @@
 	{"vformat", &vivo_param_vformat, CONF_TYPE_INT, 0, 0, 0, NULL},
 	{NULL, NULL, 0, 0, 0, 0, NULL}
 };
+#endif
 
 extern int    mf_w;
 extern int    mf_h;
@@ -466,6 +478,9 @@
 extern m_obj_settings_t* vf_settings;
 extern m_obj_list_t vf_obj_list;
 
+#ifdef FOR_MENCODER
+extern m_option_t mfopts_conf[];
+#else
 m_option_t mfopts_conf[]={
         {"on", "-mf on is deprecated, use mf:// instead.\n", CONF_TYPE_PRINT, 0, 0, 1, NULL},
         {"w", &mf_w, CONF_TYPE_INT, 0, 0, 0, NULL},
@@ -474,17 +489,26 @@
         {"type", &mf_type, CONF_TYPE_STRING, 0, 0, 0, NULL},
         {NULL, NULL, 0, 0, 0, 0, NULL}
 };
+#endif
 						
 extern m_obj_settings_t* vo_plugin_args;
 
 #include "libaf/af.h"
 extern af_cfg_t af_cfg; // Audio filter configuration, defined in libmpcodecs/dec_audio.c
+
+#ifdef FOR_MENCODER
+extern m_option_t audio_filter_conf[];
+#else
 m_option_t audio_filter_conf[]={       
 	{"list", &af_cfg.list, CONF_TYPE_STRING_LIST, 0, 0, 0, NULL},
         {"force", &af_cfg.force, CONF_TYPE_INT, CONF_RANGE, 0, 7, NULL},
 	{NULL, NULL, 0, 0, 0, 0, NULL}
 };
+#endif
 
+#ifdef FOR_MENCODER
+extern m_option_t msgl_config[];
+#else
 m_option_t msgl_config[]={
 	{ "all", &mp_msg_level_all, CONF_TYPE_INT, CONF_RANGE, -1, 9, NULL},
 
@@ -573,6 +597,7 @@
         "   muxer      - muxer layer\n"
         "\n", CONF_TYPE_PRINT, CONF_NOCFG, 0, 0, NULL},
 };
+#endif
 
 #ifdef WIN32
 
Index: cfg-mencoder.h
===================================================================
RCS file: /cvsroot/mplayer/main/cfg-mencoder.h,v
retrieving revision 1.107
diff -u -r1.107 cfg-mencoder.h
--- cfg-mencoder.h	19 Feb 2006 09:34:36 -0000	1.107
+++ cfg-mencoder.h	31 Mar 2006 16:44:43 -0000
@@ -3,7 +3,9 @@
  * config for cfgparser
  */
 
+#define FOR_MENCODER 1
 #include "cfg-common.h"
+#undef FOR_MENCODER
 
 #ifdef USE_FAKE_MONO
 extern int fakemono; // defined in dec_audio.c
Index: mencoder.c
===================================================================
RCS file: /cvsroot/mplayer/main/mencoder.c,v
retrieving revision 1.346
diff -u -r1.346 mencoder.c
--- mencoder.c	24 Mar 2006 08:12:01 -0000	1.346
+++ mencoder.c	31 Mar 2006 16:44:46 -0000
@@ -74,7 +74,7 @@
 
 #include "osdep/timer.h"
 
-#include "get_path.c"
+extern char *get_path(char *filename);
 
 #ifdef USE_LIBAVCODEC
 #ifdef USE_LIBAVCODEC_SO
@@ -85,30 +85,27 @@
 #endif
 
 #include "libmpcodecs/ae.h"
-int vo_doublebuffering=0;
-int vo_directrendering=0;
-int vo_config_count=0;
-int forced_subs_only=0;
+extern int forced_subs_only;
 
 //--------------------------
 
 // cache2:
-int stream_cache_size=-1;
+extern int stream_cache_size;
 #ifdef USE_STREAM_CACHE
 extern int cache_fill_status;
 
-float stream_cache_min_percent=20.0;
-float stream_cache_seek_min_percent=50.0;
+extern float stream_cache_min_percent;
+extern float stream_cache_seek_min_percent;
 #else
 #define cache_fill_status 0
 #endif
 
-int audio_id=-1;
-int video_id=-1;
-int dvdsub_id=-2;
-int vobsub_id=-1;
-char* audio_lang=NULL;
-char* dvdsub_lang=NULL;
+extern int audio_id;
+extern int video_id;
+extern int dvdsub_id;
+extern int vobsub_id;
+extern char* audio_lang;
+extern char* dvdsub_lang;
 static char* spudec_ifo=NULL;
 
 static char** audio_codec_list=NULL;  // override audio codec
@@ -130,15 +127,15 @@
 //void resync_audio_stream(sh_audio_t *sh_audio){}
 
 extern int verbose; // must be global!
-int identify=0;
-int quiet=0;
-double video_time_usage=0;
-double vout_time_usage=0;
+extern int identify;
+extern int quiet;
+extern double video_time_usage;
+extern double vout_time_usage;
 double max_video_time_usage=0;
 double max_vout_time_usage=0;
 double cur_video_time_usage=0;
 double cur_vout_time_usage=0;
-int benchmark=0;
+extern int benchmark;
 
 extern int mp_msg_levels[MSGT_MAX];
 extern int mp_msg_level_all;
@@ -159,10 +156,10 @@
 static int ignore_start=0;
 static int audio_density=2;
 
-float force_fps=0;
+extern float force_fps;
 static float force_ofps=0; // set to 24 for inverse telecine
 static int skip_limit=-1;
-float playback_speed=1.0;
+extern float playback_speed;
 
 static int force_srate=0;
 static int audio_output_format=0;
@@ -185,21 +182,21 @@
 #include "libvo/sub.h"
 
 // sub:
-char *font_name=NULL;
+extern char *font_name;
 #ifdef HAVE_FONTCONFIG
 extern int font_fontconfig;
 #endif
-float font_factor=0.75;
-char **sub_name=NULL;
-float sub_delay=0;
-float sub_fps=0;
-int   sub_auto = 0;
-int   subcc_enabled=0;
-int   suboverlap_enabled = 1;
+extern float font_factor;
+extern char **sub_name;
+extern float sub_delay;
+extern float sub_fps;
+extern int   sub_auto;
+extern int   subcc_enabled;
+extern int   suboverlap_enabled;
 
 #ifdef USE_SUB
 static sub_data* subdata=NULL;
-float sub_last_pts = -303;
+extern float sub_last_pts;
 #endif
 
 int auto_expand=1;
@@ -297,7 +294,7 @@
     exit(level);
 }
 
-void parse_cfgfiles( m_config_t* conf )
+void me_parse_cfgfiles( m_config_t* conf )
 {
   char *conffile;
   if ((conffile = get_path("mencoder.conf")) == NULL) {
@@ -358,7 +355,7 @@
 
 extern void print_wave_header(WAVEFORMATEX *h);
 
-int main(int argc,char* argv[]){
+int me_main(int argc,char* argv[]){
 
 stream_t* stream=NULL;
 demuxer_t* demuxer=NULL;
@@ -403,6 +400,7 @@
 
 audio_encoding_params_t aparams;
 audio_encoder_t *aencoder = NULL;
+ sub_auto = 0;
 
   mp_msg_init();
   mp_msg(MSGT_CPLAYER,MSGL_INFO, "MEncoder " VERSION " (C) 2000-2006 MPlayer Team\n");
@@ -461,7 +459,7 @@
 
  mconfig = m_config_new();
  m_config_register_options(mconfig,mencoder_opts);
- parse_cfgfiles(mconfig);
+ me_parse_cfgfiles(mconfig);
  filelist = m_config_parse_me_command_line(mconfig, argc, argv);
  if(!filelist) mencoder_exit(1, MSGTR_ErrorParsingCommandLine);
 
Index: mplayer.c
===================================================================
RCS file: /cvsroot/mplayer/main/mplayer.c,v
retrieving revision 1.923
diff -u -r1.923 mplayer.c
--- mplayer.c	24 Mar 2006 20:42:23 -0000	1.923
+++ mplayer.c	31 Mar 2006 16:44:56 -0000
@@ -2154,7 +2154,7 @@
     return 1;
 }
 
-int main(int argc,char* argv[]){
+int mp_main(int argc,char* argv[]){
 
 
 char * mem_ptr;
@@ -4990,3 +4990,14 @@
 
 return 1;
 }
+
+int
+main (int argc, char **argv)
+{
+#ifdef HAVE_MENCODER
+  if (strstr (argv[0], "mencoder"))
+    return me_main (argc, argv);
+  else
+#endif
+    return mp_main (argc, argv);
+}
