--- libtool-1.4.2/tests/Makefile.am.orig	2002-09-16 21:33:44.000000000 -0400
+++ libtool-1.4.2/tests/Makefile.am	2002-09-16 21:34:19.000000000 -0400
@@ -19,7 +19,7 @@
 	demo-nofast.test demo-make.test demo-exec.test \
 	demo-inst.test demo-unst.test \
 	demo-pic.test demo-make.test demo-exec.test \
-	demo-nopic.test demo-make.test demo-exec.test \
+	demo-make.test demo-exec.test \
 	depdemo-nofast.test depdemo-make.test \
 	depdemo-exec.test depdemo-inst.test depdemo-unst.test \
 	cdemo-shared.test cdemo-make.test cdemo-exec.test \
