--- a/gmic-qt/src/Widgets/SearchFieldWidget.h
+++ b/gmic-qt/src/Widgets/SearchFieldWidget.h
@@ -26,6 +26,7 @@
 #define _GMIC_QT_SEARCHFIELDWIDGET_H_
 #include <QIcon>
 #include <QWidget>
+#include <QAction>
 
 namespace Ui
 {
--- a/zart/include/DialogAbout.h
+++ b/zart/include/DialogAbout.h
@@ -47,6 +47,7 @@
 #define _DIALOGABOUT_H_
 
 #include <QDialog>
+#include <QButtonGroup>
 #include "ui_DialogAbout.h"
 
 /**
