Index: common/video/Makefile.am
===================================================================
RCS file: /cvsroot/mpeg4ip/mpeg4ip/common/video/Makefile.am,v
retrieving revision 1.13
diff -u -r1.13 Makefile.am
--- common/video/Makefile.am	9 Nov 2004 00:19:45 -0000	1.13
+++ common/video/Makefile.am	6 May 2005 22:51:37 -0000
@@ -1 +1 @@
-SUBDIRS = iso-mpeg4
+SUBDIRS = #iso-mpeg4 
Index: player/src/codec/Makefile.am
===================================================================
RCS file: /cvsroot/mpeg4ip/mpeg4ip/player/src/codec/Makefile.am,v
retrieving revision 1.3
diff -u -r1.3 Makefile.am
--- player/src/codec/Makefile.am	1 May 2002 17:41:53 -0000	1.3
+++ player/src/codec/Makefile.am	6 May 2005 22:51:45 -0000
@@ -1 +1 @@
-SUBDIRS = aa mp3 mpeg4 wav
+SUBDIRS = aa mp3 wav
