--- a/src/rules.mk
+++ b/src/rules.mk
@@ -307,8 +307,8 @@ $(STATICOTFS): %.otf: $(BUILDDIR)/%-subr
 
 forceiftagchange = $(shell cmp -s $@ - <<< "$(GitVersion)" || echo force)
 $(BUILDDIR)/last-commit: $$(forceiftagchange) | $(BUILDDIR)
-	git update-index --refresh --ignore-submodules ||:
-	git diff-index --quiet --cached HEAD -- $(SOURCES)
+	-git update-index --refresh --ignore-submodules ||:
+	-git diff-index --quiet --cached HEAD -- $(SOURCES)
 	echo $(GitVersion) > $@
 
 DISTDIR = $(PROJECT)-$(GitVersion)
