diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.4.3.orig/glib/Makefile.am poppler-0.4.3/glib/Makefile.am
--- poppler-0.4.3.orig/glib/Makefile.am	2005-05-16 17:56:31.000000000 -0400
+++ poppler-0.4.3/glib/Makefile.am	2006-01-03 12:27:24.000000000 -0500
@@ -76,7 +76,7 @@
 	poppler-private.h
 
 libpoppler_glib_la_LIBADD =			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(POPPLER_GLIB_LIBS)			\
 	$(FREETYPE_LIBS)			\
 	$(cairo_libs)
@@ -87,7 +87,7 @@
        test-poppler-glib.c
 
 test_poppler_glib_LDADD =			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-glib.la			\
 	$(POPPLER_GLIB_LIBS)			\
 	$(FREETYPE_LIBS)			\
diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.4.3.orig/Makefile.am poppler-0.4.3/Makefile.am
--- poppler-0.4.3.orig/Makefile.am	2005-06-28 06:00:09.000000000 -0400
+++ poppler-0.4.3/Makefile.am	2006-01-03 14:49:51.000000000 -0500
@@ -22,7 +22,7 @@
 utils_subdir = utils
 endif
 
-SUBDIRS = goo fofi $(splash_subdir) poppler $(utils_subdir) $(glib_subdir) $(qt_subdir) test $(qt4_subdir)
+SUBDIRS = $(glib_subdir) $(qt_subdir) $(qt4_subdir)
 
 EXTRA_DIST =					\
 	README-XPDF				\
@@ -33,9 +33,6 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA =				\
-	poppler.pc				\
-	$(cairo_pc_file)			\
-	$(splash_pc_file)			\
 	$(glib_pc_file)				\
 	$(qt_pc_file)                           \
 	$(qt4_pc_file)
diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.4.3.orig/qt/Makefile.am poppler-0.4.3/qt/Makefile.am
--- poppler-0.4.3.orig/qt/Makefile.am	2005-12-12 17:25:02.000000000 -0500
+++ poppler-0.4.3/qt/Makefile.am	2006-01-03 12:27:39.000000000 -0500
@@ -18,7 +18,7 @@
 	poppler-private.h
 
 libpoppler_qt_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)
 
@@ -32,7 +32,7 @@
        test-poppler-qt.cpp
 
 test_poppler_qt_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	libpoppler-qt.la			\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)
diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.4.3.orig/qt4/src/Makefile.am poppler-0.4.3/qt4/src/Makefile.am
--- poppler-0.4.3.orig/qt4/src/Makefile.am	2005-06-28 06:00:09.000000000 -0400
+++ poppler-0.4.3/qt4/src/Makefile.am	2006-01-03 12:28:08.000000000 -0500
@@ -17,6 +17,6 @@
 	poppler-private.h
 
 libpoppler_qt4_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(POPPLER_QT4_LIBS)
 
diff --exclude-from=/home/dang/.diffrc -u -ruN poppler-0.4.3.orig/qt4/tests/Makefile.am poppler-0.4.3/qt4/tests/Makefile.am
--- poppler-0.4.3.orig/qt4/tests/Makefile.am	2005-07-07 17:15:09.000000000 -0400
+++ poppler-0.4.3/qt4/tests/Makefile.am	2006-01-03 12:28:28.000000000 -0500
@@ -5,7 +5,7 @@
 	$(POPPLER_QT4_CXXFLAGS)
 
 LDADDS =					\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler				\
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(POPPLER_QT4_LIBS)
 
