Index: libao/audio_out_wav.c
===================================================================
RCS file: /cvsroot/liba52/a52dec/libao/audio_out_wav.c,v
retrieving revision 1.9
diff -u -r1.9 audio_out_wav.c
--- libao/audio_out_wav.c	24 Jan 2003 10:52:59 -0000	1.9
+++ libao/audio_out_wav.c	1 Aug 2006 11:09:14 -0000
@@ -133,12 +133,13 @@
     if (instance->set_params) {
 	instance->set_params = 0;
 	instance->speaker_flags = speaker_flags;
-	if (speaker_flags == 3 || speaker_flags == 4) {
+//	if (speaker_flags == 3 || speaker_flags == 4) {
 	    store2 (wav_header + 22, chans);
 	    store4 (wav_header + 24, instance->sample_rate);
 	    store4 (wav_header + 28, instance->sample_rate * 2 * chans);
 	    store2 (wav_header + 32, 2 * chans);
 	    fwrite (wav_header, sizeof (wav_header), 1, stdout);
+#if 0
 	} else {
 	    store2 (wav6_header + 22, chans);
 	    store4 (wav6_header + 24, instance->sample_rate);
@@ -147,6 +148,7 @@
 	    store4 (wav6_header + 40, speaker_flags);
 	    fwrite (wav6_header, sizeof (wav6_header), 1, stdout);
 	}
+#endif
     } else if (speaker_flags != instance->speaker_flags)
 	return 1;
 
@@ -167,15 +169,17 @@
     if (fseek (stdout, 0, SEEK_SET) < 0)
 	return;
 
-    if (instance->speaker_flags == 3 || instance->speaker_flags == 4) {
+//    if (instance->speaker_flags == 3 || instance->speaker_flags == 4) {
 	store4 (wav_header + 4, instance->size + 36);
 	store4 (wav_header + 40, instance->size);
 	fwrite (wav_header, sizeof (wav_header), 1, stdout);
+#if 0
     } else {
 	store4 (wav6_header + 4, instance->size + 60);
 	store4 (wav6_header + 64, instance->size);
 	fwrite (wav6_header, sizeof (wav6_header), 1, stdout);
     }
+#endif
 }
 
 static ao_instance_t * wav_open (int flags)
