Index: mk/objective.mk
===================================================================
--- mk/objective.mk	(revision 685)
+++ mk/objective.mk	(working copy)
@@ -38,6 +38,7 @@
 	@if test "$(OBJECTIVE_BINS)" != "none"; then \
 		for i in $(OBJECTIVE_BINS); do \
 			printf "%10s     %-20s\n" INSTALL $$i; \
+			$(INSTALL) $(INSTALL_OVERRIDE) -d $(DESTDIR)/$(BINDIR); \
 			$(INSTALL) $(INSTALL_OVERRIDE) $$i $(DESTDIR)/$(BINDIR)/$$i; \
 		done; \
 	fi;
