Index: configure
===================================================================
RCS file: /cvsroot/audacity/audacity-src/configure,v
retrieving revision 1.87
diff -u -r1.87 configure
--- configure	14 Feb 2006 08:12:25 -0000	1.87
+++ configure	2 Jun 2006 13:23:07 -0000
@@ -5598,11 +5598,6 @@
 fi
 
 
-if [ "$use_portmixer" = "yes" ] && [ "$use_portaudio" = "v19" ] ; then
-   { { echo "$as_me:$LINENO: error: *** PortAudio v19 does not work with PortMixer (at least not yet) " >&5
-echo "$as_me: error: *** PortAudio v19 does not work with PortMixer (at least not yet) " >&2;}
-   { (exit 1); exit 1; }; }
-fi
 
 
 
