diff -Naurp libdts-0.0.2.orig/configure.ac libdts-0.0.2/configure.ac
--- libdts-0.0.2.orig/configure.ac	2004-02-29 04:00:32.000000000 -0800
+++ libdts-0.0.2/configure.ac	2005-03-26 03:11:21.000000000 -0800
@@ -77,10 +77,10 @@ else
     esac
 fi
 
-#libtool# dnl Checks for libtool - this must be done after we set cflags
-#libtool# AC_DISABLE_SHARED
-#libtool# AC_LIBTOOL_WIN32_DLL
-#libtool# AC_PROG_LIBTOOL
+dnl Checks for libtool - this must be done after we set cflags
+AC_DISABLE_SHARED
+AC_LIBTOOL_WIN32_DLL
+AC_PROG_LIBTOOL
 
 dnl Build the PIC library? - to be removed once libtool is re-enabled
 AC_PROG_RANLIB
diff -Naurp libdts-0.0.2.orig/libdts/Makefile.am libdts-0.0.2/libdts/Makefile.am
--- libdts-0.0.2.orig/libdts/Makefile.am	2004-02-25 05:52:15.000000000 -0800
+++ libdts-0.0.2/libdts/Makefile.am	2005-03-26 03:05:59.000000000 -0800
@@ -1,22 +1,23 @@
 AM_CFLAGS = $(OPT_CFLAGS) $(LIBDTS_CFLAGS)
 
-#libtool# lib_LTLIBRARIES = libdts.la
-#libtool# 
-#libtool# libdts_la_SOURCES = bitstream.c parse.c downmix.c
-#libtool# libdts_la_LIBADD = $(LIBDTS_LIBS)
-#libtool# libdts_la_LDFLAGS = -no-undefined
+lib_LTLIBRARIES = libdts.la
 
-lib_LIBRARIES = libdts.a $(libdts_pic_a)
-libdts_a_SOURCES = bitstream.c parse.c downmix.c
+libdts_la_SOURCES = bitstream.c parse.c downmix.c
+libdts_la_LIBADD = $(LIBDTS_LIBS)
+libdts_la_LDFLAGS = -no-undefined
+
+#lib_LIBRARIES = libdts.a $(libdts_pic_a)
+#libdts_a_SOURCES = bitstream.c parse.c downmix.c
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libdts.pc
 
 EXTRA_DIST = configure.incl dts_internal.h bitstream.h tables.h tables_quantization.h tables_huffman.h tables_fir.h tables_adpcm.h tables_vq.h
 
-if NEED_PIC
-libdts_pic_a = libdts_pic.a
-endif
-libdts_pic_a_SOURCES = $(libdts_a_SOURCES)
-libdts_pic_a_CPPFLAGS = -fPIC -DPIC
+if NEED_PIC
+#libdts_pic_a = libdts_pic.a
+#endif
+#libdts_pic_a_SOURCES = $(libdts_a_SOURCES)
+libdts_la_CPPFLAGS = -fPIC -DPIC
+endif
 
diff -Naurp libdts-0.0.2.orig/src/Makefile.am libdts-0.0.2/src/Makefile.am
--- libdts-0.0.2.orig/src/Makefile.am	2004-02-25 05:52:16.000000000 -0800
+++ libdts-0.0.2/src/Makefile.am	2005-03-26 03:06:30.000000000 -0800
@@ -2,10 +2,10 @@ AM_CFLAGS = $(DTSDEC_CFLAGS)
 
 bin_PROGRAMS = dtsdec extract_dts
 dtsdec_SOURCES = dtsdec.c getopt.c gettimeofday.c
-#libtool# dtsdec_LDADD = $(top_builddir)/libdts/libdts.la \
-#libtool# 	       $(top_builddir)/libao/libao.a $(LIBAO_LIBS)
-dtsdec_LDADD = $(top_builddir)/libdts/libdts.a \
-	       $(top_builddir)/libao/libao.a $(LIBAO_LIBS)
+dtsdec_LDADD = $(top_builddir)/libdts/libdts.la \
+ 	       $(top_builddir)/libao/libao.a $(LIBAO_LIBS)
+#dtsdec_LDADD = $(top_builddir)/libdts/libdts.a \
+#	       $(top_builddir)/libao/libao.a $(LIBAO_LIBS)
 extract_dts_SOURCES = extract_dts.c getopt.c
 
 man_MANS = dtsdec.1 extract_dts.1
