diff -ruNd libdts-0.0.2/libdts/bitstream.c libdts-svn/libdts/bitstream.c
--- libdts-0.0.2/libdts/bitstream.c	2004-02-24 22:37:15.000000000 +0200
+++ libdts-svn/libdts/bitstream.c	2004-07-18 02:20:28.000000000 +0300
@@ -38,7 +38,7 @@
     intptr_t align;
 
     align = (uintptr_t)buf & 3;
-    state->buffer_start = (uint32_t *) (buf - align);
+    state->buffer_start = (uint32_t *) ((uintptr_t)buf - align);
     state->bits_left = 0;
     state->current_word = 0;
     state->word_mode = word_mode;
diff -ruNd libdts-0.0.2/libdts/parse.c libdts-svn/libdts/parse.c
--- libdts-0.0.2/libdts/parse.c	2004-02-29 13:33:53.000000000 +0200
+++ libdts-svn/libdts/parse.c	2004-07-18 02:20:28.000000000 +0300
@@ -440,7 +440,7 @@
     return 0;
 }
 
-int dts_subframe_header (dts_state_t * state)
+static int dts_subframe_header (dts_state_t * state)
 {
     /* Primary audio coding side information */
     int j, k;
@@ -754,7 +754,7 @@
     return 0;
 }
 
-int dts_subsubframe (dts_state_t * state)
+static int dts_subsubframe (dts_state_t * state)
 {
     int k, l;
     int subsubframe = state->current_subsubframe;
@@ -979,8 +979,8 @@
     /* 32 subbands QMF */
     for (k = 0; k < state->prim_channels; k++)
     {
-        static double pcm_to_float[8] =
-            {32768.0, 32768.0, 524288.0, 524288.0, 0, 8388608.0, 8388608.0};
+        /*static double pcm_to_float[8] =
+            {32768.0, 32768.0, 524288.0, 524288.0, 0, 8388608.0, 8388608.0};*/
 
         qmf_32_subbands (state, k,
                          subband_samples[k],
@@ -1017,7 +1017,7 @@
     return 0;
 }
 
-int dts_subframe_footer (dts_state_t * state)
+static int dts_subframe_footer (dts_state_t * state)
 {
     int aux_data_count = 0, i;
     int lfe_samples;
diff -ruNd libdts-0.0.2/libdts/tables.h libdts-svn/libdts/tables.h
--- libdts-0.0.2/libdts/tables.h	2004-02-08 21:54:07.000000000 +0200
+++ libdts-svn/libdts/tables.h	2004-07-18 02:20:25.000000000 +0300
@@ -41,7 +41,7 @@
     1, 2, 2, 2, 2, 3, 3, 4, 4, 5, 6, 6, 6, 7, 8, 8
 };
 
-static uint8_t dts_bits_per_sample[] =
+static const uint8_t dts_bits_per_sample[] =
 {
     16, 16, 20, 20, 0, 24, 24
 };
diff -ruNd libdts-0.0.2/libdts/tables_huffman.h libdts-svn/libdts/tables_huffman.h
--- libdts-0.0.2/libdts/tables_huffman.h	2004-02-08 21:54:42.000000000 +0200
+++ libdts-svn/libdts/tables_huffman.h	2004-07-18 02:20:26.000000000 +0300
@@ -1045,7 +1045,7 @@
     bitalloc_e_129, bitalloc_f_129, bitalloc_g_129, 0 },
 };
 
-int InverseQ( dts_state_t * state, huff_entry_t * huff )
+static int InverseQ( dts_state_t * state, huff_entry_t * huff )
 {
     int value = 0;
     int length = 0, j;
