--- a/flowblade
+++ b/flowblade
@@ -22,6 +22,7 @@
 
 import os
 import sys
+import site
 
 print "FLOWBLADE MOVIE EDITOR repository 1.7"
 print "-------------------------------------"
@@ -43,6 +44,7 @@ else:
     print "Running from filesystem..."
     modules_path = launch_dir + "/Flowblade"
 
+modules_path = site.getsitepackages()[0] + "/Flowblade"
 sys.path.insert(0, modules_path)
 sys.path.insert(0, modules_path + "/vieweditor")
 sys.path.insert(0, modules_path + "/tools")
