Index: configure.ac
===================================================================
--- configure.ac	(revision 903)
+++ configure.ac	(working copy)
@@ -116,7 +116,16 @@
 AC_SUBST(SAMPLERATE_CFLAGS)
 AC_SUBST(SAMPLERATE_LIBS)
 
+PKG_CHECK_MODULES(XML2, libxml-2.0,
+            ac_cv_libxml2=1, ac_cv_libxml2=0)
+AC_SUBST(XML2_CFLAGS)
+AC_SUBST(XML2_LIBS)
 
+PKG_CHECK_MODULES(PORTAUDIO, portaudio-2.0 >= 19,
+            ac_cv_portaudio=1, ac_cv_portaudio=0)
+AC_SUBST(PORTAUDIO_CFLAGS)
+AC_SUBST(PORTAUDIO_LIBS)
+
 # Checks for typedefs, structures, and compiler characteristics.
 AC_HEADER_STDBOOL
 AC_C_CONST
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 903)
+++ src/Makefile.am	(working copy)
@@ -21,9 +21,10 @@
 #	portaudio
 
 
-DEPS_LIBS = libs/WiredWidgets/src/.libs/libWiredWidgets.so libs/WiredAkai/src/.libs/libWiredAkai.so
+DEPS_LIBS = -Llibs/WiredWidgets/src -lWiredWidgets -Llibs/WiredAkai/src -lWiredAkai
 
-wired_LDFLAGS = $(shell wx-config --libs) -lsndfile -lasound -Lmidi/portmidi $(DEPS_LIBS) -Lfileloader -lfileloader -L/usr/pkg/lib/ `xml2-config --libs` @SAMPLERATE_LIBS@ -lportaudio
+wired_LDFLAGS = -Lmidi/portmidi -Lfileloader
+EXTRA_LIBS = $(DEPS_LIBS) -lfileloader -lportaudio -lsndfile -lasound @XML2_LIBS@ @SAMPLERATE_LIBS@
 
 wired_CPPFLAGS = $(shell wx-config --cflags) -Igui -Iplugins -Ifileloader -I/opt/gnome/include -Isequencer -Iredist -Iaudio -Iengine -Imixer -Imidi -Imidi/portmidi/pm_common -Imidi/portmidi/porttime -I. -Iasm -Ieditmidi -Iundo -I.  -I/usr/pkg/include/libxml2/ -I/usr/pkg/include `xml2-config --cflags` @SAMPLERATE_CFLAGS@
 
@@ -49,4 +49,5 @@
 	samplerate/libwiredsamplerate.a		\
 	codec/libwiredCodec.a			\
 	wiredvideo/libwiredvideo.a		\
-	dssi/libwireddssi.a
+	dssi/libwireddssi.a \
+	$(EXTRA_LIBS)
