diff --git a/sources/Montserrat-Italic.glyphs b/sources/Montserrat-Italic.glyphs
index de74777..a8a6376 100644
--- a/sources/Montserrat-Italic.glyphs
+++ b/sources/Montserrat-Italic.glyphs
@@ -135371,7 +135371,7 @@ nodes = (
 );
 };
 layerId = "9ABB2ABC-5A7C-4B76-9DAB-8AF6A3B474BD";
-name = "{-170-} Sep 19 17, 22:59";
+name = "-170- Sep 19 17, 22:59";
 paths = (
 {
 closed = 1;
diff --git a/sources/Montserrat.glyphs b/sources/Montserrat.glyphs
index 7523f01..58f6c11 100644
--- a/sources/Montserrat.glyphs
+++ b/sources/Montserrat.glyphs
@@ -130583,7 +130583,7 @@ nodes = (
 );
 };
 layerId = "9ABB2ABC-5A7C-4B76-9DAB-8AF6A3B474BD";
-name = "{-170-} Sep 19 17, 22:59";
+name = "-170- Sep 19 17, 22:59";
 paths = (
 {
 closed = 1;
