diff --git a/avidemux/qt4/CMakeLists.txt b/avidemux/qt4/CMakeLists.txt
index 0fe076e..6759de8 100644
--- a/avidemux/qt4/CMakeLists.txt
+++ b/avidemux/qt4/CMakeLists.txt
@@ -148,6 +148,11 @@ IF (GETTEXT_FOUND)
 	TARGET_LINK_LIBRARIES(avidemux3_qt4 ${GETTEXT_LIBRARY_DIR})
 ENDIF (GETTEXT_FOUND)
 
+IF (GETTEXT)
+	INCLUDE(Ts)
+	COMPILE_TS_FILES(${AVIDEMUX_TOP_SOURCE_DIR}/po QM_FILES)
+	ADD_CUSTOM_TARGET(qmfiles ALL DEPENDS ${QM_FILES})
+ENDIF (GETTEXT)
 
 # SDL
 IF (USE_SDL)
diff --git a/cmake/Ts.cmake b/cmake/Ts.cmake
index 7468323..420baa3 100644
--- a/cmake/Ts.cmake
+++ b/cmake/Ts.cmake
@@ -34,7 +34,7 @@ MACRO(INSTALL_I18N _files)
         IF(WIN32)
                 SET(i18dir "${CMAKE_INSTALL_PREFIX}/${BIN_DIR}/i18n")
         ELSE(WIN32)
-                SET(i18dir "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
+                SET(i18dir "${CMAKE_INSTALL_PREFIX}/share/avidemux6/i18n")
         ENDIF(WIN32)
         INSTALL(FILES ${_files} DESTINATION "${i18dir}")
 ENDMACRO(INSTALL_I18N _files)
