Index: configure.ac
===================================================================
--- configure.ac	(revision 302)
+++ configure.ac	(working copy)
@@ -53,6 +53,12 @@
                        gstreamer-0.10 >= $GSTREAMER_REQUIRED
                        gnome-keyring-1 >= $GNOME_KEYRING_REQUIRED)
 
+# Check for libnotify >= 0.7
+PKG_CHECK_MODULES(LIBNOTIFY_07, [libnotify >= 0.7], [have_libnotify_07=yes],[have_libnotify_07=no])
+if test x"$have_libnotify_07" = "xyes"; then
+    AC_DEFINE(HAVE_LIBNOTIFY_07, 1, [Define if you have libnotify 0.7 or later])
+fi
+
 AC_SUBST(GGN_CFLAGS)
 AC_SUBST(GGN_LIBS)
 
Index: src/ggn-icon.c
===================================================================
--- src/ggn-icon.c	(revision 302)
+++ src/ggn-icon.c	(working copy)
@@ -571,12 +571,16 @@
   /* create a new notification. */
   icon->priv->note = notify_notification_new (icon->priv->title,
                                               icon->priv->summary,
+  #if HAVE_LIBNOTIFY_07
+                                              icon->priv->icon_file);
+  #else
                                               icon->priv->icon_file,
                                               NULL);
 
   /* attach the notification to our icon. */
   notify_notification_attach_to_status_icon (icon->priv->note,
                                              icon->priv->icon);
+  #endif
 
   /* set the notification timeout. */
   notify_notification_set_timeout (icon->priv->note, (4 * 1000));
