diff -r 6d83826a4de6 Makefile.am
--- a/Makefile.am	Sun Oct 25 17:39:10 2009 +0100
+++ b/Makefile.am	Tue Oct 27 21:59:46 2009 +0200
@@ -30,5 +30,5 @@
 
 EXTRA_DIST = STYLE transcode.spec
 
-docsdir = $(datadir)/doc/transcode
+docsdir = @docdir@
 docs_DATA = ChangeLog
diff -r 6d83826a4de6 docs/Makefile.am
--- a/docs/Makefile.am	Sun Oct 25 17:39:10 2009 +0100
+++ b/docs/Makefile.am	Tue Oct 27 21:59:46 2009 +0200
@@ -15,7 +15,7 @@
 	html \
 	tech
 
-docsdir = $(datadir)/doc/transcode
+docsdir = @docdir@
 
 docs_DATA = \
 	DV-to-DVD-HOWTO.txt \
diff -r 6d83826a4de6 docs/html/Makefile.am
--- a/docs/html/Makefile.am	Sun Oct 25 17:39:10 2009 +0100
+++ b/docs/html/Makefile.am	Tue Oct 27 21:59:46 2009 +0200
@@ -2,7 +2,7 @@
 	$(docs_DATA)
 
 docsdir = \
-	$(datadir)/doc/transcode/html
+	@docdir@/html
 
 docs_DATA =  \
 	dv.html \
