Index: encode/Makefile.am
===================================================================
RCS file: /cvstc/transcode/encode/Makefile.am,v
retrieving revision 1.17
diff -u -r1.17 Makefile.am
--- encode/Makefile.am	27 Jan 2008 09:49:38 -0000	1.17
+++ encode/Makefile.am	24 Mar 2008 15:29:18 -0000
@@ -83,7 +83,7 @@
 encode_im_la_LIBADD = $(IMAGEMAGICK_LIBS) -lm
 
 encode_lavc_la_SOURCES = encode_lavc.c
-encode_lavc_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVCODEC_CFLAGS)
+encode_lavc_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS)
 encode_lavc_la_LDFLAGS = -module -avoid-version
 encode_lavc_la_LIBADD = $(LIBAVCODEC_LIBS)
 
Index: export/Makefile.am
===================================================================
RCS file: /cvstc/transcode/export/Makefile.am,v
retrieving revision 1.68
diff -u -r1.68 Makefile.am
--- export/Makefile.am	22 Mar 2008 11:09:33 -0000	1.68
+++ export/Makefile.am	24 Mar 2008 15:29:26 -0000
@@ -70,12 +70,12 @@
 export_ac3_la_LDFLAGS = -module -avoid-version
 
 export_divx5_la_SOURCES = export_divx5.c aud_aux.c divx4_vbr.c
-export_divx5_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_divx5_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_divx5_la_LDFLAGS = -module -avoid-version
 export_divx5_la_LIBADD = $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
 export_dv_la_SOURCES = export_dv.c aud_aux.c
-export_dv_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBDV_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_dv_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBDV_CFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_dv_la_LDFLAGS = -module -avoid-version
 export_dv_la_LIBADD = $(LIBDV_LIBS) $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
@@ -85,7 +85,7 @@
 export_dvraw_la_LIBADD = $(LIBDV_LIBS) -lm
 
 export_ffmpeg_la_SOURCES = export_ffmpeg.c aud_aux.c ffmpeg_cfg.c
-export_ffmpeg_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_ffmpeg_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_ffmpeg_la_LDFLAGS = -module -avoid-version
 export_ffmpeg_la_LIBADD = $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
@@ -103,7 +103,7 @@
 export_lame_la_LDFLAGS = -module -avoid-version
 
 export_lzo_la_SOURCES = export_lzo.c aud_aux.c
-export_lzo_la_CPPFLAGS = $(AM_CPPFLAGS) $(LZO_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_lzo_la_CPPFLAGS = $(AM_CPPFLAGS) $(LZO_CFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_lzo_la_LDFLAGS = -module -avoid-version
 export_lzo_la_LIBADD = $(LZO_LIBS) $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
@@ -135,7 +135,7 @@
 export_pcm_la_LDFLAGS = -module -avoid-version
 
 export_ppm_la_SOURCES = export_ppm.c aud_aux.c
-export_ppm_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_ppm_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_ppm_la_LDFLAGS = -module -avoid-version
 export_ppm_la_LIBADD = $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
@@ -147,12 +147,12 @@
 export_pvn_la_LDFLAGS = -module -avoid-version
 
 export_raw_la_SOURCES = export_raw.c aud_aux.c
-export_raw_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_raw_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_raw_la_LDFLAGS = -module -avoid-version
 export_raw_la_LIBADD = $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
 export_tcaud_la_SOURCES = export_tcaud.c aud_aux.c
-export_tcaud_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_tcaud_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_tcaud_la_LDFLAGS = -module -avoid-version
 export_tcaud_la_LIBADD = $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
@@ -164,12 +164,12 @@
 export_wav_la_LIBADD = -lm
 
 export_xvid4_la_SOURCES = export_xvid4.c aud_aux.c
-export_xvid4_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_xvid4_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_xvid4_la_LDFLAGS = -module -avoid-version
 export_xvid4_la_LIBADD = $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
 export_yuv4mpeg_la_SOURCES = export_yuv4mpeg.c aud_aux.c
-export_yuv4mpeg_la_CPPFLAGS = $(AM_CPPFLAGS) $(MJPEGTOOLS_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
+export_yuv4mpeg_la_CPPFLAGS = $(AM_CPPFLAGS) $(MJPEGTOOLS_CFLAGS) $(LIBAVFORMAT_CFLAGS) $(LIBAVCODEC_CFLAGS) $(LAME_CFLAGS)
 export_yuv4mpeg_la_LDFLAGS = -module -avoid-version
 export_yuv4mpeg_la_LIBADD = $(MJPEGTOOLS_LIBS) $(LIBAVCODEC_LIBS) $(LAME_LIBS)
 
Index: export/ffmpeg_cfg.h
===================================================================
RCS file: /cvstc/transcode/export/ffmpeg_cfg.h,v
retrieving revision 1.15
diff -u -r1.15 ffmpeg_cfg.h
--- export/ffmpeg_cfg.h	17 Apr 2006 15:48:43 -0000	1.15
+++ export/ffmpeg_cfg.h	24 Mar 2008 15:29:26 -0000
@@ -2,7 +2,11 @@
 #define __FFMPEG_CFG_H
 
 #include "libtc/cfgfile.h"
+#ifdef HAVE_LIBAVHEADERS
+#include <avcodec.h>
+#else
 #include <ffmpeg/avcodec.h>
+#endif
 
 //char *lavc_param_vcodec = "mpeg4";
 //extern int lavc_param_vbitrate;
Index: filter/filter_resample.c
===================================================================
RCS file: /cvstc/transcode/filter/filter_resample.c,v
retrieving revision 1.22
diff -u -r1.22 filter_resample.c
--- filter/filter_resample.c	15 Dec 2007 09:49:14 -0000	1.22
+++ filter/filter_resample.c	24 Mar 2008 15:29:26 -0000
@@ -37,7 +37,7 @@
 #include "libtc/optstr.h"
 #include "libtc/tcmodule-plugin.h"
 
-#include <ffmpeg/avcodec.h>
+#include <avcodec.h>
 
 typedef struct
 {
Index: import/Makefile.am
===================================================================
RCS file: /cvstc/transcode/import/Makefile.am,v
retrieving revision 1.99
diff -u -r1.99 Makefile.am
--- import/Makefile.am	15 Dec 2007 10:42:52 -0000	1.99
+++ import/Makefile.am	24 Mar 2008 15:29:33 -0000
@@ -331,6 +331,7 @@
 	$(LIBTC_LIBS)
 
 tcdecode_CFLAGS = $(AM_CFLAGS) \
+	$(LIBAVFORMAT_CFLAGS) \
 	$(LIBAVCODEC_CFLAGS) \
 	$(LIBDV_CFLAGS) \
 	$(LAME_CFLAGS) \
Index: import/import_ffmpeg.c
===================================================================
RCS file: /cvstc/transcode/import/import_ffmpeg.c,v
retrieving revision 1.71
diff -u -r1.71 import_ffmpeg.c
--- import/import_ffmpeg.c	27 Jan 2008 09:11:57 -0000	1.71
+++ import/import_ffmpeg.c	24 Mar 2008 15:29:36 -0000
@@ -42,8 +42,13 @@
 #define MOD_PRE ffmpeg
 #include "import_def.h"
 
+#ifdef HAVE_LIBAVHEADERS
+#include <avcodec.h>
+#include <avformat.h>
+#else
 #include <ffmpeg/avcodec.h>
 #include <ffmpeg/avformat.h>
+#endif
 
 /*
  * libavcodec is not thread-safe. We must protect concurrent access to it.
Index: libtcext/tc_avcodec.h
===================================================================
RCS file: /cvstc/transcode/libtcext/tc_avcodec.h,v
retrieving revision 1.1
diff -u -r1.1 tc_avcodec.h
--- libtcext/tc_avcodec.h	27 Jan 2008 09:06:45 -0000	1.1
+++ libtcext/tc_avcodec.h	24 Mar 2008 15:29:41 -0000
@@ -27,9 +27,15 @@
 #include "config.h"
 #endif
 
+#ifdef HAVE_LIBAVHEADERS
+#include <avformat.h>
+#include <avcodec.h>
+#include <avutil.h>
+#else
 #include <ffmpeg/avformat.h>
 #include <ffmpeg/avcodec.h>
 #include <ffmpeg/avutil.h>
+#endif
 
 /*************************************************************************/
 
