diff -r 6d83826a4de6 libtcext/Makefile.am
--- a/libtcext/Makefile.am	Sun Oct 25 17:39:10 2009 +0100
+++ b/libtcext/Makefile.am	Tue Oct 27 21:59:46 2009 +0200
@@ -2,6 +2,7 @@
 
 AM_CPPFLAGS = \
 	$(PTHREAD_CFLAGS) \
+	$(GRAPHICSMAGICK_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/libtc \
@@ -10,6 +11,7 @@
 noinst_LTLIBRARIES = libtcext.la
 
 libtcext_la_SOURCES = ext.c
+libtcext_la_LIBADD = $(GRAPHICSMAGICK_LIBS)
 
 EXTRA_DIST = \
 	tc_avcodec.h \
