Index: Makefile
===================================================================
--- Makefile	(revision 676)
+++ Makefile	(working copy)
@@ -1,7 +1,7 @@
 
-prefix=/usr/local
-bindir=$(prefix)/bin
-datadir=$(prefix)/share
+prefix=/usr/lib
+bindir=/usr/bin
+datadir=$(prefix)
 docdir=$(datadir)/doc
 
 DESTDIR=
@@ -13,14 +13,15 @@
 UNINSTALL=rm
 UNINSTALL_DIR=rm -r
 
-pkg_datadir=$(datadir)/cjc
+pkg_datadir=$(datadir)
 pkg_docdir=$(docdir)/cjc
 
 VERSION=1.0.0
 SNAPSHOT=
 
-PY_DIRS=cjc cjc/ui plugins
-DOCS=doc/manual.html COPYING ChangeLog README TODO
+PY_DIRS=cjc cjc/ui
+PY_PLUGINDIRS=plugins
+DOCS=COPYING ChangeLog README TODO
 
 EXTRA_DIST=cjc.in cjc.py doc/manual.xml doc/Makefile
 
@@ -58,9 +59,10 @@
 		$(INSTALL_DIR) $(DESTDIR)$(pkg_datadir)/$$d ; \
 		$(INSTALL_DATA) $$d/*.py $(DESTDIR)$(pkg_datadir)/$$d ; \
 	done
-	python -c "import compileall; compileall.compile_dir('$(DESTDIR)$(pkg_datadir)', ddir='$(pkg_datadir)')" 
-	$(INSTALL_DIR) $(DESTDIR)$(pkg_docdir)
-	$(INSTALL_DATA) $(DOCS) $(DESTDIR)$(pkg_docdir)
+	for d in $(PY_PLUGINDIRS) ; do \
+		$(INSTALL_DIR) $(DESTDIR)$(pkg_datadir)/cjc/$$d ; \
+		$(INSTALL_DATA) $$d/*.py $(DESTDIR)$(pkg_datadir)/cjc/$$d ; \
+	done
 	$(INSTALL_DIR) $(DESTDIR)$(bindir)
 	$(INSTALL) cjc.inst $(DESTDIR)$(bindir)/cjc
 
