Index: doc/Makefile
===================================================================
--- doc/Makefile	(revision 676)
+++ doc/Makefile	(working copy)
@@ -2,14 +2,13 @@
 .SUFFIXES: .xml .fo .dvi .ps .pdf .html
 
 XSLTPROC=/usr/bin/xsltproc
-XSL_DIR=/usr/share/sgml/docbook/xsl-stylesheets/
+XSL_DIR=http://docbook.sourceforge.net/release/xsl/current
 HTML_STYLESHEET=$(XSL_DIR)/xhtml/docbook.xsl
 FO_STYLESHEET=$(XSL_DIR)/fo/docbook.xsl
 
 all: 
 	[ "x$(XSLTPROC)" != "x" ] \
 		&& [ -x "$(XSLTPROC)" ] \
-		&& [ -f "$(HTML_STYLESHEET)" ] \
 		&& make manual.html || :
 
 .fo.pdf:
