--- cjc/main.py.orig	2006-06-03 11:27:11.000000000 -0400
+++ cjc/main.py	2006-06-03 11:27:27.000000000 -0400
@@ -164,7 +164,7 @@
         else:
             home=os.environ.get("HOME","")
             self.home_dir=os.path.join(home,".cjc")
-        self.plugin_dirs=[os.path.join(base_dir,"plugins"),
+        self.plugin_dirs=[os.path.join(base_dir,"cjc/plugins"),
                     os.path.join(self.home_dir,"plugins")]
         self.plugins={}
         self.plugin_modules={}
