--- ./configure.in.orig	2007-12-25 11:25:31.000000000 +0200
+++ ./configure.in	2007-12-25 11:25:36.000000000 +0200
@@ -138,8 +138,8 @@
 dnl For debugging and to be user friendly
     AC_MSG_CHECKING([Hackish check for FLTK])
 dnl Call the config sripts.
-    LIBFLTK_CXXFLAGS=`fltk-config --cxxflags`
-    LIBFLTK_LIBS=`fltk-config --ldflags`
+    LIBFLTK_CXXFLAGS=`fltk2-config --cxxflags`
+    LIBFLTK_LIBS=`fltk2-config --ldflags`
 
 
 dnl ----------------
