--- ./configure.in.orig	2007-12-25 11:33:12.000000000 +0200
+++ ./configure.in	2007-12-25 11:36:52.000000000 +0200
@@ -425,11 +425,11 @@
       fi
     else
       CFLAGS="$CFLAGS `$PKG_CONFIG xft --cflags`"
-      LDFLAGS="$LDFLAGS `$PKG_CONFIG xft --libs`"
+      LIBS="$LIBS `$PKG_CONFIG xft --libs`"
     fi
   else
     CFLAGS="$CFLAGS `$XFT_CONFIG --cflags`"
-    LDFLAGS="$LDFLAGS `$XFT_CONFIG --libs`"
+    LIBS="$LIBS `$XFT_CONFIG --libs`"
   fi
 fi
 if test "x$enable_cookies" = "xno" ; then
