diff -ru dillo2/dpi/Makefile.am dillo2_nu/dpi/Makefile.am
--- dillo2/dpi/Makefile.am	2007-10-07 01:36:34.000000000 +0300
+++ dillo2_nu/dpi/Makefile.am	2007-10-23 01:38:07.000000000 +0300
@@ -1,11 +1,11 @@
-bookmarksdir = $(libdir)/dillo/dpi/bookmarks
-downloadsdir = $(libdir)/dillo/dpi/downloads
-ftpdir = $(libdir)/dillo/dpi/ftp
-httpsdir = $(libdir)/dillo/dpi/https
-hellodir = $(libdir)/dillo/dpi/hello
-filedir = $(libdir)/dillo/dpi/file
-cookiesdir = $(libdir)/dillo/dpi/cookies
-datauridir = $(libdir)/dillo/dpi/datauri
+bookmarksdir = $(libdir)/dillo2/dpi/bookmarks
+downloadsdir = $(libdir)/dillo2/dpi/downloads
+ftpdir = $(libdir)/dillo2/dpi/ftp
+httpsdir = $(libdir)/dillo2/dpi/https
+hellodir = $(libdir)/dillo2/dpi/hello
+filedir = $(libdir)/dillo2/dpi/file
+cookiesdir = $(libdir)/dillo2/dpi/cookies
+datauridir = $(libdir)/dillo2/dpi/datauri
 bookmarks_PROGRAMS = bookmarks.dpi
 downloads_PROGRAMS = downloads.dpi
 ftp_PROGRAMS = ftp.filter.dpi
diff -ru dillo2/dpid/dpi.c dillo2_nu/dpid/dpi.c
--- dillo2/dpid/dpi.c	2007-10-07 01:36:34.000000000 +0300
+++ dillo2_nu/dpid/dpi.c	2007-10-23 01:42:56.000000000 +0300
@@ -90,7 +90,7 @@
 
    dirfile_path = a_Dpi_sockdir_file();
    sockdir = dStrstrip(a_Dpi_rd_dpi_socket_dir(dirfile_path));
-   srs_name = dStrconcat(sockdir, "/", "dpid.srs", NULL);
+   srs_name = dStrconcat(sockdir, "/", "dpid2.srs", NULL);
    dFree(sockdir);
    dFree(dirfile_path);
    return (srs_name);
diff -ru dillo2/dpid/dpidc dillo2_nu/dpid/dpidc
--- dillo2/dpid/dpidc	2007-10-07 01:36:34.000000000 +0300
+++ dillo2_nu/dpid/dpidc	2007-10-23 01:42:45.000000000 +0300
@@ -14,7 +14,7 @@
 my $dir = <DSD>;
 close(DSD);
 
-my $socket = IO::Socket::UNIX->new(Peer => "$dir/dpid.srs", Type => SOCK_STREAM, Timeout => 1000 ) or die "new: $@";
+my $socket = IO::Socket::UNIX->new(Peer => "$dir/dpid2.srs", Type => SOCK_STREAM, Timeout => 1000 ) or die "new: $@";
 
 $socket->autoflush(1);
 
diff -ru dillo2/dpid/dpid.h dillo2_nu/dpid/dpid.h
--- dillo2/dpid/dpid.h	2007-10-07 01:36:34.000000000 +0300
+++ dillo2_nu/dpid/dpid.h	2007-10-23 01:44:15.000000000 +0300
@@ -27,7 +27,7 @@
 #define CMDLEN 20
 #define MSGLEN 50
 /*! \todo: Should read this from dillorc */
-#define SRS_NAME "dpid.srs"
+#define SRS_NAME "dpid2.srs"
 char *srs_name;
 
 /*! dpid service request socket */
diff -ru dillo2/dpid/Makefile.am dillo2_nu/dpid/Makefile.am
--- dillo2/dpid/Makefile.am	2007-10-07 01:36:34.000000000 +0300
+++ dillo2_nu/dpid/Makefile.am	2007-10-23 01:40:38.000000000 +0300
@@ -1,12 +1,12 @@
-AM_CPPFLAGS=-DDPIDRC_SYS='"$(sysconfdir)/dpidrc"'
+AM_CPPFLAGS=-DDPIDRC_SYS='"$(sysconfdir)/dpidrc2"'
 
-bin_PROGRAMS = dpid
-dpid_LDADD = ../dpip/libDpip.a ../dlib/libDlib.a
+bin_PROGRAMS = dpid2
+dpid2_LDADD = ../dpip/libDpip.a ../dlib/libDlib.a
 
-EXTRA_DIST = dpidc
-bin_SCRIPTS = dpidc
+EXTRA_DIST = dpidc2
+bin_SCRIPTS = dpidc2
 
-dpid_SOURCES = \
+dpid2_SOURCES = \
 	dpi.h \
 	dpi_service.h \
 	dpi_socket_dir.h \
@@ -23,9 +23,9 @@
 
 install-data-local :
 	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)
-	echo dpi_dir=$(libdir)/dillo/dpi > $(DESTDIR)$(sysconfdir)/dpidrc
-	echo >> $(DESTDIR)$(sysconfdir)/dpidrc
-	echo "proto.file=file/file.dpi" >> $(DESTDIR)$(sysconfdir)/dpidrc
-	echo "proto.ftp=ftp/ftp.filter.dpi" >> $(DESTDIR)$(sysconfdir)/dpidrc
-	echo "proto.https=https/https.filter.dpi" >> $(DESTDIR)$(sysconfdir)/dpidrc
-	echo "proto.data=datauri/datauri.filter.dpi" >> $(DESTDIR)$(sysconfdir)/dpidrc
+	echo dpi_dir=$(libdir)/dillo2/dpi > $(DESTDIR)$(sysconfdir)/dpidrc2
+	echo >> $(DESTDIR)$(sysconfdir)/dpidrc2
+	echo "proto.file=file/file.dpi" >> $(DESTDIR)$(sysconfdir)/dpidrc2
+	echo "proto.ftp=ftp/ftp.filter.dpi" >> $(DESTDIR)$(sysconfdir)/dpidrc2
+	echo "proto.https=https/https.filter.dpi" >> $(DESTDIR)$(sysconfdir)/dpidrc2
+	echo "proto.data=datauri/datauri.filter.dpi" >> $(DESTDIR)$(sysconfdir)/dpidrc2
diff -ru dillo2/Makefile.am dillo2_nu/Makefile.am
--- dillo2/Makefile.am	2007-10-07 01:36:34.000000000 +0300
+++ dillo2_nu/Makefile.am	2007-10-23 01:38:07.000000000 +0300
@@ -1,4 +1,4 @@
-SUBDIRS = doc dlib dpip src dpid dpi
+SUBDIRS = ../dw-testbed doc dlib dpip src dpid dpi
 
 EXTRA_DIST = ChangeLog.old dillorc2 install-dpi-local README-port
 
diff -ru dillo2/src/IO/dpi.c dillo2_nu/src/IO/dpi.c
--- dillo2/src/IO/dpi.c	2007-10-07 01:36:34.000000000 +0300
+++ dillo2_nu/src/IO/dpi.c	2007-10-23 01:46:32.000000000 +0300
@@ -305,11 +305,11 @@
    pid = fork();
    if (pid == 0) {
       /* This is the child process.  Execute the command. */
-      char *path1 = dStrconcat(dGethomedir(), "/.dillo/dpid", NULL);
+      char *path1 = dStrconcat(dGethomedir(), "/.dillo/dpid2", NULL);
       Dpi_close_fd(st_pipe[0]);
-      if (execl(path1, "dpid", NULL) == -1) {
+      if (execl(path1, "dpid2", NULL) == -1) {
          dFree(path1);
-         if (execlp("dpid", "dpid", NULL) == -1) {
+         if (execlp("dpid2", "dpid2", NULL) == -1) {
             DEBUG_MSG(4, "Dpi_start_dpid (child): %s\n", dStrerror(errno));
             do
                n = write(st_pipe[1], "ERROR", 5);
@@ -407,7 +407,7 @@
    char *dpid_uds_dir, *dpid_uds_name = NULL;
 
    if ((dpid_uds_dir = Dpi_get_dpid_uds_dir()) != NULL)
-      dpid_uds_name= dStrconcat(dpid_uds_dir, "/", "dpid.srs", NULL);
+      dpid_uds_name= dStrconcat(dpid_uds_dir, "/", "dpid2.srs", NULL);
 
    dFree(dpid_uds_dir);
    return dpid_uds_name;
@@ -498,7 +498,7 @@
       /* Get the server's uds name from dpid */
       sock = socket(AF_LOCAL, SOCK_STREAM, 0);
       dpid.sun_family = AF_LOCAL;
-      dpid_uds_name = dStrconcat(dpid_uds_dir, "/", "dpid.srs", NULL);
+      dpid_uds_name = dStrconcat(dpid_uds_dir, "/", "dpid2.srs", NULL);
       _MSG("dpid_uds_name = [%s]\n", dpid_uds_name);
       strncpy(dpid.sun_path, dpid_uds_name, sizeof(dpid.sun_path));
 
diff -ru dillo2/src/Makefile.am dillo2_nu/src/Makefile.am
--- dillo2/src/Makefile.am	2007-10-07 01:36:34.000000000 +0300
+++ dillo2_nu/src/Makefile.am	2007-10-23 01:38:07.000000000 +0300
@@ -1,4 +1,4 @@
-AM_CPPFLAGS=-DDILLORC_SYS='"$(sysconfdir)/dillorc"' @LIBJPEG_CPPFLAGS@
+AM_CPPFLAGS=-DDILLORC_SYS='"$(sysconfdir)/dillorc2"' @LIBJPEG_CPPFLAGS@
 AM_CFLAGS = -I../../dw-testbed @LIBPNG_CFLAGS@
 AM_CXXFLAGS = -I../../dw-testbed @LIBPNG_CFLAGS@ @LIBFLTK_CXXFLAGS@
 
