--- a/scfbuild/builder.py
+++ b/scfbuild/builder.py
@@ -11,7 +11,6 @@ import os
 import sys
 import tempfile
 import time
-from distutils.version import StrictVersion
 import xml.etree.ElementTree as ET
 
 import fontTools
@@ -28,9 +27,6 @@ logger = logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
 # Support for SVG tables was added to fontTools in version 2.5
-if StrictVersion('2.5') > StrictVersion(fontTools.version):
-    logger.exception("The FontTools module version must be 2.5 or higher.")
-    sys.exit(1)
 # todo: Check FontForge version
 
 
