--- a/sources/RobotoSlab.designspace
+++ b/sources/RobotoSlab.designspace
@@ -42,7 +42,7 @@
     <instances>
         <!-- instances that map directly to masters -->
         <instance familyname="RobotoSlab"
-                  filename="instances/RobotoSlab-Light.ufo"
+                  filename="instances/RobotoSlab-Thin.ufo"
                   name="Thin"
                   stylename="Thin">
             <location>
@@ -82,8 +82,8 @@
             <info />
         </instance>
         <!-- interpolated instances -->
-        <instance familyname="RobotoSLab"
-                  filename="instances/RobotoSLab-ExtraLight.ufo"
+        <instance familyname="RobotoSlab"
+                  filename="instances/RobotoSlab-ExtraLight.ufo"
                   name="ExtraLight"
                   stylename="ExtraLight">
             <location>
@@ -92,8 +92,8 @@
             <kerning />
             <info />
         </instance>
-            <instance familyname="RobotoSLab"
-                  filename="instances/RobotoSLab-Medium.ufo"
+            <instance familyname="RobotoSlab"
+                  filename="instances/RobotoSlab-Medium.ufo"
                   name="Medium"
                   stylename="Medium">
             <location>
