Index: SConstruct
===================================================================
--- SConstruct	(revision 962)
+++ SConstruct	(working copy)
@@ -127,7 +127,7 @@
 env.Append(LIBPATH=[iulibprefix+"/lib"])
 if not noeditline:
     env.Append(CXXFLAGS=" -DLUA_USE_EDITLINE ")
-    libs += ["editline"]
+    libs += ["readline"]
 if not notesseract:
     env.Append(CPPPATH=[tessprefix+"/include/tesseract"])
     env.Append(LIBPATH=[tessprefix+"/lib"])
