Index: ocroscript/lepton.pkg
===================================================================
--- ocroscript/lepton.pkg	(revision 912)
+++ ocroscript/lepton.pkg	(working copy)
@@ -29,8 +29,8 @@
 
 $#include "colib.h"
 $ extern "C" {
-$#include "leptonica/allheaders.h"
-$#include "leptonica/morph.h"
+$#include "liblept/allheaders.h"
+$#include "liblept/morph.h"
 $ using namespace colib;
 $ extern int count_pixRasterops;
 $ }
