Index: Jamrules.in
===================================================================
--- Jamrules.in	(revision 912)
+++ Jamrules.in	(working copy)
@@ -33,8 +33,9 @@
 # do NOT enable -DUNSAFE globally, except to detect high overhead error checking
 OPTIM = -g -O2 ;
 }
-ldflags = @LDFLAGS@ @LIBS@ ;
+ldflags = @LDFLAGS@ ;
 LINK = @CXX@ $(ldflags) ;
+LINKLIBS = @LIBS@ ;
 
 TESSHEADERS = @tessheaders@ ;
 TESSLIBS = @tesslibs@ ;
