diff --git a/src/log.cpp b/src/log.cpp
index aaa3b28..1deefc4 100644
--- a/src/log.cpp
+++ b/src/log.cpp
@@ -68,6 +68,7 @@ Logger::Logger(gint level)
 	g_log_set_default_handler(log, this);
 }
 
+#if 0
 void g_info(const gchar *fmt, ...)
 {
 	va_list va;
@@ -76,3 +77,4 @@ void g_info(const gchar *fmt, ...)
 	g_logv(G_LOG_DOMAIN, G_LOG_LEVEL_INFO, fmt, va);
 	va_end(va);
 }
+#endif
diff --git a/src/log.hpp b/src/log.hpp
index b1e042a..989001b 100644
--- a/src/log.hpp
+++ b/src/log.hpp
@@ -19,6 +19,5 @@ private:
 			gpointer user_data);
 };
 
-extern void g_info(const gchar *fmt, ...);
 
 #endif//!_LOG_HPP_
