--- a/buildVFs.sh
+++ b/buildVFs.sh
@@ -31,11 +31,11 @@ function build_var_font {
 	sfntedit -a CFF2=$1/.tb_cff2 $1/$2.otf
 
 	# comment out STAT feature file which cannot be digested by fontmake
-	sed -i '' 's/^/#/' $1/../STAT.fea
+	sed -i 's/^/#/' $1/../STAT.fea
 	# build variable TTF with fontmake. Use set +e so we can un-hack the
 	# STAT.fea before exiting.
-	set +e
 	fontmake -m $1/$2.designspace -o variable --production-names --output-path $1/$2.ttf --feature-writer None
+if false; then
 	fm_status=$?
 	# undo changes to STAT feature file
 	sed -i '' 's/#//' $1/../STAT.fea
@@ -44,6 +44,9 @@ function build_var_font {
 		exit $fm_status
 	fi
 	set -e
+else
+	sed -i 's/#//' $1/../STAT.fea
+fi
 	# use DSIG, name, OS/2, hhea, post, and STAT tables from OTFs
 	sfntedit -x DSIG=$1/.tb_DSIG,name=$1/.tb_name,OS/2=$1/.tb_os2,hhea=$1/.tb_hhea,post=$1/.tb_post,STAT=$1/.tb_STAT,fvar=$1/.tb_fvar $1/$2.otf
 	sfntedit -a DSIG=$1/.tb_DSIG,name=$1/.tb_name,OS/2=$1/.tb_os2,hhea=$1/.tb_hhea,post=$1/.tb_post,STAT=$1/.tb_STAT,fvar=$1/.tb_fvar $1/$2.ttf
