Make install targets more friendly
--- svgalib-1.9.21/Makefile
+++ svgalib-1.9.21/Makefile
@@ -78,10 +78,6 @@ default:
 
 installheaders:
 	@echo Installing header files in $(includedir).
-	@if [ -f /usr/include/vga.h ]; then \
-		rm -f /usr/include/vga.h /usr/include/vgagl.h; \
-		echo Old header files in /usr/include removed.; \
-	fi
 	@mkdir -p $(includedir)
 	@cp $(SRCDIR)/src/vga.h $(includedir)/vga.h
 	@chmod a+r $(includedir)/vga.h
@@ -95,16 +91,6 @@ installheaders:
 	@chmod a+r $(includedir)/vgakeyboard.h
 
 installsharedlib: $(SHAREDLIBS) $(SVGALIBSHAREDSTUBS)
-	@if [ -f /usr/lib/libvga.sa ]; then \
-		rm -f /usr/lib/libvga.sa /usr/lib/libvgagl.sa; \
-		echo Old stubs in /usr/lib removed.; \
-	fi
-ifndef KEEPSHAREDLIBS
-	@echo "Removing shared library images (old & current)..."
-	@for i in $(OBSOLETELDIRS); do \
-	    rm -f `echo $(OBSOLETESHAREDIMAGES) | sed s?/lib/?$$i?g`; \
-         done
-endif
 	@echo Installing shared library image as \
 		$(addprefix $(sharedlibdir)/,$(notdir $(SHAREDLIBS))).
 	@for foo in $(notdir $(SHAREDLIBS)); do \
@@ -118,10 +104,8 @@ endif
 installstaticlib: static
 	@echo Installing static libraries in $(libdir).
 	@mkdir -p $(libdir)
-	@rm -f /usr/lib/libvga.a
 	@$(INSTALL_DATA) staticlib/libvga.a $(libdir)/libvga.a
 	@chmod a+r $(libdir)/libvga.a
-	@rm -f /usr/lib/libvgagl.a
 	@$(INSTALL_DATA) staticlib/libvgagl.a $(libdir)/libvgagl.a
 	@chmod a+r $(libdir)/libvgagl.a
 
@@ -130,14 +114,6 @@ installutils: textutils $(LRMI)
 		echo No $(bindir) directory, creating it.; \
 		mkdir -p $(bindir); \
 	fi
-	@if [ -f /usr/bin/restorefont ]; then \
-		echo Removing old utilities in /usr/bin.; \
-		for x in $(UTILS); do rm -f /usr/bin/$$x; done; \
-	fi
-	@if [ -f /usr/bin/convfont ]; then \
-		echo Removing inappropriate utilities in /usr/bin.; \
-		rm -f /usr/bin/convfont /usr/bin/setmclk; \
-	fi
 	@echo Installing textmode utilities in $(bindir):
 	@echo "restorefont:      Save/restore textmode font."
 	@$(INSTALL_PROGRAM) utils/restorefont $(bindir)
--- svgalib-1.9.21/Makefile.cfg
+++ svgalib-1.9.21/Makefile.cfg
@@ -44,7 +44,7 @@ CONFDIR = $(SRCDIR)/src/config
 # Common prefix for installation directories.
 # NOTE: This directory must exist when you start the install.
 TOPDIR=
-prefix = $(TOPDIR)/usr/local
+prefix = $(TOPDIR)/usr
 exec_prefix = $(prefix)
 
 # Directory where the shared stubs and static library will be installed.
@@ -63,7 +63,7 @@
 includedir = $(prefix)/include
 
 # Directory where the man files will be installed.
-mandir = $(prefix)/man
+mandir = $(prefix)/share/man
 
 # Target binary format.
 TARGET_FORMAT = elf
@@ -126,7 +126,7 @@ INSTALLMAN = installman
 # MAKEWHATIS = makewhatis # Beware, this will really need a few minutes!
 
 # Comment this out if you use devfs only and don't want to make svga nodes
-INSTALLDEV = installdev
+#INSTALLDEV = installdev
 
 # Select which makefile to use to install the module, or comment both out if
 # you are going to install the kernel module manually
