--- a/cmake/modules/vvencTests.cmake
+++ b/cmake/modules/vvencTests.cmake
@@ -19,7 +19,9 @@ add_test( NAME Test_vvenclibtest-sdk_def
 add_test( NAME Test_vvenclibtest-sdk_stringapi_interface COMMAND vvenclibtest 5 )
 add_test( NAME Test_vvenclibtest-timestamps              COMMAND vvenclibtest 6 )
 
+if( NOT BUILD_SHARED_LIBS )
 add_test( NAME Test_vvenc_unit_test COMMAND vvenc_unit_test )
+endif()
 
 set( CLEANUP_TEST_FILES "" )
 
