--- a/verdafile.js
+++ b/verdafile.js
@@ -569,6 +569,12 @@ async function buildCompositeTtc(out, in
 	);
 }
 
+const CollectionExport = task.group("collection-export", async (target, gr) => {
+	const [collectPlans] = await target.need(CollectPlans, de`${DIST}/ttc-collect/${gr}`);
+	const sourceGroups = collectPlans.groupDecomposition[gr];
+	const ttcFiles = Array.from(new Set(collectPlans.ttcContents[gr]));
+	await target.need(ttcFiles.map(pt => ExportTtcFile(gr, pt)));
+});
 const ExportSuperTtc = file.make(
 	gr => `${DIST_SUPER_TTC}/${gr}.ttc`,
 	async (target, out, gr) => {
