--- a/verdafile.js
+++ b/verdafile.js
@@ -446,6 +446,13 @@
 	await run("ttfautohint", tmpTtc, out.full);
 	await rm(tmpTtc);
 }
+
+const CollectionExport = task.group("collection-export", async (target, gr) => {
+	const [collectPlans] = await target.need(CollectPlans, de`${DIST}/ttc-collect/${gr}`);
+	const sourceGroups = collectPlans.groupDecomposition[gr];
+	const ttcFiles = Array.from(new Set(collectPlans.ttcContents[gr]));
+	await target.need(ttcFiles.map(pt => CollectedTtcFile(gr, pt)));
+});
 
 ///////////////////////////////////////////////////////////
 //////                   Archives                    //////
