Index: configure
===================================================================
RCS file: /cvsroot/gpac/gpac/configure,v
retrieving revision 1.17
diff -u -r1.17 configure
--- configure	12 Jul 2006 17:25:26 -0000	1.17
+++ configure	19 Jul 2006 13:22:06 -0000
@@ -555,6 +555,8 @@
 
 if $cc -o $TMPO $TMPC -lz -lavcodec -lavformat 2> /dev/null  ; then
 has_ffmpeg="system"
+elif $cc -o $TMPO $TMPC $(pkg-config libavformat --libs --cflags) 2> /dev/null ; then
+has_ffmpeg="system"
 elif test "$alt_macosx_dir" != "" ; then
  if $cc -o $TMPO $TMPC -I$alt_macosx_dir/include -L$alt_macosx_dir/lib -lz -lavcodec -lavformat 2> /dev/null ; then
  has_ffmpeg="system"
