--- src/format_wv.c.orig	2004-05-05 08:39:12.000000000 +0300
+++ src/format_wv.c	2005-08-07 13:17:30.893940750 +0300
@@ -77,6 +77,7 @@
   arg_add(WVUNPACK);
   arg_add("-y");
   arg_add(filename);
+  arg_add("-o");
   arg_add("-");
 
   *pid = spawn_input(&input,&output);
@@ -165,7 +166,7 @@
 
   fclose(info->input);
 
-  if (tagcmp(wph.ckID,WAVPACK_MAGIC) || wph.version < 1 || wph.version > 3 || (wph.flags & RAW_FLAG)) {
+  if (tagcmp(wph.ckID,WAVPACK_MAGIC) || wph.version < 1) {
     return NOT_WV_FILE;
   }
 
