--- a/Makefile
+++ b/Makefile
@@ -128,9 +128,9 @@
 
 EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped
 
-KBASE              ?= /lib/modules/`uname -r`
+KBASE              ?= /lib/modules/${KV_FULL}
 KBUILD_DIR         ?= $(KBASE)/build
-MDEST_DIR          ?= $(KBASE)/kernel/drivers/net/wireless
+MDEST_DIR          ?= ${ED}$(KBASE)/kernel/drivers/net/wireless
 
 all:
 	KBUILD_NOPEDANTIC=1 make -C $(KBUILD_DIR) M=`pwd`
