--- configure.orig	2006-06-27 03:24:10.000000000 +0300
+++ configure	2006-07-26 17:57:05.000000000 +0300
@@ -29935,8 +29933,8 @@
 
 
 CFLAGS="${OPTIMIZATION} ${CFLAGS}"
-LDADD="${LDADD}"
-FRONTEND_LDADD="${FRONTEND_LDADD} ${CONFIG_MATH_LIB}"
+LDADD="${LDADD} ${CONFIG_MATH_LIB}"
+FRONTEND_LDADD="${FRONTEND_LDADD}"
 
 
 
--- libmp3lame/Makefile.in.orig	2006-06-27 03:24:12.000000000 +0300
+++ libmp3lame/Makefile.in	2006-07-26 18:35:47.000000000 +0300
@@ -232,7 +232,7 @@
 @HAVE_NASM_TRUE@nasm_ldadd = $(top_builddir)/libmp3lame/@CPUTYPE@/liblameasmroutines.la
 @LIB_WITH_DECODER_FALSE@decoder_ldadd = 
 @LIB_WITH_DECODER_TRUE@decoder_ldadd = $(top_builddir)/mpglib/libmpgdecoder.la
-libmp3lame_la_LIBADD = $(nasm_ldadd) $(decoder_ldadd)
+libmp3lame_la_LIBADD = $(nasm_ldadd) $(decoder_ldadd) $(LDADD)
 libmp3lame_la_LDFLAGS = -version-info @LIB_MAJOR_VERSION@:@LIB_MINOR_VERSION@ \
 			-no-undefined
 
