--- a/debian/CMakeLists.inj
+++ b/debian/CMakeLists.inj
@@ -13,6 +13,7 @@ add_precompiled_header(lib_base ../../Te
 add_precompiled_header(lib_export ../../Telegram/SourceFiles/export/export_pch.h)
 add_precompiled_header(lib_mtproto ../../Telegram/SourceFiles/mtproto/mtp_pch.h)
 add_precompiled_header(lib_storage ../../Telegram/SourceFiles/storage/storage_pch.h)
+add_precompiled_header(lib_ui ../../Telegram/SourceFiles/ui/ui_pch.h)
 
 # Telegram uses long atomic types, so on some architectures libatomic is needed.
 include(CheckCXXSourceCompiles)
