--- a/Telegram/lib_ui/CMakeLists.txt
+++ b/Telegram/lib_ui/CMakeLists.txt
@@ -33,7 +33,6 @@ set(style_files
     ui/colors.palette
     emoji_suggestions/emoji_autocomplete.json
 
-    fonts/fonts.qrc
     qt_conf/mac.qrc
     qt_conf/win.qrc

--- a/Telegram/lib_ui/ui/style/style_core_font.cpp
+++ b/Telegram/lib_ui/ui/style/style_core_font.cpp
@@ -15,7 +15,6 @@
 #include <QtGui/QFontDatabase>
 
 void style_InitFontsResource() {
-	Q_INIT_RESOURCE(fonts);
 #ifdef Q_OS_WIN
 	Q_INIT_RESOURCE(win);
 #elif defined Q_OS_MAC // Q_OS_WIN
@@ -126,6 +126,14 @@ void StartFonts() {
 	}
 	Started = true;
 
+#ifdef Q_OS_LINUX
+	auto list = QStringList();
+	list.append("Open Sans");
+	list.append("sans-serif");
+	for (const auto &name : FontTypeNames) {
+		QFont::insertSubstitutions(name, list);
+	}
+#else
 	style_InitFontsResource();
 
 	bool areGood[FontTypesCount] = { false };
@@ -152,6 +160,7 @@ void StartFonts() {
 		//QFont::insertSubstitution(name, fallback);
 #endif // Q_OS_WIN
 	}
+#endif
 #ifdef Q_OS_MAC
 	auto list = QStringList();
 	list.append(".SF NS Text");
