--- a/Telegram/CMakeLists.txt
+++ b/Telegram/CMakeLists.txt
@@ -1034,7 +1034,7 @@ elseif (LINUX)
     if (NOT TDESKTOP_DISABLE_GTK_INTEGRATION)
         find_package(PkgConfig REQUIRED)
 
-        pkg_search_module(GTK REQUIRED gtk+-2.0 gtk+-3.0)
+        pkg_search_module(GTK REQUIRED gtk+-3.0)
         target_include_directories(Telegram PRIVATE ${GTK_INCLUDE_DIRS})
         target_compile_options(Telegram PRIVATE -Wno-register)
 
--- a/Telegram/lib_spellcheck/CMakeLists.txt
+++ b/Telegram/lib_spellcheck/CMakeLists.txt
@@ -44,4 +44,5 @@ if (LINUX)
 
     pkg_search_module(ENCHANT REQUIRED enchant-2 enchant)
     target_include_directories(lib_spellcheck PRIVATE ${ENCHANT_INCLUDE_DIRS})
+    target_link_libraries(lib_spellcheck PRIVATE dl)
 endif()
--- a/Telegram/lib_ui/CMakeLists.txt
+++ b/Telegram/lib_ui/CMakeLists.txt
@@ -192,6 +192,8 @@ target_include_directories(lib_ui
 PUBLIC
     ${src_loc}
     ${src_loc}/emoji_suggestions
+    ${Qt5Gui_PRIVATE_INCLUDE_DIRS}
+    ${Qt5Core_PRIVATE_INCLUDE_DIRS}
 )
 
 target_link_libraries(lib_ui
