--- a/cmake/external/crash_reports/CMakeLists.txt
+++ b/cmake/external/crash_reports/CMakeLists.txt
@@ -8,6 +8,12 @@ add_library(external_crash_reports INTER
 add_library(desktop-app::external_crash_reports ALIAS external_crash_reports)
 
 if (NOT DESKTOP_APP_DISABLE_CRASH_REPORTS)
+if (DESKTOP_APP_USE_PACKAGED)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(BREAKPAD REQUIRED breakpad-client)
+    target_include_directories(external_crash_reports INTERFACE ${BREAKPAD_INCLUDE_DIRS})
+    target_link_libraries(external_crash_reports INTERFACE ${BREAKPAD_LIBRARIES})
+else()
     if (WIN32 OR LINUX OR build_macstore)
         add_subdirectory(breakpad)
         target_link_libraries(external_crash_reports
@@ -22,3 +28,4 @@ if (NOT DESKTOP_APP_DISABLE_CRASH_REPORT
         )
     endif()
 endif()
+endif()
--- a/Telegram/SourceFiles/stdafx.h
+++ b/Telegram/SourceFiles/stdafx.h
@@ -83,7 +83,7 @@ https://github.com/telegramdesktop/tdesk
 #endif // OS_MAC_OLD
 
 // Fix Google Breakpad build for Mac App Store version
-#ifdef Q_OS_MAC
+#if defined Q_OS_MAC || defined Q_OS_LINUX
 #define __STDC_FORMAT_MACROS
 #endif // Q_OS_MAC
 
