--- configure.orig	2006-01-29 00:04:47.000000000 +0200
+++ configure	2006-12-17 11:53:52.000000000 +0200
@@ -30334,7 +30334,7 @@
 # Check whether --enable-oggvorbis or --disable-oggvorbis was given.
 if test "${enable_oggvorbis+set}" = set; then
   enableval="$enable_oggvorbis"
-   ac_enable_oggvorbis=no
+   ac_enable_oggvorbis=$enableval
 else
    ac_enable_oggvorbis=yes
 fi;
@@ -30343,7 +30343,7 @@
 # Check whether --enable-speex or --disable-speex was given.
 if test "${enable_speex+set}" = set; then
   enableval="$enable_speex"
-   ac_enable_speex=no
+   ac_enable_speex=$enableval
 else
    ac_enable_speex=yes
 fi;
@@ -32158,7 +32158,7 @@
 # Check whether --enable-mad or --disable-mad was given.
 if test "${enable_mad+set}" = set; then
   enableval="$enable_mad"
-   ac_enable_mad=no
+   ac_enable_mad=$enableval
 else
    ac_enable_mad=yes
 fi;
@@ -32398,7 +32398,7 @@
 # Check whether --enable-src or --disable-src was given.
 if test "${enable_src+set}" = set; then
   enableval="$enable_src"
-   ac_enable_src=no
+   ac_enable_src=$enableval
 else
    ac_enable_src=yes
 fi;
@@ -32473,7 +32473,9 @@
 # Check whether --enable-alsa or --disable-alsa was given.
 if test "${enable_alsa+set}" = set; then
   enableval="$enable_alsa"
-   ac_enable_alsa=yes
+   ac_enable_alsa=$enableval
+else
+   ac_enable_alsa=no
 fi;
 
 if test "x${ac_enable_alsa}" != xno ; then
