--- a/Makefile
+++ b/Makefile
@@ -43,14 +43,16 @@ ifeq (,$(wildcard $(BROTLI)/*))
   $(error Brotli dependency not found : you must initialize the Git submodule)
 endif
 
-all : $(OBJS) $(EXECUTABLES) $(ARCHIVES)
+all : $(EXECUTABLES)
 
 $(ARCHIVES) : $(ARCHIVE_OBJS) $(OBJS) deps
 	$(AR) $(ARFLAGS) $(SRCDIR)/$@.a $(OBJS) \
 	      $(COMMONOBJ) $(ENCOBJ) $(DECOBJ) $(SRCDIR)/$@.o
 
-$(EXECUTABLES) : $(EXE_OBJS) deps
-	$(CXX) $(LFLAGS) $(OBJS) $(COMMONOBJ) $(ENCOBJ) $(DECOBJ) $(SRCDIR)/$@.o -o $@
+libwoff2.a : $(OBJS)
+	$(AR) $(ARFLAGS) $@ $^
+$(EXECUTABLES) : libwoff2.a $(EXE_OBJS)
+	$(CXX) $(LDFLAGS) $(SRCDIR)/$@.o -o $@ $< $(LIBS)
 
 deps :
 	$(MAKE) -C $(BROTLI) lib
