--- a/Lib/fontTools/feaLib/lexer.py
+++ b/Lib/fontTools/feaLib/lexer.py
@@ -28,7 +28,7 @@ class Lexer(object):
     CHAR_NAME_START_ = CHAR_LETTER_ + "_+*:.^~!\\"
     CHAR_NAME_CONTINUATION_ = CHAR_LETTER_ + CHAR_DIGIT_ + "_.+*:^~!/-"
 
-    RE_GLYPHCLASS = re.compile(r"^[A-Za-z_0-9.]+$")
+    RE_GLYPHCLASS = re.compile(r"^[A-Za-z_0-9.-]+$")
 
     MODE_NORMAL_ = "NORMAL"
     MODE_FILENAME_ = "FILENAME"
