--- a/host-linux/chrome-token-signing.pro
+++ b/host-linux/chrome-token-signing.pro
@@ -28,4 +28,4 @@ extensionpol.path = /etc/opt/chrome/policies/managed
 extensionpol.files += ee.ria.chrome-token-signing.policy.json
 ffextension.path = /usr/share/mozilla/extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}
 ffextension.files += ../{443830f0-1fff-4f9a-aa1e-444bafbc7319}.xpi
-INSTALLS += target hostconf ffconf extension ffextension
+INSTALLS += target hostconf ffconf extensionpol ffextension
--- a/host-linux/ee.ria.chrome-token-signing.policy.json
+++ b/host-linux/ee.ria.chrome-token-signing.policy.json
@@ -1,5 +1,3 @@
 {
-  "ExtensionInstallForcelist": [
-    "ckjefchnfjhjfedoccjbhjpbncimppeg;https://clients2.google.com/service/update2/crx"
-  ]
+"ExtensionSettings":{"ckjefchnfjhjfedoccjbhjpbncimppeg":{"installation_mode":"normal_installed","update_url":"https://clients2.google.com/service/update2/crx"}}
 }
