=== modified file 'CMakeLists.txt'
--- a/CMakeLists.txt	2016-02-04 19:54:52 +0000
+++ b/CMakeLists.txt	2016-08-17 19:36:07 +0000
@@ -112,24 +112,10 @@
                   sqlite3>=3.6.19
                   gmodule-2.0
                   gio-2.0>=2.32.3
-                  libsoup-gnome-2.4>=2.37.1
                   )
 add_definitions("-DHAVE_LIBXML")
 add_definitions("-DGIO_VERSION=\"${DEPS_gio-2.0_VERSION}\"")
-add_definitions("-DLIBSOUP_VERSION=\"${DEPS_libsoup-gnome-2.4_VERSION}\"")
 set(PKGS posix linux libxml-2.0 sqlite3 gmodule-2.0 gio-2.0 libsoup-2.4)
-if (${DEPS_libsoup-gnome-2.4_VERSION} VERSION_GREATER "2.40.0")
-    # valac 0.16 didn't have the bindings yet
-    # For consistency we need to ensure C code makes the same assumptions
-    if (${VALA_VERSION} VERSION_GREATER "0.17.0")
-        add_definitions("-DHAVE_LIBSOUP_2_40_0")
-        set(VALAFLAGS ${VALAFLAGS} -D HAVE_LIBSOUP_2_40_0)
-    endif ()
-endif ()
-if (${DEPS_libsoup-gnome-2.4_VERSION} VERSION_GREATER "2.48.0")
-    add_definitions("-DHAVE_LIBSOUP_2_48_0")
-    set(VALAFLAGS ${VALAFLAGS} -D HAVE_LIBSOUP_2_48_0)
-endif ()
 
 if (${DEPS_gio-2.0_VERSION} VERSION_GREATER "2.40.0" OR WIN32)
     add_definitions("-DLIBNOTIFY_VERSION=\"No\"")
@@ -219,9 +205,23 @@
                       gtk+-2.0>=2.24.0
                       webkit-1.0>=1.8.1
                       javascriptcoregtk-1.0
+                      libsoup-gnome-2.4>=2.37.1
                       )
     add_definitions("-DGTK_VERSION=\"${DEPS_GTK_gtk+-2.0_VERSION}\"")
     add_definitions("-DWEBKIT_VERSION=\"${DEPS_GTK_webkit-1.0_VERSION}\"")
+    add_definitions("-DLIBSOUP_VERSION=\"${DEPS_libsoup-gnome-2.4_VERSION}\"")
+    if (${DEPS_libsoup-gnome-2.4_VERSION} VERSION_GREATER "2.40.0")
+        # valac 0.16 didn't have the bindings yet
+        # For consistency we need to ensure C code makes the same assumptions
+        if (${VALA_VERSION} VERSION_GREATER "0.17.0")
+            add_definitions("-DHAVE_LIBSOUP_2_40_0")
+            set(VALAFLAGS ${VALAFLAGS} -D HAVE_LIBSOUP_2_40_0)
+        endif ()
+    endif ()
+    if (${DEPS_libsoup-gnome-2.4_VERSION} VERSION_GREATER "2.48.0")
+        add_definitions("-DHAVE_LIBSOUP_2_48_0")
+        set(VALAFLAGS ${VALAFLAGS} -D HAVE_LIBSOUP_2_48_0)
+    endif ()
     set(PKGS ${PKGS} gtk+-2.0)
     set(EXTRA_VAPIS ${EXTRA_VAPIS} "${CMAKE_SOURCE_DIR}/midori/webkitgtk-3.0.vapi")
 endif ()

=== modified file 'extensions/cookie-manager/cookie-manager.c'
--- a/extensions/cookie-manager/cookie-manager.c	2013-07-29 21:05:02 +0000
+++ b/extensions/cookie-manager/cookie-manager.c	2016-08-17 19:53:55 +0000
@@ -12,7 +12,6 @@
 #include "config.h"
 #include <midori/midori.h>
 #include "katze/katze.h"
-#include <libsoup/soup-cookie-jar-sqlite.h>
 
 #include "cookie-manager.h"
 #include "cookie-manager-page.h"
@@ -282,7 +281,7 @@
 	/* setup soup */
 #ifdef HAVE_WEBKIT2
 	gchar *filename = midori_paths_get_config_filename_for_writing ("cookies.db");
-	priv->jar = soup_cookie_jar_sqlite_new (filename, FALSE);
+	priv->jar = soup_cookie_jar_db_new (filename, FALSE);
 	g_free(filename);
 #else
 	SoupSession *session = webkit_get_default_session();

=== modified file 'midori/midori-session.c'
--- a/midori/midori-session.c	2015-03-23 11:33:24 +0000
+++ b/midori/midori-session.c	2016-08-17 19:36:07 +0000
@@ -17,13 +17,13 @@
 #include "sokoke.h"
 
 #include <glib/gi18n-lib.h>
+#ifndef HAVE_WEBKIT2
 #include <libsoup/soup-cookie-jar-sqlite.h>
 #include <libsoup/soup-gnome-features.h>
 
     #define LIBSOUP_USE_UNSTABLE_REQUEST_API
     #include <libsoup/soup-cache.h>
 
-#ifndef HAVE_WEBKIT2
 static void
 midori_soup_session_set_proxy_uri (SoupSession* session,
                                    const gchar* uri)

=== modified file 'midori/midori-view.c'
--- a/midori/midori-view.c	2016-02-28 00:07:54 +0000
+++ b/midori/midori-view.c	2016-08-17 19:36:07 +0000
@@ -3840,11 +3840,11 @@
 #ifndef HAVE_WEBKIT2
     midori_view_add_version (markup, html, g_strdup_printf ("WebKitGTK+ %s (%u.%u.%u)\tlibSoup %s",
         WEBKIT_VERSION, webkit_major_version (), webkit_minor_version (), webkit_micro_version (),
-#else
-    midori_view_add_version (markup, html, g_strdup_printf ("WebKit2GTK+ %s (%u.%u.%u)\tlibSoup %s",
-        WEBKIT_VERSION, webkit_get_major_version (), webkit_get_minor_version (), webkit_get_micro_version (),
-#endif
         LIBSOUP_VERSION));
+#else
+    midori_view_add_version (markup, html, g_strdup_printf ("WebKit2GTK+ %s (%u.%u.%u)",
+        WEBKIT_VERSION, webkit_get_major_version (), webkit_get_minor_version (), webkit_get_micro_version ()));
+#endif
     midori_view_add_version (markup, html, g_strdup_printf ("cairo %s (%s)\tlibnotify %s",
         CAIRO_VERSION_STRING, cairo_version_string (),
         LIBNOTIFY_VERSION));

