--- a/python/psautohint/__init__.py
+++ b/python/psautohint/__init__.py
@@ -15,10 +15,9 @@ log = logging.getLogger(__name__)
 
 
 AUTOHINTEXE = os.path.join(
-    os.path.dirname(__file__),
     "autohintexe" + (".exe" if sys.platform in ("win32", "cygwin") else "")
 )
-if not os.path.isfile(AUTOHINTEXE) or not os.access(AUTOHINTEXE, os.X_OK):
+if False:
     import warnings
     warnings.warn(
         "embedded 'autohintexe' executable not found: %r" % AUTOHINTEXE
--- a/setup.py
+++ b/setup.py
@@ -564,6 +564,7 @@ ext_modules = [
         sources=[
             "python/psautohint/_psautohint.c",
         ],
+        libraries = ['psautohint'],
     ),
 ]
 
