Index: configure.ac
===================================================================
--- configure.ac	(revision 225)
+++ configure.ac	(working copy)
@@ -6,8 +6,6 @@
 
 AM_PROG_LIBTOOL
 
-CFLAGS="$CFLAGS -O3 -fomit-frame-pointer -fPIC"
-
 AC_C_BIGENDIAN(,CFLAGS="$CFLAGS -DMPC_LITTLE_ENDIAN",)
 
 AC_HEADER_STDC
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 225)
+++ src/Makefile.am	(working copy)
@@ -20,3 +20,4 @@
 sample_LDADD = libmpcdec.la
 
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/src
+AM_CFLAGS= -O3 -fomit-frame-pointer
