Index: libmpcdec/Makefile.am
===================================================================
--- libmpcdec/Makefile.am	(revision 449)
+++ libmpcdec/Makefile.am	(working copy)
@@ -2,9 +2,6 @@
 
 AM_CPPFLAGS = -I$(top_srcdir)/include
 
-if HAVE_VISIBILITY
-  AM_CFLAGS = -fvisibility=hidden
-endif
 
 common_sources = ../common/crc32.c \
 		../common/huffman-bcl.c
Index: configure.in
===================================================================
--- configure.in	(revision 449)
+++ configure.in	(working copy)
@@ -30,22 +30,11 @@
 AM_CONDITIONAL([MPC_CHAP], [test "x$enable_mpcchap" = xyes])
 
 
-CHECK_VISIBILITY
 
 AC_CONFIG_FILES([
 Makefile
 include/Makefile
 libmpcdec/Makefile
-libmpcenc/Makefile
-libmpcpsy/Makefile
-libwavformat/Makefile
-mpc2sv8/Makefile
-mpcchap/Makefile
-mpccut/Makefile
-mpcdec/Makefile
-mpcenc/Makefile
-mpcgain/Makefile
-wavcmp/Makefile
 ])
 
 AC_OUTPUT
Index: Makefile.am
===================================================================
--- Makefile.am	(revision 449)
+++ Makefile.am	(working copy)
@@ -1,6 +1,5 @@
 AUTOMAKE_OPTIONS = foreign
 
-ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = CMakeLists.txt libmpc.kdevelop docs win32
 
@@ -13,15 +12,5 @@
 
 SUBDIRS = \
 include \
-libmpcdec \
-libmpcenc \
-libmpcpsy \
-libwavformat \
-mpcenc \
-mpc2sv8 \
-mpccut \
-mpcdec \
-wavcmp \
-$(MPCCHAP_DIR) \
-$(MPCGAIN_DIR)
+libmpcdec
 
