--- libmpcdec-1.2.6/configure.ac	2007-04-17 22:05:32.000000000 +0300
+++ libmpcdec-1.2.6_nu/configure.ac	2010-03-05 17:58:23.000000000 +0200
@@ -6,8 +6,6 @@
 
 AM_PROG_LIBTOOL
 
-CFLAGS="$CFLAGS -O3 -fomit-frame-pointer -fPIC"
-
 AC_C_BIGENDIAN(,CFLAGS="$CFLAGS -DMPC_LITTLE_ENDIAN",)
 
 AC_HEADER_STDC
--- libmpcdec-1.2.6/src/Makefile.am	2007-04-17 21:28:07.000000000 +0300
+++ libmpcdec-1.2.6_nu/src/Makefile.am	2010-03-05 16:42:02.000000000 +0200
@@ -2,9 +2,9 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-lib_LTLIBRARIES = libmpcdec.la
+lib_LTLIBRARIES = libmpcdecsv7.la
 
-libmpcdec_la_SOURCES = \
+libmpcdecsv7_la_SOURCES = \
 	huffsv46.c \
 	huffsv7.c \
 	idtag.c \
@@ -13,11 +13,6 @@
 	requant.c \
 	streaminfo.c \
 	synth_filter.c
-libmpcdec_la_LDFLAGS = -no-undefined -version-info 5:2:0
-
-noinst_PROGRAMS = sample
-sample_SOURCES = sample.cpp
-sample_LDADD = libmpcdec.la
-sample_LDFLAGS = -static
+libmpcdecsv7_la_LDFLAGS = -no-undefined -version-info 5:2:0
 
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/src
