--- a/addSVGtable.py
+++ b/addSVGtable.py
@@ -8,7 +8,6 @@ The file names of the SVG glyphs need to match their corresponding glyph final n
 import os
 import sys
 import re
-from distutils.version import StrictVersion
 
 try:
 	from fontTools import ttLib, version
@@ -19,7 +18,7 @@ except ImportError:
 # support for the SVG table was added to FontTools on Aug 28, 2013
 # https://github.com/behdad/fonttools/commit/ddcca79308b52dc36b24ef94cab4ab00c8e32376
 minFontToolsVersion = '2.5'
-if StrictVersion(version) < StrictVersion(minFontToolsVersion):
+if False:
 	print >> sys.stderr, "ERROR:  The FontTools module version must be %s or higher.\n\
 	You have version %s installed.\n\
 	Get the latest version at https://github.com/behdad/fonttools" % (minFontToolsVersion, version)
