Index: src/FbiStuffPng.cpp
===================================================================
--- src/FbiStuffPng.cpp	(revision 557)
+++ src/FbiStuffPng.cpp	(working copy)
@@ -97,7 +97,7 @@
 	goto oops;
 
     fim_png_fp=fp;
-    h->png->read_data_fn=&fim_png_rw_ptr;
+    png_set_read_fn(h->png, NULL, &fim_png_rw_ptr);
     png_init_io(h->png, h->infile);
     png_read_info(h->png, h->info);
     png_get_IHDR(h->png, h->info, &h->w, &h->h,
