--- a/src/FbiStuffLoader.cpp
+++ b/src/FbiStuffLoader.cpp
@@ -206,7 +206,7 @@
     		FIM_FPRINTF(stream,"%s\n",buf);
 	}
 
-	string fim_loaders_to_string(void)
+	std::string fim_loaders_to_string(void)
     	{
 		fim_char_t buf[FIM_MAXLINE_BUFSIZE];	// NOTE: a larger buffer would be ok (e.g.: user configurable)...
 		fim_loaders_to_buffer(buf);
--- a/src/FbiStuffLoader.h
+++ b/src/FbiStuffLoader.h
@@ -162,7 +162,7 @@
 void fim_write_register(struct ida_writer *writer);
 void fim_loaders_to_stderr(FILE * stream);
 
-string fim_loaders_to_string(void);
+std::string fim_loaders_to_string(void);
 }
 #endif /* FIM_STUFF_LOADER_H */
 
