--- ./configure.ac.orig	2013-04-15 09:47:42.231583873 +0300
+++ ./configure.ac	2013-04-15 10:00:00.876636004 +0300
@@ -120,8 +120,8 @@
   [
 if test x"$fim_handle_graphicsmagick" = x"yes" ; then 
   FIM_LIB_OBJECTS="$FIM_LIB_OBJECTS FbiStuffMagick.o";
-  LIBS="$LIBS `GraphicsMagick-config --cppflags --ldflags --libs|tr '\n' ' '` ";
-  CXXFLAGS="$CXXFLAGS `GraphicsMagick-config --cppflags |tr '\n' ' '` ";
+  LIBS="$LIBS `pkg-config GraphicsMagick --libs` ";
+  CXXFLAGS="$CXXFLAGS `pkg-config GraphicsMagick --cflags` ";
   AC_DEFINE([HAVE_LIBGRAPHICSMAGICK],[1],[We have libGraphicsMagick])
 fi
   ], [])
