Index: configure.ac
===================================================================
--- configure.ac	(revision 110)
+++ configure.ac	(working copy)
@@ -469,7 +469,7 @@
 
 dnl ==================
 
-fim_cv_warnings=no
+fim_cv_screen=no
 AC_ARG_ENABLE(screen, AC_HELP_STRING([--enable-screen], [Fim will work under screen (EXPERIMENTAL!)]),
 [if test x$enableval = xyes; then
         fim_cv_screen=yes
