Index: configure.ac
===================================================================
--- configure.ac	(revision 255)
+++ configure.ac	(working copy)
@@ -1403,10 +1403,6 @@
 dnl ###############################################################################
 dnl this is experimental stuff
 dnl ###############################################################################
-AH_TEMPLATE([HAVE_MATRIX_MARKET_DECODER])
-AC_DEFINE([HAVE_MATRIX_MARKET_DECODER])
-CXXFLAGS="$CXXFLAGS -I${HOME}/src/libmmvbr/"
-LIBS="$LIBS -L${HOME}/src/libmmvbr/ -lmmvbr"
 dnl ###############################################################################
 
 
