--- a/src/Makefile.am	(revision 276)
+++ b/src/Makefile.am	(working copy)
@@ -18,9 +18,11 @@
 	@$(ECHO) "Applying a horrible hack to avoid the flex non-throwing isatty() redeclaration in 'extern "C" int isatty (int );'"
 	$(SED) -i 's/^.*extern.*isatty.*int.*$$//g' $@
 
-yacc.tab.hpp yacc.tab.cpp: yacc.ypp lex.lex
+yacc.tab.cpp: yacc.ypp lex.lex
 	LC_ALL=C $(YACC) $(YFLAGS) $< -o $@
 
+yacc.tab.hpp: yacc.tab.cpp
+
 FIM_LIB_OBJECTS = yacc.tab.o lex.yy.o
 
 conf.h:       fimrc
