--- ./src/FbiStuffGif.cpp.orig	2014-03-01 12:40:13.699509296 +0200
+++ ./src/FbiStuffGif.cpp	2014-03-01 12:42:46.668587469 +0200
@@ -58,14 +58,22 @@
 #if FIM_GIFLIB_RETIRED_PrintGifError  
 void
 FimPrintGifError(struct gif_state * gs) {
+#if FIM_GIFLIB_STATE_HAS_ERRCODE
     int ErrorCode = gs->gif->Error;
     /* On the basis of giflib-5.0.5/util/qprintf.c suggestion, after retirement of PrintGifError  */
+#else
+    int ErrorCode = GifError();
+#endif
     char *Err = NULL;
 
     if(ErrorCode == 0)   
 	    return;
 
+#if FIM_GIFLIB_STATE_HAS_ERRCODE
     Err = GifErrorString(ErrorCode);
+#else
+    Err = GifErrorString();
+#endif
                                                                                                                               
     if (Err != NULL)
         fprintf(stderr, "GIF-LIB error: %s.\n", Err);
