--- a/Ponomar/hp-generate.py
+++ b/Ponomar/hp-generate.py
@@ -14,7 +14,12 @@ fontforge.setPrefs ("ClearInstrsBigChanges",False )
 fontforge.setPrefs ( "CopyTTFInstrs",True )
 
 ## open up the font
-font = fontforge.open(base_name + ".sfd")
+from sys import argv
+sfd = argv[1]
+family = sfd.split('/')[0]
+base_name = family + "Unicode"
+full_name  = family + " Unicode"
+font = fontforge.open(sfd)
 
 ## Evidently, this can break Evince, so it may need to be commented out. Not sure about that, though
 font.head_optimized_for_cleartype = True
