Index: configure
===================================================================
--- configure	(revision 5743)
+++ configure	(working copy)
@@ -1554,10 +1554,10 @@
 echo "#define FFMPEG_CONFIGURATION "'"'"$FFMPEG_CONFIGURATION"'"' >> $TMPH
 
 echo "prefix=\$(DESTDIR)$prefix" >> config.mak
-echo "libdir=\$(DESTDIR)$libdir" >> config.mak
-echo "incdir=\$(DESTDIR)$incdir" >> config.mak
-echo "bindir=\$(DESTDIR)$bindir" >> config.mak
-echo "mandir=\$(DESTDIR)$mandir" >> config.mak
+echo "libdir=$libdir" >> config.mak
+echo "incdir=$incdir" >> config.mak
+echo "bindir=$bindir" >> config.mak
+echo "mandir=$mandir" >> config.mak
 echo "MAKE=$make" >> config.mak
 echo "CC=$cc" >> config.mak
 echo "AR=$ar" >> config.mak
