--- a/sources/Oswald.glyphs
+++ b/sources/Oswald.glyphs
@@ -109090,10 +109090,6 @@ weightClass = Bold;
 {
 customParameters = (
 {
-name = familyName;
-value = "Oswald Heavy";
-},
-{
 name = "Remove Glyphs";
 value = (
 "Iishorttail-cy",
@@ -109219,7 +109215,8 @@ interpolationWeight = 800;
 instanceInterpolations = {
 "3FB85DAF-C7EF-4F81-BE96-1E190958879E" = 1;
 };
-name = Regular;
+name = Heavy;
+weightClass = Heavy;
 }
 );
 kerning = {
