--- src/utils/x86_asm/cpuid.asm.orig	2006-12-11 19:58:36.000000000 +0200
+++ src/utils/x86_asm/cpuid.asm	2006-12-30 01:04:27.000000000 +0200
@@ -51,7 +51,7 @@
 %define CPUID_MMX               0x00800000
 %define CPUID_SSE               0x02000000
 %define CPUID_SSE2              0x04000000
-%define CPUID_SSE3              0x00000001
+%define CPUID_SSE3              0x00000200
 
 %define EXT_CPUID_3DNOW         0x80000000
 %define EXT_CPUID_AMD_3DNOWEXT  0x40000000
