--- build/generic/Makefile.orig	2005-05-23 12:29:43.000000000 +0300
+++ build/generic/Makefile	2007-01-19 02:35:36.000000000 +0200
@@ -140,6 +142,8 @@
 	@$(INSTALL) -d $(DESTDIR)$(libdir)
 	@echo "  I: $(libdir)/$(SHARED_LIB)"
 	@$(INSTALL) -m 644 $(BUILD_DIR)/$(SHARED_LIB) $(DESTDIR)$(libdir)/$(SHARED_LIB)
+	@ln -sf $(SHARED_LIB) $(DESTDIR)$(libdir)/libxvidcore.$(SHARED_EXTENSION).$(API_MAJOR)
+	@ln -sf $(SHARED_LIB) $(DESTDIR)$(libdir)/libxvidcore.$(SHARED_EXTENSION)
 	@echo "  I: $(libdir)/$(STATIC_LIB)"
 	@$(INSTALL) -m 644 $(BUILD_DIR)/$(STATIC_LIB) $(DESTDIR)$(libdir)/$(STATIC_LIB)
 	@echo "  D: $(includedir)"
