--- marlin/Makefile.am.orig	2006-09-10 17:13:06.000000000 +0300
+++ marlin/Makefile.am	2006-09-11 23:54:04.000000000 +0300
@@ -51,6 +51,9 @@
 	-DDATADIR=\""$(datadir)"\"		\
 	-DG_LOG_DOMAIN=\"libmarlin\"
 
+libmarlin_la_LIBADD = 				\
+	$(MARLIN_LIBS)
+
 marlin_headers =				\
 	marlin-a11y-helper.h			\
 	marlin-base-window.h			\
@@ -122,49 +125,44 @@
 	-DMARLIN_PLUGINS_DIR=\""$(plugindir)"\"	\
 	-DG_LOG_DOMAIN=\"libmarlinx\"
 
+libmarlinx_la_LIBADD =				\
+	$(MARLINX_LIBS)
+
 libmarlinincludedir = $(includedir)/libmarlin
 libmarlininclude_HEADERS = 			\
 	$(marlin_headers)
 
 test_gst_alsa_SOURCES = test-gst-alsa.c
-test_gst_alsa_LDADD =				\
-	$(MARLIN_LIBS)
 test_gst_alsa_CFLAGS = $(MARLIN_CFLAGS)
 
 test_play_pipeline_SOURCES = test-play-pipeline.c
 test_play_pipeline_LDADD =			\
-	libmarlin.la libmarlinx.la		\
-	$(MARLIN_LIBS) $(MARLINX_LIBS)
+	libmarlin.la libmarlinx.la
 test_play_pipeline_CFLAGS = $(MARLIN_CFLAGS) $(MARLINX_CFLAGS)
 
 test_sample_view_SOURCES = test-sample-view.c
 test_sample_view_LDADD =			\
-	libmarlin.la libmarlinx.la		\
-	$(MARLIN_LIBS) $(MARLINX_LIBS)
+	libmarlin.la libmarlinx.la
 test_sample_view_CFLAGS = $(MARLIN_CFLAGS) $(MARLINX_CFLAGS)
 
 test_sample_SOURCES = test-sample.c
 test_sample_LDADD =				\
-	libmarlin.la				\
-	$(MARLIN_LIBS)
+	libmarlin.la
 test_sample_CFLAGS = $(MARLIN_CFLAGS)
 
 test_splitter_SOURCES = test-splitter.c
 test_splitter_LDADD =				\
-	libmarlin.la				\
-	$(MARLIN_LIBS)
+	libmarlin.la
 test_splitter_CFLAGS = $(MARLIN_CFLAGS)
 
 test_record_pipeline_SOURCES = test-record-pipeline.c
 test_record_pipeline_LDADD =			\
-	libmarlin.la libmarlinx.la		\
-	$(MARLIN_LIBS) $(MARLINX_LIBS)
+	libmarlin.la libmarlinx.la
 test_record_pipeline_CFLAGS = $(MARLIN_CFLAGS) $(MARLINX_CFLAGS)
 
 test_save_pipeline_SOURCES = test-save-pipeline.c
 test_save_pipeline_LDADD =			\
-	libmarlin.la libmarlinx.la		\
-	$(MARLIN_LIBS) $(MARLINX_LIBS)
+	libmarlin.la libmarlinx.la
 test_save_pipeline_CFLAGS = $(MARLIN_CFLAGS) $(MARLINX_CFLAGS)
 
 marlin-marshal.h: marlin-marshal.list $(GLIB_GENMARSHAL)
