--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,6 @@ FetchContent_Declare(
   ninja
   SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/ninja-upstream
 )
-FetchContent_MakeAvailable(ninja)
 
 if(RUN_NINJA_TEST)
   add_custom_target(
@@ -28,5 +27,4 @@ if(RUN_NINJA_TEST)
   )
 endif()
 
-install(TARGETS ninja COMPONENT python DESTINATION "${SKBUILD_SCRIPTS_DIR}")
-install(FILES "${ninja_SOURCE_DIR}/misc/ninja_syntax.py" COMPONENT python DESTINATION ninja)
+install(FILES "@_NINJA_SRC_DIR_@/misc/ninja_syntax.py" COMPONENT python DESTINATION ninja)
