--- a/mp4v2/configure.ac
+++ b/mp4v2/configure.ac
@@ -19,11 +19,7 @@ m4_define([PRJ_snapshot],
     m4_bregexp(m4_esyscmd([(test "$SNAPSHOT" = "1" && date +%Y%m%d) 2>/dev/null]),[\([0-9]+\)],[\1]))
 m4_define([PRJ_repo_type],ifelse(m4_len(PRJ_snapshot),8,[snapshot],PRJ_repo_type))
 
-m4_define([PRJ_version],ifelse(
-    PRJ_repo_type,[stable],m4_format([%d.%d.%d],PRJ_version_major,PRJ_version_minor,PRJ_version_point),
-    PRJ_repo_type,[snapshot],m4_format([%d.%d-%s],PRJ_version_major,PRJ_version_minor,PRJ_snapshot),
-    PRJ_repo_type,[developer],m4_format([%d.%d-r%d],PRJ_version_major,PRJ_version_minor,PRJ_repo_rev),
-    m4_format([%d.%d-r%d],PRJ_version_major,PRJ_version_minor,PRJ_repo_rev)))
+m4_define([PRJ_version],m4_format([%d.%d.%d],PRJ_version_major,PRJ_version_minor,PRJ_version_point))
 
 ###############################################################################
 # initialization
