Only in cmus-2.2.0.orig: cmus-2.2.0
diff -ur cmus-2.2.0.orig/configure cmus-2.2.0/configure
--- cmus-2.2.0.orig/configure	2007-07-27 17:52:13.000000000 +0300
+++ cmus-2.2.0/configure	2009-07-26 16:16:43.000000000 +0300
@@ -61,7 +61,7 @@
 
 check_mpc()
 {
-	check_library MPC "" "-lmpcdec"
+	check_library MPC "" "-lmpcdecsv7"
 	return $?
 }
 
diff -ur cmus-2.2.0.orig/mpc.c cmus-2.2.0/mpc.c
--- cmus-2.2.0.orig/mpc.c	2007-07-27 17:52:13.000000000 +0300
+++ cmus-2.2.0/mpc.c	2009-07-26 16:17:32.000000000 +0300
@@ -11,7 +11,7 @@
 #include "xmalloc.h"
 #include "read_wrapper.h"
 
-#include <mpcdec/mpcdec.h>
+#include <mpcdecsv7/mpcdec.h>
 #include <inttypes.h>
 #include <errno.h>
 
@@ -25,7 +25,7 @@
 	int samples_pos;
 	int samples_avail;
 
-	/* mpcdec/mpcdec.h
+	/* mpcdecsv7/mpcdec.h
 	 *
 	 * the api doc says this is pcm samples per mpc frame
 	 * but it's really pcm _frames_ per mpc frame
